# Generated by the gRPC Python protocol compiler plugin. DO NOT EDIT!
"""Client and server classes corresponding to protobuf-defined services."""
import grpc

from google.protobuf import empty_pb2 as google_dot_protobuf_dot_empty__pb2
from google.protobuf import wrappers_pb2 as google_dot_protobuf_dot_wrappers__pb2
from ncbi.datasets.v2 import datasets_pb2 as ncbi_dot_datasets_dot_v2_dot_datasets__pb2
from ncbi.datasets.v2.reports import assembly_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__pb2
from ncbi.datasets.v2.reports import biosample_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_biosample__pb2
from ncbi.datasets.v2.reports import feature_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_feature__pb2
from ncbi.datasets.v2.reports import gene_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2
from ncbi.datasets.v2.reports import taxonomy_names_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__names__pb2
from ncbi.datasets.v2.reports import taxonomy_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__pb2
from ncbi.datasets.v2.reports import virus_annotation_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__annotation__pb2
from ncbi.datasets.v2.reports import virus_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__pb2


class BioSampleDatasetStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetBioSampleDataReports = channel.unary_unary(
                '/ncbi.datasets.v2.BioSampleDataset/GetBioSampleDataReports',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.BioSampleDatasetReportsRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_biosample__pb2.BioSampleDataReportPage.FromString,
                )


class BioSampleDatasetServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetBioSampleDataReports(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_BioSampleDatasetServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetBioSampleDataReports': grpc.unary_unary_rpc_method_handler(
                    servicer.GetBioSampleDataReports,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.BioSampleDatasetReportsRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_biosample__pb2.BioSampleDataReportPage.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.BioSampleDataset', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class BioSampleDataset(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetBioSampleDataReports(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.BioSampleDataset/GetBioSampleDataReports',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.BioSampleDatasetReportsRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_biosample__pb2.BioSampleDataReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class AssemblyDatasetStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetDownloadSummary = channel.unary_unary(
                '/ncbi.datasets.v2.AssemblyDataset/GetDownloadSummary',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
                )
        self.GetAssemblyDataReports = channel.unary_unary(
                '/ncbi.datasets.v2.AssemblyDataset/GetAssemblyDataReports',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetReportsRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__pb2.AssemblyDataReportPage.FromString,
                )
        self.GetAssemblyRevisionHistory = channel.unary_unary(
                '/ncbi.datasets.v2.AssemblyDataset/GetAssemblyRevisionHistory',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyRevisionHistoryRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyRevisionHistory.FromString,
                )
        self.GetSequenceReports = channel.unary_unary(
                '/ncbi.datasets.v2.AssemblyDataset/GetSequenceReports',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblySequenceReportsRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SequenceReportPage.FromString,
                )
        self.GetAssemblyLinks = channel.unary_unary(
                '/ncbi.datasets.v2.AssemblyDataset/GetAssemblyLinks',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyLinksRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyLinksReply.FromString,
                )
        self.GetCheckMHistogram = channel.unary_unary(
                '/ncbi.datasets.v2.AssemblyDataset/GetCheckMHistogram',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyCheckMHistogramRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyCheckMHistogramReply.FromString,
                )
        self.Sleep = channel.unary_unary(
                '/ncbi.datasets.v2.AssemblyDataset/Sleep',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepReply.FromString,
                )
        self.SleepStream = channel.unary_stream(
                '/ncbi.datasets.v2.AssemblyDataset/SleepStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepReply.FromString,
                )


class AssemblyDatasetServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetDownloadSummary(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetAssemblyDataReports(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetAssemblyRevisionHistory(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetSequenceReports(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetAssemblyLinks(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetCheckMHistogram(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def Sleep(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def SleepStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_AssemblyDatasetServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetDownloadSummary': grpc.unary_unary_rpc_method_handler(
                    servicer.GetDownloadSummary,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.SerializeToString,
            ),
            'GetAssemblyDataReports': grpc.unary_unary_rpc_method_handler(
                    servicer.GetAssemblyDataReports,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetReportsRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__pb2.AssemblyDataReportPage.SerializeToString,
            ),
            'GetAssemblyRevisionHistory': grpc.unary_unary_rpc_method_handler(
                    servicer.GetAssemblyRevisionHistory,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyRevisionHistoryRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyRevisionHistory.SerializeToString,
            ),
            'GetSequenceReports': grpc.unary_unary_rpc_method_handler(
                    servicer.GetSequenceReports,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblySequenceReportsRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SequenceReportPage.SerializeToString,
            ),
            'GetAssemblyLinks': grpc.unary_unary_rpc_method_handler(
                    servicer.GetAssemblyLinks,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyLinksRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyLinksReply.SerializeToString,
            ),
            'GetCheckMHistogram': grpc.unary_unary_rpc_method_handler(
                    servicer.GetCheckMHistogram,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyCheckMHistogramRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyCheckMHistogramReply.SerializeToString,
            ),
            'Sleep': grpc.unary_unary_rpc_method_handler(
                    servicer.Sleep,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepReply.SerializeToString,
            ),
            'SleepStream': grpc.unary_stream_rpc_method_handler(
                    servicer.SleepStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepReply.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.AssemblyDataset', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class AssemblyDataset(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetDownloadSummary(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.AssemblyDataset/GetDownloadSummary',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetAssemblyDataReports(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.AssemblyDataset/GetAssemblyDataReports',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetReportsRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__pb2.AssemblyDataReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetAssemblyRevisionHistory(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.AssemblyDataset/GetAssemblyRevisionHistory',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyRevisionHistoryRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyRevisionHistory.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetSequenceReports(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.AssemblyDataset/GetSequenceReports',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblySequenceReportsRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SequenceReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetAssemblyLinks(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.AssemblyDataset/GetAssemblyLinks',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyLinksRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyLinksReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetCheckMHistogram(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.AssemblyDataset/GetCheckMHistogram',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyCheckMHistogramRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyCheckMHistogramReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def Sleep(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.AssemblyDataset/Sleep',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def SleepStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.AssemblyDataset/SleepStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SleepReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class VirusDownloadStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetVirusDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.VirusDownload/GetVirusDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )
        self.GetSars2ProteinDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.VirusDownload/GetSars2ProteinDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )


class VirusDownloadServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetVirusDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetSars2ProteinDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_VirusDownloadServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetVirusDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetVirusDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
            'GetSars2ProteinDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetSars2ProteinDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.VirusDownload', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class VirusDownload(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetVirusDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.VirusDownload/GetVirusDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetSars2ProteinDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.VirusDownload/GetSars2ProteinDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class DatasetDownloadStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetAssemblyDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.DatasetDownload/GetAssemblyDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )
        self.GetGeneDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.DatasetDownload/GetGeneDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )
        self.GetMicroBiggeDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.DatasetDownload/GetMicroBiggeDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.MicroBiggeDatasetRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )
        self.GetProkaryoteGeneDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.DatasetDownload/GetProkaryoteGeneDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.ProkaryoteGeneRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )
        self.GetGenomeAnnotationDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.DatasetDownload/GetGenomeAnnotationDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )
        self.GetTaxonomyDatasetStream = channel.unary_stream(
                '/ncbi.datasets.v2.DatasetDownload/GetTaxonomyDatasetStream',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyDatasetRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )
        self.CheckAssemblyAvailability = channel.unary_unary(
                '/ncbi.datasets.v2.DatasetDownload/CheckAssemblyAvailability',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetAvailability.FromString,
                )


class DatasetDownloadServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetAssemblyDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetGeneDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetMicroBiggeDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetProkaryoteGeneDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetGenomeAnnotationDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetTaxonomyDatasetStream(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def CheckAssemblyAvailability(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_DatasetDownloadServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetAssemblyDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetAssemblyDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
            'GetGeneDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetGeneDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
            'GetMicroBiggeDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetMicroBiggeDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.MicroBiggeDatasetRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
            'GetProkaryoteGeneDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetProkaryoteGeneDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.ProkaryoteGeneRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
            'GetGenomeAnnotationDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetGenomeAnnotationDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
            'GetTaxonomyDatasetStream': grpc.unary_stream_rpc_method_handler(
                    servicer.GetTaxonomyDatasetStream,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyDatasetRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
            'CheckAssemblyAvailability': grpc.unary_unary_rpc_method_handler(
                    servicer.CheckAssemblyAvailability,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetAvailability.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.DatasetDownload', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class DatasetDownload(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetAssemblyDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.DatasetDownload/GetAssemblyDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetGeneDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.DatasetDownload/GetGeneDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetMicroBiggeDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.DatasetDownload/GetMicroBiggeDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.MicroBiggeDatasetRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetProkaryoteGeneDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.DatasetDownload/GetProkaryoteGeneDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.ProkaryoteGeneRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetGenomeAnnotationDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.DatasetDownload/GetGenomeAnnotationDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetTaxonomyDatasetStream(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.DatasetDownload/GetTaxonomyDatasetStream',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyDatasetRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def CheckAssemblyAvailability(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.DatasetDownload/CheckAssemblyAvailability',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.AssemblyDatasetAvailability.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class GeneDatasetStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetGeneDataReportsPaged = channel.unary_unary(
                '/ncbi.datasets.v2.GeneDataset/GetGeneDataReportsPaged',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetReportsRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2.GeneDataReportPage.FromString,
                )
        self.GetDownloadSummary = channel.unary_unary(
                '/ncbi.datasets.v2.GeneDataset/GetDownloadSummary',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
                )
        self.GetGeneCountsByTaxon = channel.unary_unary(
                '/ncbi.datasets.v2.GeneDataset/GetGeneCountsByTaxon',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneCountsByTaxonRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneCountsByTaxonReply.FromString,
                )
        self.GetOrthologSet = channel.unary_unary(
                '/ncbi.datasets.v2.GeneDataset/GetOrthologSet',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.OrthologRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2.GeneDataReportPage.FromString,
                )
        self.GetGeneLinks = channel.unary_unary(
                '/ncbi.datasets.v2.GeneDataset/GetGeneLinks',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneLinksRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneLinksReply.FromString,
                )
        self.GetChromsomeSummaryForTaxonAndAnnotation = channel.unary_unary(
                '/ncbi.datasets.v2.GeneDataset/GetChromsomeSummaryForTaxonAndAnnotation',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneChromosomeSummaryRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneChromosomeSummaryReply.FromString,
                )


class GeneDatasetServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetGeneDataReportsPaged(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetDownloadSummary(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetGeneCountsByTaxon(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetOrthologSet(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetGeneLinks(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetChromsomeSummaryForTaxonAndAnnotation(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_GeneDatasetServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetGeneDataReportsPaged': grpc.unary_unary_rpc_method_handler(
                    servicer.GetGeneDataReportsPaged,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetReportsRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2.GeneDataReportPage.SerializeToString,
            ),
            'GetDownloadSummary': grpc.unary_unary_rpc_method_handler(
                    servicer.GetDownloadSummary,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.SerializeToString,
            ),
            'GetGeneCountsByTaxon': grpc.unary_unary_rpc_method_handler(
                    servicer.GetGeneCountsByTaxon,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneCountsByTaxonRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneCountsByTaxonReply.SerializeToString,
            ),
            'GetOrthologSet': grpc.unary_unary_rpc_method_handler(
                    servicer.GetOrthologSet,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.OrthologRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2.GeneDataReportPage.SerializeToString,
            ),
            'GetGeneLinks': grpc.unary_unary_rpc_method_handler(
                    servicer.GetGeneLinks,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneLinksRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneLinksReply.SerializeToString,
            ),
            'GetChromsomeSummaryForTaxonAndAnnotation': grpc.unary_unary_rpc_method_handler(
                    servicer.GetChromsomeSummaryForTaxonAndAnnotation,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneChromosomeSummaryRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneChromosomeSummaryReply.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.GeneDataset', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class GeneDataset(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetGeneDataReportsPaged(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GeneDataset/GetGeneDataReportsPaged',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetReportsRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2.GeneDataReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetDownloadSummary(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GeneDataset/GetDownloadSummary',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneDatasetRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetGeneCountsByTaxon(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GeneDataset/GetGeneCountsByTaxon',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneCountsByTaxonRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneCountsByTaxonReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetOrthologSet(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GeneDataset/GetOrthologSet',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.OrthologRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2.GeneDataReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetGeneLinks(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GeneDataset/GetGeneLinks',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneLinksRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneLinksReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetChromsomeSummaryForTaxonAndAnnotation(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GeneDataset/GetChromsomeSummaryForTaxonAndAnnotation',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneChromosomeSummaryRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GeneChromosomeSummaryReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class VirusDatasetStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetDownloadSummary = channel.unary_unary(
                '/ncbi.datasets.v2.VirusDataset/GetDownloadSummary',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
                )
        self.GetSars2ProteinDownloadSummary = channel.unary_unary(
                '/ncbi.datasets.v2.VirusDataset/GetSars2ProteinDownloadSummary',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
                )
        self.GetVirusGenomeMetadataAsTable = channel.unary_stream(
                '/ncbi.datasets.v2.VirusDataset/GetVirusGenomeMetadataAsTable',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TabularOutput.FromString,
                )
        self.GetSars2ProteinMetadataAsTable = channel.unary_stream(
                '/ncbi.datasets.v2.VirusDataset/GetSars2ProteinMetadataAsTable',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TabularOutput.FromString,
                )
        self.GetVirusDataReports = channel.unary_unary(
                '/ncbi.datasets.v2.VirusDataset/GetVirusDataReports',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDataReportRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__pb2.VirusDataReportPage.FromString,
                )
        self.GetVirusAnnotationReports = channel.unary_unary(
                '/ncbi.datasets.v2.VirusDataset/GetVirusAnnotationReports',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAnnotationReportRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__annotation__pb2.VirusAnnotationReportPage.FromString,
                )
        self.CheckVirusAvailability = channel.unary_unary(
                '/ncbi.datasets.v2.VirusDataset/CheckVirusAvailability',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAvailabilityRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAvailability.FromString,
                )


class VirusDatasetServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetDownloadSummary(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetSars2ProteinDownloadSummary(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetVirusGenomeMetadataAsTable(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetSars2ProteinMetadataAsTable(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetVirusDataReports(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetVirusAnnotationReports(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def CheckVirusAvailability(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_VirusDatasetServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetDownloadSummary': grpc.unary_unary_rpc_method_handler(
                    servicer.GetDownloadSummary,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.SerializeToString,
            ),
            'GetSars2ProteinDownloadSummary': grpc.unary_unary_rpc_method_handler(
                    servicer.GetSars2ProteinDownloadSummary,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.SerializeToString,
            ),
            'GetVirusGenomeMetadataAsTable': grpc.unary_stream_rpc_method_handler(
                    servicer.GetVirusGenomeMetadataAsTable,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TabularOutput.SerializeToString,
            ),
            'GetSars2ProteinMetadataAsTable': grpc.unary_stream_rpc_method_handler(
                    servicer.GetSars2ProteinMetadataAsTable,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TabularOutput.SerializeToString,
            ),
            'GetVirusDataReports': grpc.unary_unary_rpc_method_handler(
                    servicer.GetVirusDataReports,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDataReportRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__pb2.VirusDataReportPage.SerializeToString,
            ),
            'GetVirusAnnotationReports': grpc.unary_unary_rpc_method_handler(
                    servicer.GetVirusAnnotationReports,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAnnotationReportRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__annotation__pb2.VirusAnnotationReportPage.SerializeToString,
            ),
            'CheckVirusAvailability': grpc.unary_unary_rpc_method_handler(
                    servicer.CheckVirusAvailability,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAvailabilityRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAvailability.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.VirusDataset', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class VirusDataset(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetDownloadSummary(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.VirusDataset/GetDownloadSummary',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetSars2ProteinDownloadSummary(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.VirusDataset/GetSars2ProteinDownloadSummary',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetVirusGenomeMetadataAsTable(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.VirusDataset/GetVirusGenomeMetadataAsTable',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDatasetRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TabularOutput.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetSars2ProteinMetadataAsTable(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.VirusDataset/GetSars2ProteinMetadataAsTable',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.Sars2ProteinDatasetRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TabularOutput.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetVirusDataReports(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.VirusDataset/GetVirusDataReports',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusDataReportRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__pb2.VirusDataReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetVirusAnnotationReports(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.VirusDataset/GetVirusAnnotationReports',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAnnotationReportRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_virus__annotation__pb2.VirusAnnotationReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def CheckVirusAvailability(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.VirusDataset/CheckVirusAvailability',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAvailabilityRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VirusAvailability.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class HttpDatafileStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.Fetch = channel.unary_stream(
                '/ncbi.datasets.v2.HttpDatafile/Fetch',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.MethodPayloadRequest.SerializeToString,
                response_deserializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
                )


class HttpDatafileServicer(object):
    """Missing associated documentation comment in .proto file."""

    def Fetch(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_HttpDatafileServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'Fetch': grpc.unary_stream_rpc_method_handler(
                    servicer.Fetch,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.MethodPayloadRequest.FromString,
                    response_serializer=google_dot_protobuf_dot_wrappers__pb2.BytesValue.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.HttpDatafile', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class HttpDatafile(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def Fetch(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_stream(request, target, '/ncbi.datasets.v2.HttpDatafile/Fetch',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.MethodPayloadRequest.SerializeToString,
            google_dot_protobuf_dot_wrappers__pb2.BytesValue.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class TaxonomyDatasetStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetOrganismsThatMatch = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyDataset/GetOrganismsThatMatch',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.OrganismQueryRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SciNameAndIds.FromString,
                )
        self.GetTaxonomyLinks = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyDataset/GetTaxonomyLinks',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyLinksRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyLinksResponse.FromString,
                )
        self.GetTaxonomyImage = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyDataset/GetTaxonomyImage',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.HttpBody.FromString,
                )
        self.GetTaxonomyImageMetadata = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyDataset/GetTaxonomyImageMetadata',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageMetadataRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageMetadataResponse.FromString,
                )


class TaxonomyDatasetServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetOrganismsThatMatch(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetTaxonomyLinks(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetTaxonomyImage(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetTaxonomyImageMetadata(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_TaxonomyDatasetServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetOrganismsThatMatch': grpc.unary_unary_rpc_method_handler(
                    servicer.GetOrganismsThatMatch,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.OrganismQueryRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SciNameAndIds.SerializeToString,
            ),
            'GetTaxonomyLinks': grpc.unary_unary_rpc_method_handler(
                    servicer.GetTaxonomyLinks,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyLinksRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyLinksResponse.SerializeToString,
            ),
            'GetTaxonomyImage': grpc.unary_unary_rpc_method_handler(
                    servicer.GetTaxonomyImage,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.HttpBody.SerializeToString,
            ),
            'GetTaxonomyImageMetadata': grpc.unary_unary_rpc_method_handler(
                    servicer.GetTaxonomyImageMetadata,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageMetadataRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageMetadataResponse.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.TaxonomyDataset', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class TaxonomyDataset(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetOrganismsThatMatch(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyDataset/GetOrganismsThatMatch',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.OrganismQueryRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.SciNameAndIds.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetTaxonomyLinks(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyDataset/GetTaxonomyLinks',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyLinksRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyLinksResponse.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetTaxonomyImage(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyDataset/GetTaxonomyImage',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.HttpBody.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetTaxonomyImageMetadata(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyDataset/GetTaxonomyImageMetadata',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageMetadataRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyImageMetadataResponse.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class GenomeAnnotationDatasetStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetGenomeAnnotations = channel.unary_unary(
                '/ncbi.datasets.v2.GenomeAnnotationDataset/GetGenomeAnnotations',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_feature__pb2.GenomeAnnotationReportPage.FromString,
                )
        self.GetGenomeAnnotationSummary = channel.unary_unary(
                '/ncbi.datasets.v2.GenomeAnnotationDataset/GetGenomeAnnotationSummary',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationTableSummaryReply.FromString,
                )
        self.GetDownloadSummary = channel.unary_unary(
                '/ncbi.datasets.v2.GenomeAnnotationDataset/GetDownloadSummary',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
                )


class GenomeAnnotationDatasetServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetGenomeAnnotations(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetGenomeAnnotationSummary(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetDownloadSummary(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_GenomeAnnotationDatasetServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetGenomeAnnotations': grpc.unary_unary_rpc_method_handler(
                    servicer.GetGenomeAnnotations,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_feature__pb2.GenomeAnnotationReportPage.SerializeToString,
            ),
            'GetGenomeAnnotationSummary': grpc.unary_unary_rpc_method_handler(
                    servicer.GetGenomeAnnotationSummary,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationTableSummaryReply.SerializeToString,
            ),
            'GetDownloadSummary': grpc.unary_unary_rpc_method_handler(
                    servicer.GetDownloadSummary,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.GenomeAnnotationDataset', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class GenomeAnnotationDataset(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetGenomeAnnotations(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GenomeAnnotationDataset/GetGenomeAnnotations',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_feature__pb2.GenomeAnnotationReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetGenomeAnnotationSummary(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GenomeAnnotationDataset/GetGenomeAnnotationSummary',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationTableSummaryReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetDownloadSummary(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.GenomeAnnotationDataset/GetDownloadSummary',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.GenomeAnnotationRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.DownloadSummary.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class TaxonomyCppDatasetStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.GetTaxonomyMetadataFor = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyCppDataset/GetTaxonomyMetadataFor',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataResponse.FromString,
                )
        self.GetTaxonomyDataReportFor = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyCppDataset/GetTaxonomyDataReportFor',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__pb2.TaxonomyDataReportPage.FromString,
                )
        self.GetTaxonomyNamesDataReportFor = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyCppDataset/GetTaxonomyNamesDataReportFor',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__names__pb2.TaxonomyNamesDataReportPage.FromString,
                )
        self.GetRelatedTaxids = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyCppDataset/GetRelatedTaxids',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyRelatedIdRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyTaxIdsPage.FromString,
                )
        self.GetMinimumFilteredSubtree = channel.unary_unary(
                '/ncbi.datasets.v2.TaxonomyCppDataset/GetMinimumFilteredSubtree',
                request_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyFilteredSubtreeRequest.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyFilteredSubtreeResponse.FromString,
                )


class TaxonomyCppDatasetServicer(object):
    """Missing associated documentation comment in .proto file."""

    def GetTaxonomyMetadataFor(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetTaxonomyDataReportFor(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetTaxonomyNamesDataReportFor(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetRelatedTaxids(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')

    def GetMinimumFilteredSubtree(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_TaxonomyCppDatasetServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'GetTaxonomyMetadataFor': grpc.unary_unary_rpc_method_handler(
                    servicer.GetTaxonomyMetadataFor,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataResponse.SerializeToString,
            ),
            'GetTaxonomyDataReportFor': grpc.unary_unary_rpc_method_handler(
                    servicer.GetTaxonomyDataReportFor,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__pb2.TaxonomyDataReportPage.SerializeToString,
            ),
            'GetTaxonomyNamesDataReportFor': grpc.unary_unary_rpc_method_handler(
                    servicer.GetTaxonomyNamesDataReportFor,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__names__pb2.TaxonomyNamesDataReportPage.SerializeToString,
            ),
            'GetRelatedTaxids': grpc.unary_unary_rpc_method_handler(
                    servicer.GetRelatedTaxids,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyRelatedIdRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyTaxIdsPage.SerializeToString,
            ),
            'GetMinimumFilteredSubtree': grpc.unary_unary_rpc_method_handler(
                    servicer.GetMinimumFilteredSubtree,
                    request_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyFilteredSubtreeRequest.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyFilteredSubtreeResponse.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.TaxonomyCppDataset', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class TaxonomyCppDataset(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def GetTaxonomyMetadataFor(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyCppDataset/GetTaxonomyMetadataFor',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataResponse.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetTaxonomyDataReportFor(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyCppDataset/GetTaxonomyDataReportFor',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__pb2.TaxonomyDataReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetTaxonomyNamesDataReportFor(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyCppDataset/GetTaxonomyNamesDataReportFor',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyMetadataRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_reports_dot_taxonomy__names__pb2.TaxonomyNamesDataReportPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetRelatedTaxids(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyCppDataset/GetRelatedTaxids',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyRelatedIdRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyTaxIdsPage.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)

    @staticmethod
    def GetMinimumFilteredSubtree(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.TaxonomyCppDataset/GetMinimumFilteredSubtree',
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyFilteredSubtreeRequest.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.TaxonomyFilteredSubtreeResponse.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)


class GatewayStub(object):
    """Missing associated documentation comment in .proto file."""

    def __init__(self, channel):
        """Constructor.

        Args:
            channel: A grpc.Channel.
        """
        self.Version = channel.unary_unary(
                '/ncbi.datasets.v2.Gateway/Version',
                request_serializer=google_dot_protobuf_dot_empty__pb2.Empty.SerializeToString,
                response_deserializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VersionReply.FromString,
                )


class GatewayServicer(object):
    """Missing associated documentation comment in .proto file."""

    def Version(self, request, context):
        """Missing associated documentation comment in .proto file."""
        context.set_code(grpc.StatusCode.UNIMPLEMENTED)
        context.set_details('Method not implemented!')
        raise NotImplementedError('Method not implemented!')


def add_GatewayServicer_to_server(servicer, server):
    rpc_method_handlers = {
            'Version': grpc.unary_unary_rpc_method_handler(
                    servicer.Version,
                    request_deserializer=google_dot_protobuf_dot_empty__pb2.Empty.FromString,
                    response_serializer=ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VersionReply.SerializeToString,
            ),
    }
    generic_handler = grpc.method_handlers_generic_handler(
            'ncbi.datasets.v2.Gateway', rpc_method_handlers)
    server.add_generic_rpc_handlers((generic_handler,))


 # This class is part of an EXPERIMENTAL API.
class Gateway(object):
    """Missing associated documentation comment in .proto file."""

    @staticmethod
    def Version(request,
            target,
            options=(),
            channel_credentials=None,
            call_credentials=None,
            insecure=False,
            compression=None,
            wait_for_ready=None,
            timeout=None,
            metadata=None):
        return grpc.experimental.unary_unary(request, target, '/ncbi.datasets.v2.Gateway/Version',
            google_dot_protobuf_dot_empty__pb2.Empty.SerializeToString,
            ncbi_dot_datasets_dot_v2_dot_datasets__pb2.VersionReply.FromString,
            options, channel_credentials,
            insecure, call_credentials, compression, wait_for_ready, timeout, metadata)
