# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v2/datasets.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from google.protobuf import timestamp_pb2 as google_dot_protobuf_dot_timestamp__pb2
from ncbi.datasets.options import openapi_pb2 as ncbi_dot_datasets_dot_options_dot_openapi__pb2
from ncbi.datasets.options import request_pb2 as ncbi_dot_datasets_dot_options_dot_request__pb2
from ncbi.datasets.v2.reports import assembly_revision_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__revision__pb2
from ncbi.datasets.v2.reports import assembly_sequence_info_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_assembly__sequence__info__pb2
from ncbi.datasets.v2.reports import common_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\x1fncbi/datasets/v2/datasets.proto\x12\x10ncbi.datasets.v2\x1a\x1fgoogle/protobuf/timestamp.proto\x1a#ncbi/datasets/options/openapi.proto\x1a#ncbi/datasets/options/request.proto\x1a\x30ncbi/datasets/v2/reports/assembly_revision.proto\x1a\x35ncbi/datasets/v2/reports/assembly_sequence_info.proto\x1a%ncbi/datasets/v2/reports/common.proto\"H\n\x14MethodPayloadRequest\x12\x16\n\x06method\x18\x01 \x01(\tR\x06method\x12\x18\n\x07payload\x18\x02 \x01(\tR\x07payload\"\xb2\x06\n\x1e\x41ssemblySequenceReportsRequest\x12X\n\taccession\x18\x01 \x01(\tB:\xc2\xf0\x19\x36\x12\x19\n\x10GCF_000001635.27\x12\x05Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\taccession\x12s\n\x0b\x63hromosomes\x18\x02 \x03(\tBQ\xc2\xf0\x19M\x12,\n\x01\x31\n\x01\x32\n\x01\x33\n\x01X\n\x01Y\n\x02MT\x12\x17Selected chromosomes #1\x12\x1d\n\x01X\n\x01Y\x12\x15Human sex chromosomesR\x0b\x63hromosomes\x12\xaf\x01\n\x0crole_filters\x18\x04 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12T\n\x12\x61ssembled-molecule\n\x14unlocalized-scaffold\n\x11unplaced-scaffold\x12\x15\x41ll types of sequence\x12.\n\x12\x61ssembled-molecule\x12\x18Only assembled moleculesR\x0broleFilters\x12\x82\x01\n\x0ctable_fields\x18\x03 \x03(\tB_\xc2\xf0\x19[\x12;\n\taccession\n\x08\x63hr-name\x12$Genome Accession and Chromosome Name\x12\x1c\n\x0fucsc-style-name\x12\tUCSC NameR\x0btableFields\x12\x43\n\x17\x63ount_assembly_unplaced\x18\x05 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\x15\x63ountAssemblyUnplaced\x12%\n\tpage_size\x18\x08 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\t \x01(\tR\tpageToken\x12\\\n\x16include_tabular_header\x18\x0c \x01(\x0e\x32&.ncbi.datasets.v2.IncludeTabularHeaderR\x14includeTabularHeader\x12!\n\x0ctable_format\x18\r \x01(\tR\x0btableFormat\"\xb1\x02\n\x12SequenceReportPage\x12@\n\x07reports\x18\x01 \x03(\x0b\x32&.ncbi.datasets.v2.reports.SequenceInfoR\x07reports\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\x12&\n\x0fnext_page_token\x18\x04 \x01(\tR\rnextPageToken\x12!\n\x0c_report_type\x18\xf4\x03 \x01(\tR\nReportType\x12%\n\x0e_report_fields\x18\xf5\x03 \x03(\tR\x0cReportFields\x12\x1f\n\x0b_first_page\x18\xf6\x03 \x01(\x08R\tFirstPage\x12%\n\x0e_report_format\x18\xf7\x03 \x01(\tR\x0cReportFormat\"\xac\x06\n\tSortField\x12\xdf\x05\n\x05\x66ield\x18\x01 \x01(\tB\xc8\x05\xc2\xf0\x19\xc3\x05\x12\x0b\n\taccession\x12\x17\n\x15versionless-accession\x12\r\n\x0bsearch-text\x12\x08\n\x06tax-id\x12\x13\n\x11tax-ordinal-range\x12\x0f\n\rorganism-name\x12\x0b\n\tsubmitter\x12\x15\n\x13\x61ssembly-level-enum\x12\x11\n\x0f\x61ssembly-status\x12\x0e\n\x0crelease-date\x12\x0f\n\ris-ref-genome\x12\x0f\n\ris-rep-genome\x12\x0b\n\tis-refseq\x12\x0c\n\nis-primary\x12\x0c\n\nis-current\x12\x1c\n\x1apaired-assembly-is-current\x12!\n\x1fpaired-assembly-annotation-name\x12\x10\n\x0ehas-annotation\x12\r\n\x0bis-atypical\x12\x16\n\x14\x62ioproject-accession\x12\x17\n\x15total-sequence-length\x12\x11\n\x0fpair-is-current\x12\x1d\n\x1bpair-has-genbank-annotation\x12\x1c\n\x1apair-has-refseq-annotation\x12\x15\n\x13pair-has-annotation\x12\x0c\n\ncontig-n50\x12\x0e\n\x0cscaffold-n50\x12\x10\n\x0e\x62usco-complete\x12\x13\n\x11genbank-accession\x12\x12\n\x10refseq-accession\x12\x11\n\x0f\x61nnotation-name\x12\x0f\n\rwgs-accession\x12\x11\n\x0fsequencing-tech\x12\x15\n\x13\x62iosample-accession\x12\x0c\n\ngene-count\x12\x0f\n\rprotein-count\x12\n\n\x08modifierR\x05\x66ield\x12=\n\tdirection\x18\x02 \x01(\x0e\x32\x1f.ncbi.datasets.v2.SortDirectionR\tdirection\"n\n\x1e\x42ioSampleDatasetReportsRequest\x12L\n\naccessions\x18\x01 \x03(\tB,\xc2\xf0\x19(\x12&\n\x0eSAMEA103927779\x12\x14\x42ioSample IdentifierR\naccessions\"\xf1\x08\n\x1d\x41ssemblyDatasetReportsRequest\x12\x8c\x01\n\x06taxons\x18\x06 \x03(\tBt\xc2\xf0\x19p\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12+\n\nchimpanzee\n\x0f\x65\x61stern gorilla\x12\x0c\x43ommon Names\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x06taxons\x12u\n\x0b\x62ioprojects\x18\x08 \x03(\tBS\xc2\xf0\x19O\x12)\n\x0bPRJNA489243\x12\x1aVertebrate Genomes Project\x12\"\n\nPRJNA31257\x12\x14Human Genome ProjectR\x0b\x62ioprojects\x12v\n\naccessions\x18\x01 \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\x12L\n\x07\x66ilters\x18\x07 \x01(\x0b\x32\x32.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilterR\x07\x66ilters\x12\x33\n\x0ftax_exact_match\x18\x0e \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\rtaxExactMatch\x12$\n\x0b\x63hromosomes\x18\x02 \x03(\tB\x02\x18\x01R\x0b\x63hromosomes\x12\x8d\x01\n\x0ctable_fields\x18\x03 \x03(\tBj\xc2\xf0\x19\x66\x12>\n\x12\x61ssminfo-accession\n\rassminfo-name\x12\x19Genome Accession and Name\x12$\n\x12\x61ssminfo-accession\x12\x0eJust accessionR\x0btableFields\x12\x66\n\x10returned_content\x18\n \x01(\x0e\x32;.ncbi.datasets.v2.AssemblyDatasetReportsRequest.ContentTypeR\x0freturnedContent\x12%\n\tpage_size\x18\x04 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\x05 \x01(\tR\tpageToken\x12/\n\x04sort\x18\x0b \x03(\x0b\x32\x1b.ncbi.datasets.v2.SortFieldR\x04sort\x12\\\n\x16include_tabular_header\x18\x0c \x01(\x0e\x32&.ncbi.datasets.v2.IncludeTabularHeaderR\x14includeTabularHeader\x12!\n\x0ctable_format\x18\r \x01(\tR\x0btableFormat\"9\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08\x41SSM_ACC\x10\x01\x12\x0e\n\nPAIRED_ACC\x10\x02\"\x95\x11\n AssemblyDatasetDescriptorsFilter\x12\x9d\x01\n\x0ereference_only\x18\x01 \x01(\x08\x42v\xc2\xf0\x19r\n\x05\x66\x61lse\x12\x33\n\x04true\x12+Limit to reference & representative genomes\x12\x34\n\x05\x66\x61lse\x12+Include reference and non-reference genomesR\rreferenceOnly\x12\xd1\x01\n\x0f\x61ssembly_source\x18\x03 \x01(\x0e\x32\x41.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.AssemblySourceBe\xc2\xf0\x19\x61\n\x03\x61ll\x12!\n\x06refseq\x12\x17Limit to RefSeq genomes\x12#\n\x07genbank\x12\x18Limit to GenBank genomes\x12\x12\n\x03\x61ll\x12\x0b\x41ll genomesR\x0e\x61ssemblySource\x12y\n\x0ehas_annotation\x18\x07 \x01(\x08\x42R\xc2\xf0\x19N\n\x05\x66\x61lse\x12%\n\x04true\x12\x1dLimit to annotated assemblies\x12\x1e\n\x05\x66\x61lse\x12\x15Return all assembliesR\rhasAnnotation\x12\x96\x01\n\x16\x65xclude_paired_reports\x18\t \x01(\x08\x42`\xc2\xf0\x19\\\n\x05\x66\x61lse\x12\x1e\n\x05\x66\x61lse\x12\x15Return all assemblies\x12\x33\n\x04true\x12+Only return single record for GCA/GCF pairsR\x14\x65xcludePairedReports\x12\xc7\x01\n\x10\x65xclude_atypical\x18\x0b \x01(\x08\x42\x9b\x01\xc2\xf0\x19\x96\x01\n\x05\x66\x61lse\x12\x46\n\x05\x66\x61lse\x12=Return all selected genomes, including ones that are atypical\x12\x45\n\x04true\x12=Only return genomes that have not been classified as atypicalR\x0f\x65xcludeAtypical\x12\xf5\x01\n\x10\x61ssembly_version\x18\n \x01(\x0e\x32\x42.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.AssemblyVersionB\x85\x01\xc2\xf0\x19\x80\x01\n\x07\x63urrent\x12)\n\x07\x63urrent\x12\x1eOnly return current assemblies\x12J\n\x0e\x61ll_assemblies\x12\x38Return all assemblies, including suppressed and replacedR\x0f\x61ssemblyVersion\x12\xd2\x01\n\x0e\x61ssembly_level\x18\x04 \x03(\x0e\x32\'.ncbi.datasets.v2.reports.AssemblyLevelB\x81\x01\xc2\xf0\x19}\x12=\n\nchromosome\n\x0f\x63omplete_genome\x12\x1e\x43hromosome and complete genome\x12<\n\x08scaffold\n\nchromosome\n\x0f\x63omplete_genome\x12\x13Scaffold and higherR\rassemblyLevel\x12j\n\x12\x66irst_release_date\x18\x05 \x01(\x0b\x32\x1a.google.protobuf.TimestampB \xc2\xf0\x19\x1c\x12\x1a\n\n01/10/2015\x12\x0cJan 10, 2015R\x10\x66irstReleaseDate\x12h\n\x11last_release_date\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB \xc2\xf0\x19\x1c\x12\x1a\n\n01/10/2021\x12\x0cJan 10, 2021R\x0flastReleaseDate\x12\x89\x01\n\x0bsearch_text\x18\x08 \x03(\tBh\xc2\xf0\x19\x64\x12:\n\x1bGenome Reference Consortium\x12\x1bGenome Reference Consortium\x12\x10\n\x06GRCh38\x12\x06GRCh38\x12\x14\n\x08\x43\x35\x37\x42L/6J\x12\x08\x43\x35\x37\x42L/6JR\nsearchText\x12\x91\x02\n\x15is_metagenome_derived\x18\x0c \x01(\x0e\x32J.ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter.MetagenomeDerivedFilterB\x90\x01\xc2\xf0\x19\x8b\x01\x12\x44\n\x17metagenome_derived_only\x12)Return only metagenome derived assemblies\x12\x43\n\x1ametagenome_derived_exclude\x12%Exclude metagenome derived assembliesR\x13isMetagenomeDerived\"2\n\x0e\x41ssemblySource\x12\x07\n\x03\x61ll\x10\x00\x12\n\n\x06refseq\x10\x01\x12\x0b\n\x07genbank\x10\x02\"2\n\x0f\x41ssemblyVersion\x12\x0b\n\x07\x63urrent\x10\x00\x12\x12\n\x0e\x61ll_assemblies\x10\x01\"t\n\x17MetagenomeDerivedFilter\x12\x1c\n\x18METAGENOME_DERIVED_UNSET\x10\x00\x12\x1b\n\x17metagenome_derived_only\x10\x01\x12\x1e\n\x1ametagenome_derived_exclude\x10\x02\"\x84\x01\n\nAccessions\x12v\n\naccessions\x18\x01 \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\"\x95\x01\n\x17\x41ssemblyRevisionHistory\x12Y\n\x12\x61ssembly_revisions\x18\x01 \x03(\x0b\x32*.ncbi.datasets.v2.reports.AssemblyRevisionR\x11\x61ssemblyRevisions\x12\x1f\n\x0btotal_count\x18\x02 \x01(\rR\ntotalCount\"_\n\x1e\x41ssemblyRevisionHistoryRequest\x12=\n\taccession\x18\x01 \x01(\tB\x1f\xc2\xf0\x19\x1b\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\taccession\"l\n\x1e\x41ssemblyCheckMHistogramRequest\x12J\n\rspecies_taxon\x18\x01 \x01(\tB%\xc2\xf0\x19!\x12\x1f\n\x06\x32\x30\x32\x39\x35\x36\x12\x15\x41\x63inetobacter towneriR\x0cspeciesTaxon\"\xc0\x02\n\x1c\x41ssemblyCheckMHistogramReply\x12J\n\rspecies_taxid\x18\x01 \x01(\rB%\xc2\xf0\x19!\x12\x1f\n\x06\x32\x30\x32\x39\x35\x36\x12\x15\x41\x63inetobacter towneriR\x0cspeciesTaxid\x12q\n\x13histogram_intervals\x18\x02 \x03(\x0b\x32@.ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramIntervalR\x12histogramIntervals\x1a\x61\n\x11HistogramInterval\x12\x1b\n\tstart_pos\x18\x01 \x01(\x02R\x08startPos\x12\x19\n\x08stop_pos\x18\x02 \x01(\x02R\x07stopPos\x12\x14\n\x05\x63ount\x18\x03 \x01(\x02R\x05\x63ount\"\x98\x01\n\nBioProject\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12\x14\n\x05title\x18\x02 \x01(\tR\x05title\x12)\n\x10parent_accession\x18\x03 \x01(\tR\x0fparentAccession\x12+\n\x11parent_accessions\x18\x04 \x03(\tR\x10parentAccessions\"S\n\x11\x42ioProjectLineage\x12>\n\x0b\x62ioprojects\x18\x01 \x03(\x0b\x32\x1c.ncbi.datasets.v2.BioProjectR\x0b\x62ioprojects\"\x85\x02\n\tBuscoStat\x12#\n\rbusco_lineage\x18\x01 \x01(\tR\x0c\x62uscoLineage\x12\x1b\n\tbusco_ver\x18\x02 \x01(\tR\x08\x62uscoVer\x12\x1a\n\x08\x63omplete\x18\x03 \x01(\x02R\x08\x63omplete\x12\x1f\n\x0bsingle_copy\x18\x04 \x01(\x02R\nsingleCopy\x12\x1e\n\nduplicated\x18\x05 \x01(\x02R\nduplicated\x12\x1e\n\nfragmented\x18\x06 \x01(\x02R\nfragmented\x12\x18\n\x07missing\x18\x07 \x01(\x02R\x07missing\x12\x1f\n\x0btotal_count\x18\x08 \x01(\x04R\ntotalCount\"\x94\x05\n\x15\x41nnotationForAssembly\x12\x12\n\x04name\x18\x01 \x01(\tR\x04name\x12\x16\n\x06source\x18\x02 \x01(\tR\x06source\x12%\n\x0erelease_number\x18\x03 \x01(\tR\rreleaseNumber\x12!\n\x0crelease_date\x18\x04 \x01(\tR\x0breleaseDate\x12\x1d\n\nreport_url\x18\x05 \x01(\tR\treportUrl\x12@\n\x04\x66ile\x18\x06 \x03(\x0b\x32,.ncbi.datasets.v2.AnnotationForAssembly.FileR\x04\x66ile\x12\x35\n\x05stats\x18\x07 \x01(\x0b\x32\x1f.ncbi.datasets.v2.FeatureCountsR\x05stats\x12\x31\n\x05\x62usco\x18\x08 \x01(\x0b\x32\x1b.ncbi.datasets.v2.BuscoStatR\x05\x62usco\x1ao\n\x04\x46ile\x12@\n\x04type\x18\x01 \x01(\x0e\x32,.ncbi.datasets.v2.AnnotationForAssembly.TypeR\x04type\x12%\n\x0e\x65stimated_size\x18\x02 \x01(\x04R\restimatedSize\"\xc8\x01\n\x04Type\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x0e\n\nGENOME_GFF\x10\x01\x12\x0f\n\x0bGENOME_GBFF\x10\x02\x12\x11\n\tGENOME_GB\x10\x02\x1a\x02\x08\x01\x12\r\n\tRNA_FASTA\x10\x03\x12\x0e\n\nPROT_FASTA\x10\x04\x12\x0e\n\nGENOME_GTF\x10\x05\x12\r\n\tCDS_FASTA\x10\x06\x12\x16\n\nASM_STRUCT\x10\x07\x1a\x06\xc2\xf0\x19\x02\x08\x01\x12\x10\n\x0cGENOME_FASTA\x10\x08\x12\x13\n\x0fSEQUENCE_REPORT\x10\t\x1a\x02\x10\x01\"N\n\rFeatureCounts\x12=\n\x0bgene_counts\x18\x01 \x01(\x0b\x32\x1c.ncbi.datasets.v2.GeneCountsR\ngeneCounts\"I\n\nGeneCounts\x12\x14\n\x05total\x18\x01 \x01(\rR\x05total\x12%\n\x0eprotein_coding\x18\x02 \x01(\rR\rproteinCoding\"\x8b\x06\n\x16\x41ssemblyDatasetRequest\x12v\n\naccessions\x18\n \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions\x12s\n\x0b\x63hromosomes\x18\x02 \x03(\tBQ\xc2\xf0\x19M\x12,\n\x01\x31\n\x01\x32\n\x01\x33\n\x01X\n\x01Y\n\x02MT\x12\x17Selected chromosomes #1\x12\x1d\n\x01X\n\x01Y\x12\x15Human sex chromosomesR\x0b\x63hromosomes\x12\xa0\x02\n\x17include_annotation_type\x18\x08 \x03(\x0e\x32,.ncbi.datasets.v2.AnnotationForAssembly.TypeB\xb9\x01\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\xae\x01\x12M\n\x0cGENOME_FASTA\n\nPROT_FASTA\n\tRNA_FASTA\x12&Genomic, Transcript & Protein Sequence\x12]\n\nGENOME_GFF\n\x0bGENOME_GBFF\n\nGENOME_GTF\n\nPROT_FASTA\n\tRNA_FASTA\n\x1a\x43\x44S_FASTA, SEQUENCE_REPORT\x12\x03\x41llR\x15includeAnnotationType\x12O\n\x08hydrated\x18\t \x01(\x0e\x32\x33.ncbi.datasets.v2.AssemblyDatasetRequest.ResolutionR\x08hydrated\x12\x32\n\x0binclude_tsv\x18\x0b \x01(\x08\x42\x11\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\x07\n\x05\x66\x61lseR\nincludeTsv\"6\n\nResolution\x12\x12\n\x0e\x46ULLY_HYDRATED\x10\x00\x12\x14\n\x10\x44\x41TA_REPORT_ONLY\x10\x01:$\xc2\xf0\x19 \n\x1e\x61\x63\x63\x65ssions:\n- GCF_000001405.40\"\x8f\x01\n\x1b\x41ssemblyDatasetAvailability\x12)\n\x10valid_assemblies\x18\x01 \x03(\tR\x0fvalidAssemblies\x12-\n\x12invalid_assemblies\x18\x02 \x03(\tR\x11invalidAssemblies\x12\x16\n\x06reason\x18\x03 \x01(\tR\x06reason\"\x94\x08\n\x08Organism\x12\x15\n\x06tax_id\x18\x01 \x01(\x04R\x05taxId\x12\x19\n\x08sci_name\x18\x02 \x01(\tR\x07sciName\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12\x1d\n\nblast_node\x18\x10 \x01(\x08R\tblastNode\x12\x14\n\x05\x62reed\x18\n \x01(\tR\x05\x62reed\x12\x1a\n\x08\x63ultivar\x18\x0b \x01(\tR\x08\x63ultivar\x12\x18\n\x07\x65\x63otype\x18\x0c \x01(\tR\x07\x65\x63otype\x12\x18\n\x07isolate\x18\r \x01(\tR\x07isolate\x12\x10\n\x03sex\x18\x0e \x01(\tR\x03sex\x12\x16\n\x06strain\x18\x0f \x01(\tR\x06strain\x12\x1f\n\x0bsearch_text\x18\x11 \x03(\tR\nsearchText\x12\x36\n\x04rank\x18\x05 \x01(\x0e\x32\".ncbi.datasets.v2.reports.RankTypeR\x04rank\x12\"\n\rparent_tax_id\x18\x06 \x01(\x04R\x0bparentTaxId\x12)\n\x0e\x61ssembly_count\x18\x07 \x01(\x04\x42\x02\x18\x01R\rassemblyCount\x12J\n\x0f\x61ssembly_counts\x18\x13 \x01(\x0b\x32!.ncbi.datasets.v2.Organism.CountsR\x0e\x61ssemblyCounts\x12>\n\x06\x63ounts\x18\x16 \x03(\x0b\x32&.ncbi.datasets.v2.Organism.CountByTypeR\x06\x63ounts\x12\x36\n\x08\x63hildren\x18\x08 \x03(\x0b\x32\x1a.ncbi.datasets.v2.OrganismR\x08\x63hildren\x12\x32\n\x06merged\x18\t \x03(\x0b\x32\x1a.ncbi.datasets.v2.OrganismR\x06merged\x12$\n\x0emerged_tax_ids\x18\x12 \x03(\x04R\x0cmergedTaxIds\x12\x17\n\x07min_ord\x18\x14 \x01(\rR\x06minOrd\x12\x17\n\x07max_ord\x18\x15 \x01(\rR\x06maxOrd\x12\x16\n\x06weight\x18\x17 \x01(\rR\x06weight\x12\x10\n\x03key\x18\x64 \x01(\x04R\x03key\x12\x14\n\x05title\x18\x65 \x01(\tR\x05title\x12\x12\n\x04icon\x18\x66 \x01(\x08R\x04icon\x1a\x36\n\x06\x43ounts\x12\x12\n\x04node\x18\x01 \x01(\rR\x04node\x12\x18\n\x07subtree\x18\x02 \x01(\rR\x07subtree\x1a\x81\x01\n\x0b\x43ountByType\x12\x37\n\x04type\x18\x01 \x01(\x0e\x32#.ncbi.datasets.v2.reports.CountTypeR\x04type\x12\x39\n\x06\x63ounts\x18\x02 \x01(\x0b\x32!.ncbi.datasets.v2.Organism.CountsR\x06\x63ounts\"\xd3\x04\n\x14OrganismQueryRequest\x12+\n\x0eorganism_query\x18\x01 \x01(\tB\x02\x18\x01H\x00R\rorganismQuery\x12\x33\n\x0btaxon_query\x18\x02 \x01(\tB\x10\xc2\xf0\x19\x0c\x12\n\n\x03hum\x12\x03humH\x00R\ntaxonQuery\x12k\n\x0ftax_rank_filter\x18\x03 \x01(\x0e\x32\x34.ncbi.datasets.v2.OrganismQueryRequest.TaxRankFilterB\r\xc2\xf0\x19\t\n\x07speciesR\rtaxRankFilter\x12\x8f\x01\n\x15taxon_resource_filter\x18\x04 \x01(\x0e\x32:.ncbi.datasets.v2.OrganismQueryRequest.TaxonResourceFilterB\x1f\xc2\xf0\x19\x1b\n\x19TAXON_RESOURCE_FILTER_ALLR\x13taxonResourceFilter\x12,\n\x0b\x65xact_match\x18\x05 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\nexactMatch\".\n\rTaxRankFilter\x12\x0b\n\x07species\x10\x00\x12\x10\n\x0chigher_taxon\x10\x01\"v\n\x13TaxonResourceFilter\x12\x1d\n\x19TAXON_RESOURCE_FILTER_ALL\x10\x00\x12 \n\x1cTAXON_RESOURCE_FILTER_GENOME\x10\x01\x12\x1e\n\x1aTAXON_RESOURCE_FILTER_GENE\x10\x02\x42\x04\n\x02id\"\xc4\x02\n\rSciNameAndIds\x12U\n\x10sci_name_and_ids\x18\x01 \x03(\x0b\x32,.ncbi.datasets.v2.SciNameAndIds.SciNameAndIdR\rsciNameAndIds\x1a\xdb\x01\n\x0cSciNameAndId\x12\x19\n\x08sci_name\x18\x01 \x01(\tR\x07sciName\x12\x15\n\x06tax_id\x18\x02 \x01(\x04R\x05taxId\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12!\n\x0cmatched_term\x18\x04 \x01(\tR\x0bmatchedTerm\x12\x36\n\x04rank\x18\x05 \x01(\x0e\x32\".ncbi.datasets.v2.reports.RankTypeR\x04rank\x12\x1d\n\ngroup_name\x18\x06 \x01(\tR\tgroupName\"\xb8\x03\n\x0fOrthologRequest\x12\x35\n\x07gene_id\x18\x01 \x01(\rB\x1c\xc2\xf0\x19\x18\x12\x0c\n\x04\x32\x37\x37\x38\x12\x04GNAS\x12\x08\n\x01\x32\x12\x03\x41\x32MR\x06geneId\x12X\n\x10returned_content\x18\x02 \x01(\x0e\x32-.ncbi.datasets.v2.OrthologRequest.ContentTypeR\x0freturnedContent\x12\xa2\x01\n\x0ctaxon_filter\x18\x03 \x03(\tB\x7f\xc2\xf0\x19{\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12!\n\x05human\n\x0bhouse mouse\x12\x0b\x43ommon Name\x12-\n\x0chomo sapiens\n\x0cmus musculus\x12\x0fScientific NameR\x0btaxonFilter\x12%\n\tpage_size\x18\t \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\n \x01(\tR\tpageToken\")\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\"\x9e\x07\n\x12GeneDatasetRequest\x12[\n\x08gene_ids\x18\x01 \x03(\rB@\xc2\xf0\x19<\x12\x14\n\x05\x35\x39\x30\x36\x37\x12\x0bIL21 GeneID\x12$\n\x05\x35\x39\x30\x36\x37\n\x05\x35\x30\x36\x31\x35\x12\x14IL21 & IL21R GeneIDsR\x07geneIds\x12W\n\x17include_annotation_type\x18\x06 \x03(\x0e\x32\x17.ncbi.datasets.v2.FastaB\x06\xda\xf0\x19\x02\x08\x02R\x15includeAnnotationType\x12[\n\x10returned_content\x18\x07 \x01(\x0e\x32\x30.ncbi.datasets.v2.GeneDatasetRequest.ContentTypeR\x0freturnedContent\x12!\n\x0c\x66\x61sta_filter\x18\x0c \x03(\tR\x0b\x66\x61staFilter\x12\xd9\x01\n\naux_report\x18\r \x03(\x0e\x32:.ncbi.datasets.v2.GeneDatasetRequest.GeneDatasetReportTypeB~\xc2\xf0\x19z\x12x\n\x0ePRODUCT_REPORT\x12\x66Select additional report types to include with download. The base data report will always be included.R\tauxReport\x12\x80\x01\n\x0ctable_fields\x18\x08 \x03(\tB]\xc2\xf0\x19Y\x12?\n\x07gene-id\n\tgene-type\n\x0b\x64\x65scription\x12\x1cGeneID, type and description\x12\x16\n\x07gene-id\x12\x0bJust GeneIDR\x0btableFields\x12\x66\n\x11table_report_type\x18\x0e \x01(\x0e\x32:.ncbi.datasets.v2.GeneDatasetRequest.GeneDatasetReportTypeR\x0ftableReportType\")\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\"?\n\x15GeneDatasetReportType\x12\x12\n\x0e\x44\x41TASET_REPORT\x10\x00\x12\x12\n\x0ePRODUCT_REPORT\x10\x01:\x1f\xc2\xf0\x19\x1b\n\x19gene_ids:\n- 59067\n- 50615\"\xe2\t\n\x19GeneDatasetReportsRequest\x12\x62\n\x10returned_content\x18\x01 \x01(\x0e\x32\x37.ncbi.datasets.v2.GeneDatasetReportsRequest.ContentTypeR\x0freturnedContent\x12[\n\x08gene_ids\x18\x02 \x03(\rB@\xc2\xf0\x19<\x12\x14\n\x05\x35\x39\x30\x36\x37\x12\x0bIL21 GeneID\x12$\n\x05\x35\x39\x30\x36\x37\n\x05\x35\x30\x36\x31\x35\x12\x14IL21 & IL21R GeneIDsR\x07geneIds\x12\x81\x01\n\naccessions\x18\x03 \x03(\tBa\xc2\xf0\x19]\x12.\n\x0bNM_021803.4\n\x0bNM_181078.3\x12\x12RefSeq transcripts\x12+\n\x0bNP_068575.1\n\x0bNP_851564.1\x12\x0fRefSeq proteinsR\naccessions\x12g\n\x11symbols_for_taxon\x18\x04 \x01(\x0b\x32;.ncbi.datasets.v2.GeneDatasetReportsRequest.SymbolsForTaxonR\x0fsymbolsForTaxon\x12s\n\x05taxon\x18\x05 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x12\x80\x01\n\x0ctable_fields\x18\x08 \x03(\tB]\xc2\xf0\x19Y\x12?\n\x07gene-id\n\tgene-type\n\x0b\x64\x65scription\x12\x1cGeneID, type and description\x12\x16\n\x07gene-id\x12\x0bJust GeneIDR\x0btableFields\x12!\n\x0ctable_format\x18\r \x01(\tR\x0btableFormat\x12%\n\tpage_size\x18\t \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\n \x01(\tR\tpageToken\x12\x30\n\x05query\x18\x0b \x01(\tB\x1a\xc2\xf0\x19\x16\x12\x14\n\x12\x41\x32M immunoglobulinR\x05query\x12\x63\n\x05types\x18\x0c \x03(\x0e\x32\x1a.ncbi.datasets.v2.GeneTypeB1\xc2\xf0\x19-\x12+\n\x0ePROTEIN_CODING\x12\x19Only protein coding genesR\x05types\x1a\xb4\x01\n\x0fSymbolsForTaxon\x12,\n\x07symbols\x18\x01 \x03(\tB\x12\xc2\xf0\x19\x0e\x12\x0c\n\x04GNAS\x12\x04GNASR\x07symbols\x12s\n\x05taxon\x18\x02 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\"G\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08IDS_ONLY\x10\x01\x12\x0b\n\x07PRODUCT\x10\x02\x12\x0f\n\x0b\x43OUNTS_ONLY\x10\x03:\x1f\xc2\xf0\x19\x1b\n\x19gene_ids:\n- 59067\n- 50615\"\x90\x01\n\x10GeneLinksRequest\x12[\n\x08gene_ids\x18\x01 \x03(\rB@\xc2\xf0\x19<\x12\x14\n\x05\x35\x39\x30\x36\x37\x12\x0bIL21 GeneID\x12$\n\x05\x35\x39\x30\x36\x37\n\x05\x35\x30\x36\x31\x35\x12\x14IL21 & IL21R GeneIDsR\x07geneIds:\x1f\xc2\xf0\x19\x1b\n\x19gene_ids:\n- 59067\n- 50615\"\xe8\x02\n\x0eGeneLinksReply\x12H\n\ngene_links\x18\x01 \x03(\x0b\x32).ncbi.datasets.v2.GeneLinksReply.GeneLinkR\tgeneLinks\x1a\xbe\x01\n\x08GeneLink\x12\x17\n\x07gene_id\x18\x01 \x01(\rR\x06geneId\x12S\n\x0egene_link_type\x18\x02 \x01(\x0e\x32-.ncbi.datasets.v2.GeneLinksReply.GeneLinkTypeR\x0cgeneLinkType\x12#\n\rresource_link\x18\x03 \x01(\tR\x0cresourceLink\x12\x1f\n\x0bresource_id\x18\x04 \x01(\tR\nresourceId\"K\n\x0cGeneLinkType\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\r\n\tGENE_LINK\x10\x01\x12\x0c\n\x08GDV_LINK\x10\x02\x12\x11\n\rORTHOLOG_LINK\x10\x03\"\xff\x01\n\x1cGeneChromosomeSummaryRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x12j\n\x0f\x61nnotation_name\x18\x02 \x01(\tBA\xc2\xf0\x19=\x12;\n\x1f\x41nnotation Release 109.20211119\x12\x18Human annotation releaseR\x0e\x61nnotationName\"\xfd\x01\n\x1aGeneChromosomeSummaryReply\x12~\n\x19gene_chromosome_summaries\x18\x01 \x03(\x0b\x32\x42.ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummaryR\x17geneChromosomeSummaries\x1a_\n\x15GeneChromosomeSummary\x12\x12\n\x04name\x18\x01 \x01(\tR\x04name\x12\x14\n\x05\x63ount\x18\x02 \x01(\rR\x05\x63ount\x12\x1c\n\taccession\x18\x03 \x01(\tR\taccession\"\xc4\x01\n\x14\x41ssemblyLinksRequest\x12v\n\naccessions\x18\x01 \x03(\tBV\xc2\xf0\x19R\x12\x35\n\x10GCF_000001405.40\n\x10GCF_000001635.27\x12\x0fHuman and Mouse\x12\x19\n\x10GCF_000001405.40\x12\x05HumanR\naccessions:4\xc2\xf0\x19\x30\n.accessions:\n  accessions:\n  - GCF_000001405.40\"\xb8\x03\n\x12\x41ssemblyLinksReply\x12X\n\x0e\x61ssembly_links\x18\x01 \x03(\x0b\x32\x31.ncbi.datasets.v2.AssemblyLinksReply.AssemblyLinkR\rassemblyLinks\x1a\xe5\x01\n\x0c\x41ssemblyLink\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12\x63\n\x12\x61ssembly_link_type\x18\x02 \x01(\x0e\x32\x35.ncbi.datasets.v2.AssemblyLinksReply.AssemblyLinkTypeR\x10\x61ssemblyLinkType\x12#\n\rresource_link\x18\x03 \x01(\tR\x0cresourceLink\x12-\n\x12linked_identifiers\x18\x04 \x03(\tR\x11linkedIdentifiers\"`\n\x10\x41ssemblyLinkType\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x0c\n\x08GDV_LINK\x10\x01\x12\x0c\n\x08\x46TP_LINK\x10\x02\x12\x13\n\x0f\x41SSEMBLY_PUBMED\x10\x03\x12\x0e\n\nBLAST_LINK\x10\x04\"\x91\x0c\n\x13VirusDatasetRequest\x12\x63\n\naccessions\x18\x10 \x03(\tBC\xc2\xf0\x19?\x12=\n\x0bNC_045512.2\x12.SARS-COV-2 isolate Wuhan-Hu-1, complete genomeR\naccessions\x12Z\n\x05taxon\x18\x13 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x11 \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x14 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\x08 \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\t \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\x12\x98\x01\n\x0ctable_fields\x18\n \x03(\x0e\x32!.ncbi.datasets.v2.VirusTableFieldBR\xc2\xf0\x19N\x12L\n\x14nucleotide_accession\n\x11nucleotide_length\n\x10nuc_completeness\x12\x0f\x41ssembly FieldsR\x0btableFields\x12\x8f\x01\n\x10include_sequence\x18\x16 \x03(\x0e\x32#.ncbi.datasets.v2.ViralSequenceTypeB?\xc2\xf0\x19;\x12\x39\n\x06GENOME\n\x03\x43\x44S\n\x07PROTEIN\x12!Select viral sequences to includeR\x0fincludeSequence\x12\xc3\x01\n\naux_report\x18\x17 \x03(\x0e\x32(.ncbi.datasets.v2.VirusDatasetReportTypeBz\xc2\xf0\x19v\x12t\n\nANNOTATION\x12\x66Select additional report types to include with download. The base data report will always be included.R\tauxReport\x12m\n\x06\x66ormat\x18\x0e \x01(\x0e\x32\x1d.ncbi.datasets.v2.TableFormatB6\xc2\xf0\x19\x32\n\x03tsv\x12\n\n\x03tsv\x12\x03TSV\x12\n\n\x03\x63sv\x12\x03\x43SV\x12\x13\n\x05jsonl\x12\nJSON LinesR\x06\x66ormat\x12\x17\n\x07use_psg\x18\x63 \x01(\x08R\x06usePsg\"\xdf\x07\n\x12VirusDatasetFilter\x12\xac\x01\n\naccessions\x18\x02 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\x12Z\n\x05taxon\x18\x03 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0b \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x08 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\t \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\n \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\"\xe5\x03\n\x16VirusDataReportRequest\x12<\n\x06\x66ilter\x18\x01 \x01(\x0b\x32$.ncbi.datasets.v2.VirusDatasetFilterR\x06\x66ilter\x12_\n\x10returned_content\x18\x02 \x01(\x0e\x32\x34.ncbi.datasets.v2.VirusDataReportRequest.ContentTypeR\x0freturnedContent\x12i\n\x0ctable_fields\x18\x03 \x03(\tBF\xc2\xf0\x19\x42\x12@\n\taccession\n\x0bis-complete\n\x0cis-annotated\x12\x18Virus Data Report FieldsR\x0btableFields\x12!\n\x0ctable_format\x18\x06 \x01(\tR\x0btableFormat\x12%\n\tpage_size\x18\x04 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\x05 \x01(\tR\tpageToken\"0\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x13\n\x0f\x41\x43\x43\x45SSIONS_ONLY\x10\x01:&\xc2\xf0\x19\"\n taxon: 2697049\nrefseq_only: true\"\xf4\x06\n\x15VirusAnnotationFilter\x12\xac\x01\n\naccessions\x18\x02 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\x12Z\n\x05taxon\x18\x03 \x01(\tBD\xc2\xf0\x19@\x12\x15\n\x07\x32\x36\x39\x37\x30\x34\x39\x12\nSARS-COV-2\x12\x16\n\x05\x31\x31\x31\x31\x38\x12\rCoronaviridae\x12\x0f\n\x07\x31\x33\x33\x35\x36\x32\x36\x12\x04MERSR\x05taxon\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0b \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12\x37\n\x17pangolin_classification\x18\x08 \x01(\tR\x16pangolinClassification\x12@\n\x0cgeo_location\x18\t \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\n \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\"\xd4\x02\n\x1cVirusAnnotationReportRequest\x12?\n\x06\x66ilter\x18\x01 \x01(\x0b\x32\'.ncbi.datasets.v2.VirusAnnotationFilterR\x06\x66ilter\x12\x62\n\x0ctable_fields\x18\x03 \x03(\tB?\xc2\xf0\x19;\x12\x39\n\taccession\n\x0cisolate-name\x12\x1eVirus Annotation Report FieldsR\x0btableFields\x12!\n\x0ctable_format\x18\x06 \x01(\tR\x0btableFormat\x12%\n\tpage_size\x18\x04 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\x05 \x01(\tR\tpageToken:&\xc2\xf0\x19\"\n taxon: 2697049\nrefseq_only: true\"\xc9\x01\n\x18VirusAvailabilityRequest\x12\xac\x01\n\naccessions\x18\x01 \x03(\tB\x8b\x01\xc2\xf0\x19\x86\x01\x12\x83\x01\n\x17NC_045512.2,NC_038294.1\x12hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomesR\naccessions\"\x87\x01\n\x11VirusAvailability\x12)\n\x10valid_accessions\x18\x01 \x03(\tR\x0fvalidAccessions\x12-\n\x12invalid_accessions\x18\x02 \x03(\tR\x11invalidAccessions\x12\x18\n\x07message\x18\x03 \x01(\tR\x07message\"\xfd\n\n\x1aSars2ProteinDatasetRequest\x12~\n\x08proteins\x18\x01 \x03(\tBb\xc2\xf0\x19^\x12\x1e\n\rspike protein\x12\rSpike Protein\x12<\n\rspike protein\n\x10\x65nvelope protein\n\x04RdRp\x12\x13SARS-CoV-2 proteinsR\x08proteins\x12u\n\x0brefseq_only\x18\x04 \x01(\x08\x42T\xc2\xf0\x19P\n\x05\x66\x61lse\x12\x1f\n\x04true\x12\x17Limit to RefSeq genomes\x12&\n\x05\x66\x61lse\x12\x1d\x42oth GenBank & RefSeq genomesR\nrefseqOnly\x12l\n\x0e\x61nnotated_only\x18\x05 \x01(\x08\x42\x45\xc2\xf0\x19\x41\n\x05\x66\x61lse\x12\"\n\x04true\x12\x1aLimit to annotated genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\rannotatedOnly\x12\x65\n\x0ereleased_since\x18\x06 \x01(\x0b\x32\x1a.google.protobuf.TimestampB\"\xc2\xf0\x19\x1e\x12\x1c\n\n2020-08-01\x12\x0e\x41ugust 1, 2020R\rreleasedSince\x12\x62\n\rupdated_since\x18\x0f \x01(\x0b\x32\x1a.google.protobuf.TimestampB!\xc2\xf0\x19\x1d\x12\x1b\n\n2021-07-18\x12\rJuly 18, 2021R\x0cupdatedSince\x12(\n\x04host\x18\x07 \x01(\tB\x14\xc2\xf0\x19\x10\x12\x0e\n\x05human\x12\x05humanR\x04host\x12@\n\x0cgeo_location\x18\x08 \x01(\tB\x1d\xc2\xf0\x19\x19\x12\n\n\x03USA\x12\x03USA\x12\x0b\n\x02TX\x12\x05TexasR\x0bgeoLocation\x12i\n\rcomplete_only\x18\t \x01(\x08\x42\x44\xc2\xf0\x19@\n\x05\x66\x61lse\x12!\n\x04true\x12\x19Limit to complete genomes\x12\x14\n\x05\x66\x61lse\x12\x0b\x41ll genomesR\x0c\x63ompleteOnly\x12\x98\x01\n\x0ctable_fields\x18\n \x03(\x0e\x32!.ncbi.datasets.v2.VirusTableFieldBR\xc2\xf0\x19N\x12L\n\x14nucleotide_accession\n\x11nucleotide_length\n\x10nuc_completeness\x12\x0f\x41ssembly FieldsR\x0btableFields\x12\x87\x01\n\x10include_sequence\x18\x16 \x03(\x0e\x32#.ncbi.datasets.v2.ViralSequenceTypeB7\xc2\xf0\x19\x33\x12\x31\n\x03\x43\x44S\n\x07PROTEIN\x12!Select viral sequences to includeR\x0fincludeSequence\x12\xc3\x01\n\naux_report\x18\x17 \x03(\x0e\x32(.ncbi.datasets.v2.VirusDatasetReportTypeBz\xc2\xf0\x19v\x12t\n\nANNOTATION\x12\x66Select additional report types to include with download. The base data report will always be included.R\tauxReport\x12m\n\x06\x66ormat\x18\x0e \x01(\x0e\x32\x1d.ncbi.datasets.v2.TableFormatB6\xc2\xf0\x19\x32\n\x03tsv\x12\n\n\x03tsv\x12\x03TSV\x12\n\n\x03\x63sv\x12\x03\x43SV\x12\x13\n\x05jsonl\x12\nJSON LinesR\x06\x66ormat\"\xb0\r\n\x0f\x44ownloadSummary\x12!\n\x0crecord_count\x18\x01 \x01(\rR\x0brecordCount\x12)\n\x0e\x61ssembly_count\x18\x02 \x01(\rB\x02\x18\x01R\rassemblyCount\x12J\n\x13resource_updated_on\x18\x04 \x01(\x0b\x32\x1a.google.protobuf.TimestampR\x11resourceUpdatedOn\x12\x46\n\x08hydrated\x18\x05 \x01(\x0b\x32*.ncbi.datasets.v2.DownloadSummary.HydratedR\x08hydrated\x12L\n\ndehydrated\x18\x06 \x01(\x0b\x32,.ncbi.datasets.v2.DownloadSummary.DehydratedR\ndehydrated\x12;\n\x06\x65rrors\x18\x07 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorB\x02\x18\x01R\x06\x65rrors\x12=\n\x08messages\x18\x08 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12Y\n\x0f\x61vailable_files\x18\t \x01(\x0b\x32\x30.ncbi.datasets.v2.DownloadSummary.AvailableFilesR\x0e\x61vailableFiles\x1a\x8c\x01\n\x08Hydrated\x12\x33\n\x16\x65stimated_file_size_mb\x18\x01 \x01(\rR\x13\x65stimatedFileSizeMb\x12\x10\n\x03url\x18\x02 \x01(\tR\x03url\x12\x39\n\x19\x63li_download_command_line\x18\x03 \x01(\tR\x16\x63liDownloadCommandLine\x1a\xcb\x01\n\nDehydrated\x12\x33\n\x16\x65stimated_file_size_mb\x18\x01 \x01(\rR\x13\x65stimatedFileSizeMb\x12\x10\n\x03url\x18\x02 \x01(\tR\x03url\x12\x39\n\x19\x63li_download_command_line\x18\x03 \x01(\tR\x16\x63liDownloadCommandLine\x12;\n\x1a\x63li_rehydrate_command_line\x18\x04 \x01(\tR\x17\x63liRehydrateCommandLine\x1a\x45\n\x0b\x46ileSummary\x12\x1d\n\nfile_count\x18\x01 \x01(\rR\tfileCount\x12\x17\n\x07size_mb\x18\x02 \x01(\x02R\x06sizeMb\x1a\xf1\x05\n\x0e\x41vailableFiles\x12Y\n\x11\x61ll_genomic_fasta\x18\x01 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x0f\x61llGenomicFasta\x12L\n\ngenome_gff\x18\x02 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\tgenomeGff\x12N\n\x0bgenome_gbff\x18\x03 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\ngenomeGbff\x12J\n\trna_fasta\x18\x04 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x08rnaFasta\x12L\n\nprot_fasta\x18\x05 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\tprotFasta\x12L\n\ngenome_gtf\x18\x06 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\tgenomeGtf\x12J\n\tcds_fasta\x18\x07 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x08\x63\x64sFasta\x12V\n\x0fsequence_report\x18\x08 \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x0esequenceReport\x12Z\n\x11\x61nnotation_report\x18\t \x01(\x0b\x32-.ncbi.datasets.v2.DownloadSummary.FileSummaryR\x10\x61nnotationReport\"#\n\rTabularOutput\x12\x12\n\x04\x64\x61ta\x18\x01 \x01(\tR\x04\x64\x61ta\"\xbb\x03\n\x18MicroBiggeDatasetRequest\x12,\n\x11opaque_solr_query\x18\x01 \x01(\tH\x00R\x0fopaqueSolrQuery\x12Q\n\x05\x66iles\x18\x02 \x03(\x0e\x32\x33.ncbi.datasets.v2.MicroBiggeDatasetRequest.FileTypeB\x06\xda\xf0\x19\x02\x08\x02R\x05\x66iles\x12o\n\x14\x65lement_flank_config\x18\x03 \x01(\x0b\x32=.ncbi.datasets.v2.MicroBiggeDatasetRequest.ElementFlankConfigR\x12\x65lementFlankConfig\x1a,\n\x12\x45lementFlankConfig\x12\x16\n\x06length\x18\x01 \x01(\rR\x06length\"v\n\x08\x46ileType\x12\x19\n\x15\x46ILE_TYPE_UNSPECIFIED\x10\x00\x12\x11\n\relement_fasta\x10\x01\x12\x17\n\x13\x65lement_flank_fasta\x10\x02\x12\x10\n\x0c\x63ontig_fasta\x10\x03\x12\x11\n\rprotein_fasta\x10\x04\x42\x07\n\x05query\"\x7f\n\x0cVersionReply\x12\x18\n\x07version\x18\x01 \x01(\tR\x07version\x12\x1b\n\tmajor_ver\x18\x02 \x01(\rR\x08majorVer\x12\x1b\n\tminor_ver\x18\x03 \x01(\rR\x08minorVer\x12\x1b\n\tpatch_ver\x18\x04 \x01(\rR\x08patchVer\"\x81\x05\n\x15ProkaryoteGeneRequest\x12\x93\x01\n\naccessions\x18\x01 \x03(\tBs\xc2\xf0\x19o\x12.\n\x0eWP_015878339.1\x12\x1c\x44NA-deoxyinosine glycosylase\x12=\n\x0eWP_001040085.1\x12+type II CRISPR RNA-guided endonuclease Cas9R\naccessions\x12\xf6\x01\n\x17include_annotation_type\x18\x02 \x03(\x0e\x32\x17.ncbi.datasets.v2.FastaB\xa4\x01\xda\xf0\x19\x02\x08\x02\xc2\xf0\x19\x99\x01\x12\x1e\n\nFASTA_GENE\x12\x10Gene range FASTA\x12?\n\tFASTA_RNA\n\rFASTA_PROTEIN\x12#Transcript & protein sequence FASTA\x12\x36\n\nFASTA_GENE\n\tFASTA_RNA\n\rFASTA_PROTEIN\x12\x0e\x41ll annotationR\x15includeAnnotationType\x12\x63\n\x11gene_flank_config\x18\x03 \x01(\x0b\x32\x37.ncbi.datasets.v2.ProkaryoteGeneRequest.GeneFlankConfigR\x0fgeneFlankConfig\x12\x14\n\x05taxon\x18\x04 \x01(\tR\x05taxon\x1a)\n\x0fGeneFlankConfig\x12\x16\n\x06length\x18\x01 \x01(\rR\x06length:3\xc2\xf0\x19/\n-accessions:\n- WP_000000001.1\n- WP_000000002.1\"L\n\x0cSleepRequest\x12\x1d\n\nsleep_msec\x18\x01 \x01(\x05R\tsleepMsec\x12\x1d\n\nerror_rate\x18\x02 \x01(\x02R\terrorRate\"\"\n\nSleepReply\x12\x14\n\x05reply\x18\x01 \x01(\tR\x05reply\"\xc8\x06\n\x0cTaxonomyNode\x12\x15\n\x06tax_id\x18\x01 \x01(\rR\x05taxId\x12#\n\rorganism_name\x18\x02 \x01(\tR\x0corganismName\x12\x1f\n\x0b\x63ommon_name\x18\x03 \x01(\tR\ncommonName\x12.\n\x13genbank_common_name\x18\x19 \x01(\tR\x11genbankCommonName\x12\x1a\n\x08\x61\x63ronyms\x18\x1a \x03(\tR\x08\x61\x63ronyms\x12\'\n\x0fgenbank_acronym\x18\x1b \x01(\tR\x0egenbankAcronym\x12\x1d\n\nblast_name\x18\x18 \x01(\tR\tblastName\x12\x18\n\x07lineage\x18\x04 \x03(\rR\x07lineage\x12\x1a\n\x08\x63hildren\x18\x05 \x03(\rR\x08\x63hildren\x12_\n-descendent_with_described_species_names_count\x18\x06 \x01(\rR(descendentWithDescribedSpeciesNamesCount\x12\x36\n\x04rank\x18\x07 \x01(\x0e\x32\".ncbi.datasets.v2.reports.RankTypeR\x04rank\x12;\n\x1ahas_described_species_name\x18\x08 \x01(\x08R\x17hasDescribedSpeciesName\x12\x42\n\x06\x63ounts\x18\t \x03(\x0b\x32*.ncbi.datasets.v2.TaxonomyNode.CountByTypeR\x06\x63ounts\x12\x17\n\x07min_ord\x18\x14 \x01(\rR\x06minOrd\x12\x17\n\x07max_ord\x18\x15 \x01(\rR\x06maxOrd\x12\x18\n\x07\x65xtinct\x18\x16 \x01(\x08R\x07\x65xtinct\x12\'\n\x0fgenomic_moltype\x18\x17 \x01(\tR\x0egenomicMoltype\x12$\n\x0e_indexed_names\x18\x65 \x03(\tR\x0cIndexedNames\x1a\\\n\x0b\x43ountByType\x12\x37\n\x04type\x18\x01 \x01(\x0e\x32#.ncbi.datasets.v2.reports.CountTypeR\x04type\x12\x14\n\x05\x63ount\x18\x02 \x01(\rR\x05\x63ount\"\xfe\x04\n\x17TaxonomyMetadataRequest\x12\x91\x01\n\x06taxons\x18\x01 \x03(\tBy\xc2\xf0\x19u\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12\x1b\n\x05human\n\x05mouse\x12\x0b\x43ommon Name\x12-\n\x0cHomo sapiens\n\x0cMus musculus\x12\x0fScientific NameR\x06taxons\x12`\n\x10returned_content\x18\x02 \x01(\x0e\x32\x35.ncbi.datasets.v2.TaxonomyMetadataRequest.ContentTypeR\x0freturnedContent\x12%\n\tpage_size\x18\x03 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\\\n\x16include_tabular_header\x18\x0c \x01(\x0e\x32&.ncbi.datasets.v2.IncludeTabularHeaderR\x14includeTabularHeader\x12\x1d\n\npage_token\x18\x04 \x01(\tR\tpageToken\x12X\n\x0ctable_format\x18\r \x01(\x0e\x32\x35.ncbi.datasets.v2.TaxonomyMetadataRequest.TableFormatR\x0btableFormat\"5\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\n\n\x06TAXIDS\x10\x01\x12\x0c\n\x08METADATA\x10\x02\"\x1a\n\x0bTableFormat\x12\x0b\n\x07SUMMARY\x10\x00:\x1c\xc2\xf0\x19\x18\n\x16taxons:\n- 9606\n- mouse\"\x9f\x02\n\x18TaxonomyRelatedIdRequest\x12=\n\x06tax_id\x18\x01 \x01(\rB&\xc2\xf0\x19\"\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy IdentifierR\x05taxId\x12\x34\n\x0finclude_lineage\x18\x02 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\x0eincludeLineage\x12\x34\n\x0finclude_subtree\x18\x03 \x01(\x08\x42\x0b\xc2\xf0\x19\x07\n\x05\x66\x61lseR\x0eincludeSubtree\x12%\n\tpage_size\x18\x08 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12\x1d\n\npage_token\x18\t \x01(\tR\tpageToken:\x12\xc2\xf0\x19\x0e\n\x0ctax_id: 9606\"U\n\x12TaxonomyTaxIdsPage\x12\x17\n\x07tax_ids\x18\x01 \x03(\rR\x06taxIds\x12&\n\x0fnext_page_token\x18\x04 \x01(\tR\rnextPageToken\"\xaf\x03\n\x16TaxonomyDatasetRequest\x12\x46\n\x07tax_ids\x18\x01 \x03(\rB-\xc2\xf0\x19)\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy IdentifierR\x06taxIds\x12\xef\x01\n\x0b\x61ux_reports\x18\x04 \x03(\x0e\x32;.ncbi.datasets.v2.TaxonomyDatasetRequest.TaxonomyReportTypeB\x90\x01\xc2\xf0\x19\x8b\x01\x12\x88\x01\n\x0cNAMES_REPORT\n\x10TAXONOMY_SUMMARY\x12\x66Select additional report types to include with download. The base data report will always be included.R\nauxReports\"<\n\x12TaxonomyReportType\x12\x14\n\x10TAXONOMY_SUMMARY\x10\x00\x12\x10\n\x0cNAMES_REPORT\x10\x01:\x1d\xc2\xf0\x19\x19\n\x17tax_ids:\n- 9606\n- 10090\"\xd9\x01\n\rTaxonomyMatch\x12=\n\x08warnings\x18\x01 \x03(\x0b\x32!.ncbi.datasets.v2.reports.WarningR\x08warnings\x12\x37\n\x06\x65rrors\x18\x02 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorR\x06\x65rrors\x12\x14\n\x05query\x18\x03 \x03(\tR\x05query\x12:\n\x08taxonomy\x18\x04 \x01(\x0b\x32\x1e.ncbi.datasets.v2.TaxonomyNodeR\x08taxonomy\"\xa1\x01\n\x18TaxonomyMetadataResponse\x12=\n\x08messages\x18\x01 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12\x46\n\x0etaxonomy_nodes\x18\x02 \x03(\x0b\x32\x1f.ncbi.datasets.v2.TaxonomyMatchR\rtaxonomyNodes\"\xdc\x01\n\x14TaxonomyImageRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\x12:\n\nimage_size\x18\x02 \x01(\x0e\x32\x1b.ncbi.datasets.v2.ImageSizeR\timageSize:\x13\xc2\xf0\x19\x0f\n\rtaxon:\n- 9606\"\xa8\x01\n\x1cTaxonomyImageMetadataRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon:\x13\xc2\xf0\x19\x0f\n\rtaxon:\n- 9606\"\xf2\x01\n\x1dTaxonomyImageMetadataResponse\x12\x15\n\x06tax_id\x18\x01 \x01(\tR\x05taxId\x12\x10\n\x03src\x18\x02 \x01(\tR\x03src\x12\x18\n\x07license\x18\x03 \x01(\tR\x07license\x12 \n\x0b\x61ttribution\x18\x04 \x01(\tR\x0b\x61ttribution\x12\x16\n\x06source\x18\x05 \x01(\tR\x06source\x12<\n\x0bimage_sizes\x18\x06 \x03(\x0e\x32\x1b.ncbi.datasets.v2.ImageSizeR\nimageSizes\x12\x16\n\x06\x66ormat\x18\x07 \x01(\tR\x06\x66ormat\"\xa2\x02\n\x1eTaxonomyFilteredSubtreeRequest\x12\x91\x01\n\x06taxons\x18\x01 \x03(\tBy\xc2\xf0\x19u\x12\'\n\x04\x39\x36\x30\x36\n\x05\x31\x30\x30\x39\x30\x12\x18NCBI Taxonomy Identifier\x12\x1b\n\x05human\n\x05mouse\x12\x0b\x43ommon Name\x12-\n\x0cHomo sapiens\n\x0cMus musculus\x12\x0fScientific NameR\x06taxons\x12\'\n\x0fspecified_limit\x18\x02 \x01(\x08R\x0especifiedLimit\x12\x43\n\x0brank_limits\x18\x03 \x03(\x0e\x32\".ncbi.datasets.v2.reports.RankTypeR\nrankLimits\"\xef\x04\n\x1fTaxonomyFilteredSubtreeResponse\x12\x1d\n\nroot_nodes\x18\x01 \x03(\rR\trootNodes\x12R\n\x05\x65\x64ges\x18\x02 \x03(\x0b\x32<.ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgesEntryR\x05\x65\x64ges\x12=\n\x08warnings\x18\x03 \x03(\x0b\x32!.ncbi.datasets.v2.reports.WarningR\x08warnings\x12\x37\n\x06\x65rrors\x18\x04 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorR\x06\x65rrors\x1a\xee\x01\n\x04\x45\x64ge\x12)\n\x10visible_children\x18\x01 \x03(\rR\x0fvisibleChildren\x12k\n\x0f\x63hildren_status\x18\x02 \x01(\x0e\x32\x42.ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge.ChildStatusR\x0e\x63hildrenStatus\"N\n\x0b\x43hildStatus\x12\x0f\n\x0bUNSPECIFIED\x10\x00\x12\x15\n\x11HAS_MORE_CHILDREN\x10\x01\x12\x17\n\x13NO_VISIBLE_CHILDREN\x10\x02\x1ap\n\nEdgesEntry\x12\x10\n\x03key\x18\x01 \x01(\rR\x03key\x12L\n\x05value\x18\x02 \x01(\x0b\x32\x36.ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgeR\x05value:\x02\x38\x01\"\xe9\x01\n\x0e\x44\x61tasetRequest\x12G\n\tgenome_v2\x18\x04 \x01(\x0b\x32(.ncbi.datasets.v2.AssemblyDatasetRequestH\x00R\x08genomeV2\x12?\n\x07gene_v2\x18\x05 \x01(\x0b\x32$.ncbi.datasets.v2.GeneDatasetRequestH\x00R\x06geneV2\x12\x42\n\x08virus_v2\x18\x06 \x01(\x0b\x32%.ncbi.datasets.v2.VirusDatasetRequestH\x00R\x07virusV2B\t\n\x07request\"\xa0\x01\n\x14TaxonomyLinksRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon:\x13\xc2\xf0\x19\x0f\n\rtaxon:\n- 9606\"\xb8\x03\n\x15TaxonomyLinksResponse\x12\x15\n\x06tax_id\x18\x01 \x01(\tR\x05taxId\x12\x30\n\x14\x65ncyclopedia_of_life\x18\x02 \x01(\tR\x12\x65ncyclopediaOfLife\x12W\n(global_biodiversity_information_facility\x18\x03 \x01(\tR%globalBiodiversityInformationFacility\x12 \n\x0binaturalist\x18\x04 \x01(\tR\x0binaturalist\x12\x1c\n\tviralzone\x18\x05 \x01(\tR\tviralzone\x12\x1c\n\twikipedia\x18\x06 \x01(\tR\twikipedia\x12X\n\rgeneric_links\x18\x07 \x03(\x0b\x32\x33.ncbi.datasets.v2.TaxonomyLinksResponse.GenericLinkR\x0cgenericLinks\x1a\x45\n\x0bGenericLink\x12\x1b\n\tlink_name\x18\x01 \x01(\tR\x08linkName\x12\x19\n\x08link_url\x18\x02 \x01(\tR\x07linkUrl\"\x8f\x01\n\x18GeneCountsByTaxonRequest\x12s\n\x05taxon\x18\x01 \x01(\tB]\xc2\xf0\x19Y\x12 \n\x04\x39\x36\x30\x36\x12\x18NCBI Taxonomy Identifier\x12\x14\n\x05human\x12\x0b\x43ommon Name\x12\x1f\n\x0cHomo sapiens\x12\x0fScientific NameR\x05taxon\"\xb2\x01\n\x16GeneCountsByTaxonReply\x12Q\n\x06report\x18\x01 \x03(\x0b\x32\x39.ncbi.datasets.v2.GeneCountsByTaxonReply.GeneTypeAndCountR\x06report\x1a\x45\n\x10GeneTypeAndCount\x12\x1b\n\tgene_type\x18\x01 \x01(\tR\x08geneType\x12\x14\n\x05\x63ount\x18\x02 \x01(\rR\x05\x63ount\"\xc1\x06\n\x17GenomeAnnotationRequest\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12%\n\x0e\x61nnotation_ids\x18\r \x03(\tR\rannotationIds\x12\x18\n\x07symbols\x18\x02 \x03(\tR\x07symbols\x12\x1c\n\tlocations\x18\x0f \x03(\tR\tlocations\x12\x1d\n\ngene_types\x18\x0e \x03(\tR\tgeneTypes\x12\x1f\n\x0bsearch_text\x18\x05 \x03(\tR\nsearchText\x12/\n\x04sort\x18\x06 \x03(\x0b\x32\x1b.ncbi.datasets.v2.SortFieldR\x04sort\x12p\n\x17include_annotation_type\x18\x04 \x03(\x0e\x32\x38.ncbi.datasets.v2.GenomeAnnotationRequest.AnnotationTypeR\x15includeAnnotationType\x12%\n\tpage_size\x18\x08 \x01(\x05\x42\x08\xc2\xf0\x19\x04\n\x02\x32\x30R\x08pageSize\x12!\n\x0ctable_fields\x18\n \x03(\tR\x0btableFields\x12h\n\x0ctable_format\x18\x0b \x01(\x0e\x32\x45.ncbi.datasets.v2.GenomeAnnotationRequest.GenomeAnnotationTableFormatR\x0btableFormat\x12\\\n\x16include_tabular_header\x18\x0c \x01(\x0e\x32&.ncbi.datasets.v2.IncludeTabularHeaderR\x14includeTabularHeader\x12\x1d\n\npage_token\x18\t \x01(\tR\tpageToken\"N\n\x0e\x41nnotationType\x12\x0b\n\x07\x44\x45\x46\x41ULT\x10\x00\x12\x10\n\x0cGENOME_FASTA\x10\x01\x12\r\n\tRNA_FASTA\x10\x02\x12\x0e\n\nPROT_FASTA\x10\x03\"E\n\x1bGenomeAnnotationTableFormat\x12\x0c\n\x08NO_TABLE\x10\x00\x12\x0b\n\x07SUMMARY\x10\x01\x12\x0b\n\x07PRODUCT\x10\x02\"\xa7\x01\n!GenomeAnnotationTableSummaryReply\x12\x1c\n\taccession\x18\x01 \x01(\tR\taccession\x12 \n\x0b\x63hromosomes\x18\x02 \x03(\tR\x0b\x63hromosomes\x12\x1d\n\ngene_types\x18\x03 \x03(\tR\tgeneTypes\x12#\n\rempty_columns\x18\x04 \x03(\tR\x0c\x65mptyColumns\"A\n\x08HttpBody\x12!\n\x0c\x63ontent_type\x18\x01 \x01(\tR\x0b\x63ontentType\x12\x12\n\x04\x64\x61ta\x18\x02 \x01(\x0cR\x04\x64\x61ta*\x87\x01\n\x14IncludeTabularHeader\x12*\n&INCLUDE_TABULAR_HEADER_FIRST_PAGE_ONLY\x10\x00\x12!\n\x1dINCLUDE_TABULAR_HEADER_ALWAYS\x10\x01\x12 \n\x1cINCLUDE_TABULAR_HEADER_NEVER\x10\x02*l\n\rSortDirection\x12\x1e\n\x1aSORT_DIRECTION_UNSPECIFIED\x10\x00\x12\x1c\n\x18SORT_DIRECTION_ASCENDING\x10\x01\x12\x1d\n\x19SORT_DIRECTION_DESCENDING\x10\x02*\x99\x01\n\x05\x46\x61sta\x12\x15\n\x11\x46\x41STA_UNSPECIFIED\x10\x00\x12\x0e\n\nFASTA_GENE\x10\x01\x12\r\n\tFASTA_RNA\x10\x02\x12\x11\n\rFASTA_PROTEIN\x10\x03\x12\x14\n\x10\x46\x41STA_GENE_FLANK\x10\x04\x12\r\n\tFASTA_CDS\x10\x05\x12\x10\n\x0c\x46\x41STA_5P_UTR\x10\x06\x12\x10\n\x0c\x46\x41STA_3P_UTR\x10\x07*\xb8\x01\n\x08GeneType\x12\x0b\n\x07UNKNOWN\x10\x00\x12\x08\n\x04tRNA\x10\x01\x12\x08\n\x04rRNA\x10\x02\x12\t\n\x05snRNA\x10\x03\x12\t\n\x05scRNA\x10\x04\x12\n\n\x06snoRNA\x10\x05\x12\x12\n\x0ePROTEIN_CODING\x10\x06\x12\n\n\x06PSEUDO\x10\x07\x12\x0e\n\nTRANSPOSON\x10\x08\x12\x0b\n\x07miscRNA\x10\t\x12\t\n\x05ncRNA\x10\n\x12\x15\n\x11\x42IOLOGICAL_REGION\x10\x0b\x12\n\n\x05OTHER\x10\xff\x01*\xa1\x03\n\x0fVirusTableField\x12\x0f\n\x0bunspecified\x10\x00\x12\x18\n\x14nucleotide_accession\x10\x01\x12\x12\n\x0especies_tax_id\x10\x02\x12\x10\n\x0cspecies_name\x10\x03\x12\t\n\x05genus\x10\x04\x12\n\n\x06\x66\x61mily\x10\x05\x12\x15\n\x11nucleotide_length\x10\x07\x12\x10\n\x0cisolate_name\x10\x08\x12\x11\n\rsequence_type\x10\t\x12\x14\n\x10nuc_completeness\x10\n\x12\x10\n\x0cgeo_location\x10\x0b\x12\x0c\n\x08us_state\x10\x0c\x12\r\n\thost_name\x10\r\x12\x0f\n\x0bhost_tax_id\x10\x0e\x12\x13\n\x0f\x63ollection_date\x10\x0f\x12\x0e\n\nbioproject\x10\x10\x12\r\n\tbiosample\x10\x11\x12\x14\n\x10polyprotein_name\x10\x13\x12\x10\n\x0cprotein_name\x10\x14\x12\x15\n\x11protein_accession\x10\x15\x12\x13\n\x0fprotein_synonym\x10\x16\x12\x0c\n\x08\x63\x64s_span\x10\x17**\n\x0bTableFormat\x12\x07\n\x03tsv\x10\x00\x12\x07\n\x03\x63sv\x10\x01\x12\t\n\x05jsonl\x10\x02*N\n\x11ViralSequenceType\x12\n\n\x06GENOME\x10\x00\x12\x07\n\x03\x43\x44S\x10\x01\x12\x0b\n\x07PROTEIN\x10\x02\x12\x08\n\x04NONE\x10\x03\x12\r\n\tBIOSAMPLE\x10\x04*R\n\x16VirusDatasetReportType\x12\x12\n\x0e\x44\x41TASET_REPORT\x10\x00\x12\x0e\n\nANNOTATION\x10\x01\x12\x14\n\x10\x42IOSAMPLE_REPORT\x10\x02*3\n\tImageSize\x12\x0f\n\x0bUNSPECIFIED\x10\x00\x12\t\n\x05SMALL\x10\x01\x12\n\n\x06MEDIUM\x10\x02\x42\x15Z\x10ncbi/datasets/v2\xf8\x01\x01\x62\x06proto3')

_INCLUDETABULARHEADER = DESCRIPTOR.enum_types_by_name['IncludeTabularHeader']
IncludeTabularHeader = enum_type_wrapper.EnumTypeWrapper(_INCLUDETABULARHEADER)
_SORTDIRECTION = DESCRIPTOR.enum_types_by_name['SortDirection']
SortDirection = enum_type_wrapper.EnumTypeWrapper(_SORTDIRECTION)
_FASTA = DESCRIPTOR.enum_types_by_name['Fasta']
Fasta = enum_type_wrapper.EnumTypeWrapper(_FASTA)
_GENETYPE = DESCRIPTOR.enum_types_by_name['GeneType']
GeneType = enum_type_wrapper.EnumTypeWrapper(_GENETYPE)
_VIRUSTABLEFIELD = DESCRIPTOR.enum_types_by_name['VirusTableField']
VirusTableField = enum_type_wrapper.EnumTypeWrapper(_VIRUSTABLEFIELD)
_TABLEFORMAT = DESCRIPTOR.enum_types_by_name['TableFormat']
TableFormat = enum_type_wrapper.EnumTypeWrapper(_TABLEFORMAT)
_VIRALSEQUENCETYPE = DESCRIPTOR.enum_types_by_name['ViralSequenceType']
ViralSequenceType = enum_type_wrapper.EnumTypeWrapper(_VIRALSEQUENCETYPE)
_VIRUSDATASETREPORTTYPE = DESCRIPTOR.enum_types_by_name['VirusDatasetReportType']
VirusDatasetReportType = enum_type_wrapper.EnumTypeWrapper(_VIRUSDATASETREPORTTYPE)
_IMAGESIZE = DESCRIPTOR.enum_types_by_name['ImageSize']
ImageSize = enum_type_wrapper.EnumTypeWrapper(_IMAGESIZE)
INCLUDE_TABULAR_HEADER_FIRST_PAGE_ONLY = 0
INCLUDE_TABULAR_HEADER_ALWAYS = 1
INCLUDE_TABULAR_HEADER_NEVER = 2
SORT_DIRECTION_UNSPECIFIED = 0
SORT_DIRECTION_ASCENDING = 1
SORT_DIRECTION_DESCENDING = 2
FASTA_UNSPECIFIED = 0
FASTA_GENE = 1
FASTA_RNA = 2
FASTA_PROTEIN = 3
FASTA_GENE_FLANK = 4
FASTA_CDS = 5
FASTA_5P_UTR = 6
FASTA_3P_UTR = 7
UNKNOWN = 0
tRNA = 1
rRNA = 2
snRNA = 3
scRNA = 4
snoRNA = 5
PROTEIN_CODING = 6
PSEUDO = 7
TRANSPOSON = 8
miscRNA = 9
ncRNA = 10
BIOLOGICAL_REGION = 11
OTHER = 255
unspecified = 0
nucleotide_accession = 1
species_tax_id = 2
species_name = 3
genus = 4
family = 5
nucleotide_length = 7
isolate_name = 8
sequence_type = 9
nuc_completeness = 10
geo_location = 11
us_state = 12
host_name = 13
host_tax_id = 14
collection_date = 15
bioproject = 16
biosample = 17
polyprotein_name = 19
protein_name = 20
protein_accession = 21
protein_synonym = 22
cds_span = 23
tsv = 0
csv = 1
jsonl = 2
GENOME = 0
CDS = 1
PROTEIN = 2
NONE = 3
BIOSAMPLE = 4
DATASET_REPORT = 0
ANNOTATION = 1
BIOSAMPLE_REPORT = 2
UNSPECIFIED = 0
SMALL = 1
MEDIUM = 2


_METHODPAYLOADREQUEST = DESCRIPTOR.message_types_by_name['MethodPayloadRequest']
_ASSEMBLYSEQUENCEREPORTSREQUEST = DESCRIPTOR.message_types_by_name['AssemblySequenceReportsRequest']
_SEQUENCEREPORTPAGE = DESCRIPTOR.message_types_by_name['SequenceReportPage']
_SORTFIELD = DESCRIPTOR.message_types_by_name['SortField']
_BIOSAMPLEDATASETREPORTSREQUEST = DESCRIPTOR.message_types_by_name['BioSampleDatasetReportsRequest']
_ASSEMBLYDATASETREPORTSREQUEST = DESCRIPTOR.message_types_by_name['AssemblyDatasetReportsRequest']
_ASSEMBLYDATASETDESCRIPTORSFILTER = DESCRIPTOR.message_types_by_name['AssemblyDatasetDescriptorsFilter']
_ACCESSIONS = DESCRIPTOR.message_types_by_name['Accessions']
_ASSEMBLYREVISIONHISTORY = DESCRIPTOR.message_types_by_name['AssemblyRevisionHistory']
_ASSEMBLYREVISIONHISTORYREQUEST = DESCRIPTOR.message_types_by_name['AssemblyRevisionHistoryRequest']
_ASSEMBLYCHECKMHISTOGRAMREQUEST = DESCRIPTOR.message_types_by_name['AssemblyCheckMHistogramRequest']
_ASSEMBLYCHECKMHISTOGRAMREPLY = DESCRIPTOR.message_types_by_name['AssemblyCheckMHistogramReply']
_ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL = _ASSEMBLYCHECKMHISTOGRAMREPLY.nested_types_by_name['HistogramInterval']
_BIOPROJECT = DESCRIPTOR.message_types_by_name['BioProject']
_BIOPROJECTLINEAGE = DESCRIPTOR.message_types_by_name['BioProjectLineage']
_BUSCOSTAT = DESCRIPTOR.message_types_by_name['BuscoStat']
_ANNOTATIONFORASSEMBLY = DESCRIPTOR.message_types_by_name['AnnotationForAssembly']
_ANNOTATIONFORASSEMBLY_FILE = _ANNOTATIONFORASSEMBLY.nested_types_by_name['File']
_FEATURECOUNTS = DESCRIPTOR.message_types_by_name['FeatureCounts']
_GENECOUNTS = DESCRIPTOR.message_types_by_name['GeneCounts']
_ASSEMBLYDATASETREQUEST = DESCRIPTOR.message_types_by_name['AssemblyDatasetRequest']
_ASSEMBLYDATASETAVAILABILITY = DESCRIPTOR.message_types_by_name['AssemblyDatasetAvailability']
_ORGANISM = DESCRIPTOR.message_types_by_name['Organism']
_ORGANISM_COUNTS = _ORGANISM.nested_types_by_name['Counts']
_ORGANISM_COUNTBYTYPE = _ORGANISM.nested_types_by_name['CountByType']
_ORGANISMQUERYREQUEST = DESCRIPTOR.message_types_by_name['OrganismQueryRequest']
_SCINAMEANDIDS = DESCRIPTOR.message_types_by_name['SciNameAndIds']
_SCINAMEANDIDS_SCINAMEANDID = _SCINAMEANDIDS.nested_types_by_name['SciNameAndId']
_ORTHOLOGREQUEST = DESCRIPTOR.message_types_by_name['OrthologRequest']
_GENEDATASETREQUEST = DESCRIPTOR.message_types_by_name['GeneDatasetRequest']
_GENEDATASETREPORTSREQUEST = DESCRIPTOR.message_types_by_name['GeneDatasetReportsRequest']
_GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON = _GENEDATASETREPORTSREQUEST.nested_types_by_name['SymbolsForTaxon']
_GENELINKSREQUEST = DESCRIPTOR.message_types_by_name['GeneLinksRequest']
_GENELINKSREPLY = DESCRIPTOR.message_types_by_name['GeneLinksReply']
_GENELINKSREPLY_GENELINK = _GENELINKSREPLY.nested_types_by_name['GeneLink']
_GENECHROMOSOMESUMMARYREQUEST = DESCRIPTOR.message_types_by_name['GeneChromosomeSummaryRequest']
_GENECHROMOSOMESUMMARYREPLY = DESCRIPTOR.message_types_by_name['GeneChromosomeSummaryReply']
_GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY = _GENECHROMOSOMESUMMARYREPLY.nested_types_by_name['GeneChromosomeSummary']
_ASSEMBLYLINKSREQUEST = DESCRIPTOR.message_types_by_name['AssemblyLinksRequest']
_ASSEMBLYLINKSREPLY = DESCRIPTOR.message_types_by_name['AssemblyLinksReply']
_ASSEMBLYLINKSREPLY_ASSEMBLYLINK = _ASSEMBLYLINKSREPLY.nested_types_by_name['AssemblyLink']
_VIRUSDATASETREQUEST = DESCRIPTOR.message_types_by_name['VirusDatasetRequest']
_VIRUSDATASETFILTER = DESCRIPTOR.message_types_by_name['VirusDatasetFilter']
_VIRUSDATAREPORTREQUEST = DESCRIPTOR.message_types_by_name['VirusDataReportRequest']
_VIRUSANNOTATIONFILTER = DESCRIPTOR.message_types_by_name['VirusAnnotationFilter']
_VIRUSANNOTATIONREPORTREQUEST = DESCRIPTOR.message_types_by_name['VirusAnnotationReportRequest']
_VIRUSAVAILABILITYREQUEST = DESCRIPTOR.message_types_by_name['VirusAvailabilityRequest']
_VIRUSAVAILABILITY = DESCRIPTOR.message_types_by_name['VirusAvailability']
_SARS2PROTEINDATASETREQUEST = DESCRIPTOR.message_types_by_name['Sars2ProteinDatasetRequest']
_DOWNLOADSUMMARY = DESCRIPTOR.message_types_by_name['DownloadSummary']
_DOWNLOADSUMMARY_HYDRATED = _DOWNLOADSUMMARY.nested_types_by_name['Hydrated']
_DOWNLOADSUMMARY_DEHYDRATED = _DOWNLOADSUMMARY.nested_types_by_name['Dehydrated']
_DOWNLOADSUMMARY_FILESUMMARY = _DOWNLOADSUMMARY.nested_types_by_name['FileSummary']
_DOWNLOADSUMMARY_AVAILABLEFILES = _DOWNLOADSUMMARY.nested_types_by_name['AvailableFiles']
_TABULAROUTPUT = DESCRIPTOR.message_types_by_name['TabularOutput']
_MICROBIGGEDATASETREQUEST = DESCRIPTOR.message_types_by_name['MicroBiggeDatasetRequest']
_MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG = _MICROBIGGEDATASETREQUEST.nested_types_by_name['ElementFlankConfig']
_VERSIONREPLY = DESCRIPTOR.message_types_by_name['VersionReply']
_PROKARYOTEGENEREQUEST = DESCRIPTOR.message_types_by_name['ProkaryoteGeneRequest']
_PROKARYOTEGENEREQUEST_GENEFLANKCONFIG = _PROKARYOTEGENEREQUEST.nested_types_by_name['GeneFlankConfig']
_SLEEPREQUEST = DESCRIPTOR.message_types_by_name['SleepRequest']
_SLEEPREPLY = DESCRIPTOR.message_types_by_name['SleepReply']
_TAXONOMYNODE = DESCRIPTOR.message_types_by_name['TaxonomyNode']
_TAXONOMYNODE_COUNTBYTYPE = _TAXONOMYNODE.nested_types_by_name['CountByType']
_TAXONOMYMETADATAREQUEST = DESCRIPTOR.message_types_by_name['TaxonomyMetadataRequest']
_TAXONOMYRELATEDIDREQUEST = DESCRIPTOR.message_types_by_name['TaxonomyRelatedIdRequest']
_TAXONOMYTAXIDSPAGE = DESCRIPTOR.message_types_by_name['TaxonomyTaxIdsPage']
_TAXONOMYDATASETREQUEST = DESCRIPTOR.message_types_by_name['TaxonomyDatasetRequest']
_TAXONOMYMATCH = DESCRIPTOR.message_types_by_name['TaxonomyMatch']
_TAXONOMYMETADATARESPONSE = DESCRIPTOR.message_types_by_name['TaxonomyMetadataResponse']
_TAXONOMYIMAGEREQUEST = DESCRIPTOR.message_types_by_name['TaxonomyImageRequest']
_TAXONOMYIMAGEMETADATAREQUEST = DESCRIPTOR.message_types_by_name['TaxonomyImageMetadataRequest']
_TAXONOMYIMAGEMETADATARESPONSE = DESCRIPTOR.message_types_by_name['TaxonomyImageMetadataResponse']
_TAXONOMYFILTEREDSUBTREEREQUEST = DESCRIPTOR.message_types_by_name['TaxonomyFilteredSubtreeRequest']
_TAXONOMYFILTEREDSUBTREERESPONSE = DESCRIPTOR.message_types_by_name['TaxonomyFilteredSubtreeResponse']
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE = _TAXONOMYFILTEREDSUBTREERESPONSE.nested_types_by_name['Edge']
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY = _TAXONOMYFILTEREDSUBTREERESPONSE.nested_types_by_name['EdgesEntry']
_DATASETREQUEST = DESCRIPTOR.message_types_by_name['DatasetRequest']
_TAXONOMYLINKSREQUEST = DESCRIPTOR.message_types_by_name['TaxonomyLinksRequest']
_TAXONOMYLINKSRESPONSE = DESCRIPTOR.message_types_by_name['TaxonomyLinksResponse']
_TAXONOMYLINKSRESPONSE_GENERICLINK = _TAXONOMYLINKSRESPONSE.nested_types_by_name['GenericLink']
_GENECOUNTSBYTAXONREQUEST = DESCRIPTOR.message_types_by_name['GeneCountsByTaxonRequest']
_GENECOUNTSBYTAXONREPLY = DESCRIPTOR.message_types_by_name['GeneCountsByTaxonReply']
_GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT = _GENECOUNTSBYTAXONREPLY.nested_types_by_name['GeneTypeAndCount']
_GENOMEANNOTATIONREQUEST = DESCRIPTOR.message_types_by_name['GenomeAnnotationRequest']
_GENOMEANNOTATIONTABLESUMMARYREPLY = DESCRIPTOR.message_types_by_name['GenomeAnnotationTableSummaryReply']
_HTTPBODY = DESCRIPTOR.message_types_by_name['HttpBody']
_ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE = _ASSEMBLYDATASETREPORTSREQUEST.enum_types_by_name['ContentType']
_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE = _ASSEMBLYDATASETDESCRIPTORSFILTER.enum_types_by_name['AssemblySource']
_ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION = _ASSEMBLYDATASETDESCRIPTORSFILTER.enum_types_by_name['AssemblyVersion']
_ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER = _ASSEMBLYDATASETDESCRIPTORSFILTER.enum_types_by_name['MetagenomeDerivedFilter']
_ANNOTATIONFORASSEMBLY_TYPE = _ANNOTATIONFORASSEMBLY.enum_types_by_name['Type']
_ASSEMBLYDATASETREQUEST_RESOLUTION = _ASSEMBLYDATASETREQUEST.enum_types_by_name['Resolution']
_ORGANISMQUERYREQUEST_TAXRANKFILTER = _ORGANISMQUERYREQUEST.enum_types_by_name['TaxRankFilter']
_ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER = _ORGANISMQUERYREQUEST.enum_types_by_name['TaxonResourceFilter']
_ORTHOLOGREQUEST_CONTENTTYPE = _ORTHOLOGREQUEST.enum_types_by_name['ContentType']
_GENEDATASETREQUEST_CONTENTTYPE = _GENEDATASETREQUEST.enum_types_by_name['ContentType']
_GENEDATASETREQUEST_GENEDATASETREPORTTYPE = _GENEDATASETREQUEST.enum_types_by_name['GeneDatasetReportType']
_GENEDATASETREPORTSREQUEST_CONTENTTYPE = _GENEDATASETREPORTSREQUEST.enum_types_by_name['ContentType']
_GENELINKSREPLY_GENELINKTYPE = _GENELINKSREPLY.enum_types_by_name['GeneLinkType']
_ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE = _ASSEMBLYLINKSREPLY.enum_types_by_name['AssemblyLinkType']
_VIRUSDATAREPORTREQUEST_CONTENTTYPE = _VIRUSDATAREPORTREQUEST.enum_types_by_name['ContentType']
_MICROBIGGEDATASETREQUEST_FILETYPE = _MICROBIGGEDATASETREQUEST.enum_types_by_name['FileType']
_TAXONOMYMETADATAREQUEST_CONTENTTYPE = _TAXONOMYMETADATAREQUEST.enum_types_by_name['ContentType']
_TAXONOMYMETADATAREQUEST_TABLEFORMAT = _TAXONOMYMETADATAREQUEST.enum_types_by_name['TableFormat']
_TAXONOMYDATASETREQUEST_TAXONOMYREPORTTYPE = _TAXONOMYDATASETREQUEST.enum_types_by_name['TaxonomyReportType']
_TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS = _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE.enum_types_by_name['ChildStatus']
_GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE = _GENOMEANNOTATIONREQUEST.enum_types_by_name['AnnotationType']
_GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT = _GENOMEANNOTATIONREQUEST.enum_types_by_name['GenomeAnnotationTableFormat']
MethodPayloadRequest = _reflection.GeneratedProtocolMessageType('MethodPayloadRequest', (_message.Message,), {
  'DESCRIPTOR' : _METHODPAYLOADREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.MethodPayloadRequest)
  })
_sym_db.RegisterMessage(MethodPayloadRequest)

AssemblySequenceReportsRequest = _reflection.GeneratedProtocolMessageType('AssemblySequenceReportsRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYSEQUENCEREPORTSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblySequenceReportsRequest)
  })
_sym_db.RegisterMessage(AssemblySequenceReportsRequest)

SequenceReportPage = _reflection.GeneratedProtocolMessageType('SequenceReportPage', (_message.Message,), {
  'DESCRIPTOR' : _SEQUENCEREPORTPAGE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SequenceReportPage)
  })
_sym_db.RegisterMessage(SequenceReportPage)

SortField = _reflection.GeneratedProtocolMessageType('SortField', (_message.Message,), {
  'DESCRIPTOR' : _SORTFIELD,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SortField)
  })
_sym_db.RegisterMessage(SortField)

BioSampleDatasetReportsRequest = _reflection.GeneratedProtocolMessageType('BioSampleDatasetReportsRequest', (_message.Message,), {
  'DESCRIPTOR' : _BIOSAMPLEDATASETREPORTSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.BioSampleDatasetReportsRequest)
  })
_sym_db.RegisterMessage(BioSampleDatasetReportsRequest)

AssemblyDatasetReportsRequest = _reflection.GeneratedProtocolMessageType('AssemblyDatasetReportsRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETREPORTSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetReportsRequest)
  })
_sym_db.RegisterMessage(AssemblyDatasetReportsRequest)

AssemblyDatasetDescriptorsFilter = _reflection.GeneratedProtocolMessageType('AssemblyDatasetDescriptorsFilter', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETDESCRIPTORSFILTER,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetDescriptorsFilter)
  })
_sym_db.RegisterMessage(AssemblyDatasetDescriptorsFilter)

Accessions = _reflection.GeneratedProtocolMessageType('Accessions', (_message.Message,), {
  'DESCRIPTOR' : _ACCESSIONS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Accessions)
  })
_sym_db.RegisterMessage(Accessions)

AssemblyRevisionHistory = _reflection.GeneratedProtocolMessageType('AssemblyRevisionHistory', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYREVISIONHISTORY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyRevisionHistory)
  })
_sym_db.RegisterMessage(AssemblyRevisionHistory)

AssemblyRevisionHistoryRequest = _reflection.GeneratedProtocolMessageType('AssemblyRevisionHistoryRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYREVISIONHISTORYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyRevisionHistoryRequest)
  })
_sym_db.RegisterMessage(AssemblyRevisionHistoryRequest)

AssemblyCheckMHistogramRequest = _reflection.GeneratedProtocolMessageType('AssemblyCheckMHistogramRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYCHECKMHISTOGRAMREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyCheckMHistogramRequest)
  })
_sym_db.RegisterMessage(AssemblyCheckMHistogramRequest)

AssemblyCheckMHistogramReply = _reflection.GeneratedProtocolMessageType('AssemblyCheckMHistogramReply', (_message.Message,), {

  'HistogramInterval' : _reflection.GeneratedProtocolMessageType('HistogramInterval', (_message.Message,), {
    'DESCRIPTOR' : _ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyCheckMHistogramReply.HistogramInterval)
    })
  ,
  'DESCRIPTOR' : _ASSEMBLYCHECKMHISTOGRAMREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyCheckMHistogramReply)
  })
_sym_db.RegisterMessage(AssemblyCheckMHistogramReply)
_sym_db.RegisterMessage(AssemblyCheckMHistogramReply.HistogramInterval)

BioProject = _reflection.GeneratedProtocolMessageType('BioProject', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECT,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.BioProject)
  })
_sym_db.RegisterMessage(BioProject)

BioProjectLineage = _reflection.GeneratedProtocolMessageType('BioProjectLineage', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECTLINEAGE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.BioProjectLineage)
  })
_sym_db.RegisterMessage(BioProjectLineage)

BuscoStat = _reflection.GeneratedProtocolMessageType('BuscoStat', (_message.Message,), {
  'DESCRIPTOR' : _BUSCOSTAT,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.BuscoStat)
  })
_sym_db.RegisterMessage(BuscoStat)

AnnotationForAssembly = _reflection.GeneratedProtocolMessageType('AnnotationForAssembly', (_message.Message,), {

  'File' : _reflection.GeneratedProtocolMessageType('File', (_message.Message,), {
    'DESCRIPTOR' : _ANNOTATIONFORASSEMBLY_FILE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AnnotationForAssembly.File)
    })
  ,
  'DESCRIPTOR' : _ANNOTATIONFORASSEMBLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AnnotationForAssembly)
  })
_sym_db.RegisterMessage(AnnotationForAssembly)
_sym_db.RegisterMessage(AnnotationForAssembly.File)

FeatureCounts = _reflection.GeneratedProtocolMessageType('FeatureCounts', (_message.Message,), {
  'DESCRIPTOR' : _FEATURECOUNTS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.FeatureCounts)
  })
_sym_db.RegisterMessage(FeatureCounts)

GeneCounts = _reflection.GeneratedProtocolMessageType('GeneCounts', (_message.Message,), {
  'DESCRIPTOR' : _GENECOUNTS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCounts)
  })
_sym_db.RegisterMessage(GeneCounts)

AssemblyDatasetRequest = _reflection.GeneratedProtocolMessageType('AssemblyDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetRequest)
  })
_sym_db.RegisterMessage(AssemblyDatasetRequest)

AssemblyDatasetAvailability = _reflection.GeneratedProtocolMessageType('AssemblyDatasetAvailability', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATASETAVAILABILITY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyDatasetAvailability)
  })
_sym_db.RegisterMessage(AssemblyDatasetAvailability)

Organism = _reflection.GeneratedProtocolMessageType('Organism', (_message.Message,), {

  'Counts' : _reflection.GeneratedProtocolMessageType('Counts', (_message.Message,), {
    'DESCRIPTOR' : _ORGANISM_COUNTS,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Organism.Counts)
    })
  ,

  'CountByType' : _reflection.GeneratedProtocolMessageType('CountByType', (_message.Message,), {
    'DESCRIPTOR' : _ORGANISM_COUNTBYTYPE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Organism.CountByType)
    })
  ,
  'DESCRIPTOR' : _ORGANISM,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Organism)
  })
_sym_db.RegisterMessage(Organism)
_sym_db.RegisterMessage(Organism.Counts)
_sym_db.RegisterMessage(Organism.CountByType)

OrganismQueryRequest = _reflection.GeneratedProtocolMessageType('OrganismQueryRequest', (_message.Message,), {
  'DESCRIPTOR' : _ORGANISMQUERYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.OrganismQueryRequest)
  })
_sym_db.RegisterMessage(OrganismQueryRequest)

SciNameAndIds = _reflection.GeneratedProtocolMessageType('SciNameAndIds', (_message.Message,), {

  'SciNameAndId' : _reflection.GeneratedProtocolMessageType('SciNameAndId', (_message.Message,), {
    'DESCRIPTOR' : _SCINAMEANDIDS_SCINAMEANDID,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SciNameAndIds.SciNameAndId)
    })
  ,
  'DESCRIPTOR' : _SCINAMEANDIDS,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SciNameAndIds)
  })
_sym_db.RegisterMessage(SciNameAndIds)
_sym_db.RegisterMessage(SciNameAndIds.SciNameAndId)

OrthologRequest = _reflection.GeneratedProtocolMessageType('OrthologRequest', (_message.Message,), {
  'DESCRIPTOR' : _ORTHOLOGREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.OrthologRequest)
  })
_sym_db.RegisterMessage(OrthologRequest)

GeneDatasetRequest = _reflection.GeneratedProtocolMessageType('GeneDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENEDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneDatasetRequest)
  })
_sym_db.RegisterMessage(GeneDatasetRequest)

GeneDatasetReportsRequest = _reflection.GeneratedProtocolMessageType('GeneDatasetReportsRequest', (_message.Message,), {

  'SymbolsForTaxon' : _reflection.GeneratedProtocolMessageType('SymbolsForTaxon', (_message.Message,), {
    'DESCRIPTOR' : _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneDatasetReportsRequest.SymbolsForTaxon)
    })
  ,
  'DESCRIPTOR' : _GENEDATASETREPORTSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneDatasetReportsRequest)
  })
_sym_db.RegisterMessage(GeneDatasetReportsRequest)
_sym_db.RegisterMessage(GeneDatasetReportsRequest.SymbolsForTaxon)

GeneLinksRequest = _reflection.GeneratedProtocolMessageType('GeneLinksRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENELINKSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneLinksRequest)
  })
_sym_db.RegisterMessage(GeneLinksRequest)

GeneLinksReply = _reflection.GeneratedProtocolMessageType('GeneLinksReply', (_message.Message,), {

  'GeneLink' : _reflection.GeneratedProtocolMessageType('GeneLink', (_message.Message,), {
    'DESCRIPTOR' : _GENELINKSREPLY_GENELINK,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneLinksReply.GeneLink)
    })
  ,
  'DESCRIPTOR' : _GENELINKSREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneLinksReply)
  })
_sym_db.RegisterMessage(GeneLinksReply)
_sym_db.RegisterMessage(GeneLinksReply.GeneLink)

GeneChromosomeSummaryRequest = _reflection.GeneratedProtocolMessageType('GeneChromosomeSummaryRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENECHROMOSOMESUMMARYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneChromosomeSummaryRequest)
  })
_sym_db.RegisterMessage(GeneChromosomeSummaryRequest)

GeneChromosomeSummaryReply = _reflection.GeneratedProtocolMessageType('GeneChromosomeSummaryReply', (_message.Message,), {

  'GeneChromosomeSummary' : _reflection.GeneratedProtocolMessageType('GeneChromosomeSummary', (_message.Message,), {
    'DESCRIPTOR' : _GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneChromosomeSummaryReply.GeneChromosomeSummary)
    })
  ,
  'DESCRIPTOR' : _GENECHROMOSOMESUMMARYREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneChromosomeSummaryReply)
  })
_sym_db.RegisterMessage(GeneChromosomeSummaryReply)
_sym_db.RegisterMessage(GeneChromosomeSummaryReply.GeneChromosomeSummary)

AssemblyLinksRequest = _reflection.GeneratedProtocolMessageType('AssemblyLinksRequest', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYLINKSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyLinksRequest)
  })
_sym_db.RegisterMessage(AssemblyLinksRequest)

AssemblyLinksReply = _reflection.GeneratedProtocolMessageType('AssemblyLinksReply', (_message.Message,), {

  'AssemblyLink' : _reflection.GeneratedProtocolMessageType('AssemblyLink', (_message.Message,), {
    'DESCRIPTOR' : _ASSEMBLYLINKSREPLY_ASSEMBLYLINK,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyLinksReply.AssemblyLink)
    })
  ,
  'DESCRIPTOR' : _ASSEMBLYLINKSREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.AssemblyLinksReply)
  })
_sym_db.RegisterMessage(AssemblyLinksReply)
_sym_db.RegisterMessage(AssemblyLinksReply.AssemblyLink)

VirusDatasetRequest = _reflection.GeneratedProtocolMessageType('VirusDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusDatasetRequest)
  })
_sym_db.RegisterMessage(VirusDatasetRequest)

VirusDatasetFilter = _reflection.GeneratedProtocolMessageType('VirusDatasetFilter', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATASETFILTER,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusDatasetFilter)
  })
_sym_db.RegisterMessage(VirusDatasetFilter)

VirusDataReportRequest = _reflection.GeneratedProtocolMessageType('VirusDataReportRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSDATAREPORTREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusDataReportRequest)
  })
_sym_db.RegisterMessage(VirusDataReportRequest)

VirusAnnotationFilter = _reflection.GeneratedProtocolMessageType('VirusAnnotationFilter', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSANNOTATIONFILTER,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAnnotationFilter)
  })
_sym_db.RegisterMessage(VirusAnnotationFilter)

VirusAnnotationReportRequest = _reflection.GeneratedProtocolMessageType('VirusAnnotationReportRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSANNOTATIONREPORTREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAnnotationReportRequest)
  })
_sym_db.RegisterMessage(VirusAnnotationReportRequest)

VirusAvailabilityRequest = _reflection.GeneratedProtocolMessageType('VirusAvailabilityRequest', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSAVAILABILITYREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAvailabilityRequest)
  })
_sym_db.RegisterMessage(VirusAvailabilityRequest)

VirusAvailability = _reflection.GeneratedProtocolMessageType('VirusAvailability', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSAVAILABILITY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VirusAvailability)
  })
_sym_db.RegisterMessage(VirusAvailability)

Sars2ProteinDatasetRequest = _reflection.GeneratedProtocolMessageType('Sars2ProteinDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _SARS2PROTEINDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.Sars2ProteinDatasetRequest)
  })
_sym_db.RegisterMessage(Sars2ProteinDatasetRequest)

DownloadSummary = _reflection.GeneratedProtocolMessageType('DownloadSummary', (_message.Message,), {

  'Hydrated' : _reflection.GeneratedProtocolMessageType('Hydrated', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_HYDRATED,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.Hydrated)
    })
  ,

  'Dehydrated' : _reflection.GeneratedProtocolMessageType('Dehydrated', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_DEHYDRATED,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.Dehydrated)
    })
  ,

  'FileSummary' : _reflection.GeneratedProtocolMessageType('FileSummary', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_FILESUMMARY,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.FileSummary)
    })
  ,

  'AvailableFiles' : _reflection.GeneratedProtocolMessageType('AvailableFiles', (_message.Message,), {
    'DESCRIPTOR' : _DOWNLOADSUMMARY_AVAILABLEFILES,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary.AvailableFiles)
    })
  ,
  'DESCRIPTOR' : _DOWNLOADSUMMARY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DownloadSummary)
  })
_sym_db.RegisterMessage(DownloadSummary)
_sym_db.RegisterMessage(DownloadSummary.Hydrated)
_sym_db.RegisterMessage(DownloadSummary.Dehydrated)
_sym_db.RegisterMessage(DownloadSummary.FileSummary)
_sym_db.RegisterMessage(DownloadSummary.AvailableFiles)

TabularOutput = _reflection.GeneratedProtocolMessageType('TabularOutput', (_message.Message,), {
  'DESCRIPTOR' : _TABULAROUTPUT,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TabularOutput)
  })
_sym_db.RegisterMessage(TabularOutput)

MicroBiggeDatasetRequest = _reflection.GeneratedProtocolMessageType('MicroBiggeDatasetRequest', (_message.Message,), {

  'ElementFlankConfig' : _reflection.GeneratedProtocolMessageType('ElementFlankConfig', (_message.Message,), {
    'DESCRIPTOR' : _MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.MicroBiggeDatasetRequest.ElementFlankConfig)
    })
  ,
  'DESCRIPTOR' : _MICROBIGGEDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.MicroBiggeDatasetRequest)
  })
_sym_db.RegisterMessage(MicroBiggeDatasetRequest)
_sym_db.RegisterMessage(MicroBiggeDatasetRequest.ElementFlankConfig)

VersionReply = _reflection.GeneratedProtocolMessageType('VersionReply', (_message.Message,), {
  'DESCRIPTOR' : _VERSIONREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.VersionReply)
  })
_sym_db.RegisterMessage(VersionReply)

ProkaryoteGeneRequest = _reflection.GeneratedProtocolMessageType('ProkaryoteGeneRequest', (_message.Message,), {

  'GeneFlankConfig' : _reflection.GeneratedProtocolMessageType('GeneFlankConfig', (_message.Message,), {
    'DESCRIPTOR' : _PROKARYOTEGENEREQUEST_GENEFLANKCONFIG,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.ProkaryoteGeneRequest.GeneFlankConfig)
    })
  ,
  'DESCRIPTOR' : _PROKARYOTEGENEREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.ProkaryoteGeneRequest)
  })
_sym_db.RegisterMessage(ProkaryoteGeneRequest)
_sym_db.RegisterMessage(ProkaryoteGeneRequest.GeneFlankConfig)

SleepRequest = _reflection.GeneratedProtocolMessageType('SleepRequest', (_message.Message,), {
  'DESCRIPTOR' : _SLEEPREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SleepRequest)
  })
_sym_db.RegisterMessage(SleepRequest)

SleepReply = _reflection.GeneratedProtocolMessageType('SleepReply', (_message.Message,), {
  'DESCRIPTOR' : _SLEEPREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.SleepReply)
  })
_sym_db.RegisterMessage(SleepReply)

TaxonomyNode = _reflection.GeneratedProtocolMessageType('TaxonomyNode', (_message.Message,), {

  'CountByType' : _reflection.GeneratedProtocolMessageType('CountByType', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYNODE_COUNTBYTYPE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyNode.CountByType)
    })
  ,
  'DESCRIPTOR' : _TAXONOMYNODE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyNode)
  })
_sym_db.RegisterMessage(TaxonomyNode)
_sym_db.RegisterMessage(TaxonomyNode.CountByType)

TaxonomyMetadataRequest = _reflection.GeneratedProtocolMessageType('TaxonomyMetadataRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMETADATAREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyMetadataRequest)
  })
_sym_db.RegisterMessage(TaxonomyMetadataRequest)

TaxonomyRelatedIdRequest = _reflection.GeneratedProtocolMessageType('TaxonomyRelatedIdRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYRELATEDIDREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyRelatedIdRequest)
  })
_sym_db.RegisterMessage(TaxonomyRelatedIdRequest)

TaxonomyTaxIdsPage = _reflection.GeneratedProtocolMessageType('TaxonomyTaxIdsPage', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYTAXIDSPAGE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyTaxIdsPage)
  })
_sym_db.RegisterMessage(TaxonomyTaxIdsPage)

TaxonomyDatasetRequest = _reflection.GeneratedProtocolMessageType('TaxonomyDatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYDATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyDatasetRequest)
  })
_sym_db.RegisterMessage(TaxonomyDatasetRequest)

TaxonomyMatch = _reflection.GeneratedProtocolMessageType('TaxonomyMatch', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMATCH,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyMatch)
  })
_sym_db.RegisterMessage(TaxonomyMatch)

TaxonomyMetadataResponse = _reflection.GeneratedProtocolMessageType('TaxonomyMetadataResponse', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYMETADATARESPONSE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyMetadataResponse)
  })
_sym_db.RegisterMessage(TaxonomyMetadataResponse)

TaxonomyImageRequest = _reflection.GeneratedProtocolMessageType('TaxonomyImageRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYIMAGEREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyImageRequest)
  })
_sym_db.RegisterMessage(TaxonomyImageRequest)

TaxonomyImageMetadataRequest = _reflection.GeneratedProtocolMessageType('TaxonomyImageMetadataRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYIMAGEMETADATAREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyImageMetadataRequest)
  })
_sym_db.RegisterMessage(TaxonomyImageMetadataRequest)

TaxonomyImageMetadataResponse = _reflection.GeneratedProtocolMessageType('TaxonomyImageMetadataResponse', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYIMAGEMETADATARESPONSE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyImageMetadataResponse)
  })
_sym_db.RegisterMessage(TaxonomyImageMetadataResponse)

TaxonomyFilteredSubtreeRequest = _reflection.GeneratedProtocolMessageType('TaxonomyFilteredSubtreeRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREEREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeRequest)
  })
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeRequest)

TaxonomyFilteredSubtreeResponse = _reflection.GeneratedProtocolMessageType('TaxonomyFilteredSubtreeResponse', (_message.Message,), {

  'Edge' : _reflection.GeneratedProtocolMessageType('Edge', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.Edge)
    })
  ,

  'EdgesEntry' : _reflection.GeneratedProtocolMessageType('EdgesEntry', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse.EdgesEntry)
    })
  ,
  'DESCRIPTOR' : _TAXONOMYFILTEREDSUBTREERESPONSE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyFilteredSubtreeResponse)
  })
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse)
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse.Edge)
_sym_db.RegisterMessage(TaxonomyFilteredSubtreeResponse.EdgesEntry)

DatasetRequest = _reflection.GeneratedProtocolMessageType('DatasetRequest', (_message.Message,), {
  'DESCRIPTOR' : _DATASETREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.DatasetRequest)
  })
_sym_db.RegisterMessage(DatasetRequest)

TaxonomyLinksRequest = _reflection.GeneratedProtocolMessageType('TaxonomyLinksRequest', (_message.Message,), {
  'DESCRIPTOR' : _TAXONOMYLINKSREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyLinksRequest)
  })
_sym_db.RegisterMessage(TaxonomyLinksRequest)

TaxonomyLinksResponse = _reflection.GeneratedProtocolMessageType('TaxonomyLinksResponse', (_message.Message,), {

  'GenericLink' : _reflection.GeneratedProtocolMessageType('GenericLink', (_message.Message,), {
    'DESCRIPTOR' : _TAXONOMYLINKSRESPONSE_GENERICLINK,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyLinksResponse.GenericLink)
    })
  ,
  'DESCRIPTOR' : _TAXONOMYLINKSRESPONSE,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.TaxonomyLinksResponse)
  })
_sym_db.RegisterMessage(TaxonomyLinksResponse)
_sym_db.RegisterMessage(TaxonomyLinksResponse.GenericLink)

GeneCountsByTaxonRequest = _reflection.GeneratedProtocolMessageType('GeneCountsByTaxonRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENECOUNTSBYTAXONREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCountsByTaxonRequest)
  })
_sym_db.RegisterMessage(GeneCountsByTaxonRequest)

GeneCountsByTaxonReply = _reflection.GeneratedProtocolMessageType('GeneCountsByTaxonReply', (_message.Message,), {

  'GeneTypeAndCount' : _reflection.GeneratedProtocolMessageType('GeneTypeAndCount', (_message.Message,), {
    'DESCRIPTOR' : _GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT,
    '__module__' : 'ncbi.datasets.v2.datasets_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCountsByTaxonReply.GeneTypeAndCount)
    })
  ,
  'DESCRIPTOR' : _GENECOUNTSBYTAXONREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GeneCountsByTaxonReply)
  })
_sym_db.RegisterMessage(GeneCountsByTaxonReply)
_sym_db.RegisterMessage(GeneCountsByTaxonReply.GeneTypeAndCount)

GenomeAnnotationRequest = _reflection.GeneratedProtocolMessageType('GenomeAnnotationRequest', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATIONREQUEST,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GenomeAnnotationRequest)
  })
_sym_db.RegisterMessage(GenomeAnnotationRequest)

GenomeAnnotationTableSummaryReply = _reflection.GeneratedProtocolMessageType('GenomeAnnotationTableSummaryReply', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATIONTABLESUMMARYREPLY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.GenomeAnnotationTableSummaryReply)
  })
_sym_db.RegisterMessage(GenomeAnnotationTableSummaryReply)

HttpBody = _reflection.GeneratedProtocolMessageType('HttpBody', (_message.Message,), {
  'DESCRIPTOR' : _HTTPBODY,
  '__module__' : 'ncbi.datasets.v2.datasets_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.HttpBody)
  })
_sym_db.RegisterMessage(HttpBody)

if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'Z\020ncbi/datasets/v2\370\001\001'
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['accession']._options = None
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['accession']._serialized_options = b'\302\360\0316\022\031\n\020GCF_000001635.27\022\005Mouse\022\031\n\020GCF_000001405.40\022\005Human'
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['chromosomes']._options = None
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['chromosomes']._serialized_options = b'\302\360\031M\022,\n\0011\n\0012\n\0013\n\001X\n\001Y\n\002MT\022\027Selected chromosomes #1\022\035\n\001X\n\001Y\022\025Human sex chromosomes'
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['role_filters']._options = None
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['role_filters']._serialized_options = b'\302\360\031\206\001\022T\n\022assembled-molecule\n\024unlocalized-scaffold\n\021unplaced-scaffold\022\025All types of sequence\022.\n\022assembled-molecule\022\030Only assembled molecules'
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['table_fields']._options = None
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031[\022;\n\taccession\n\010chr-name\022$Genome Accession and Chromosome Name\022\034\n\017ucsc-style-name\022\tUCSC Name'
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['count_assembly_unplaced']._options = None
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['count_assembly_unplaced']._serialized_options = b'\302\360\031\007\n\005false'
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['page_size']._options = None
  _ASSEMBLYSEQUENCEREPORTSREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _SORTFIELD.fields_by_name['field']._options = None
  _SORTFIELD.fields_by_name['field']._serialized_options = b'\302\360\031\303\005\022\013\n\taccession\022\027\n\025versionless-accession\022\r\n\013search-text\022\010\n\006tax-id\022\023\n\021tax-ordinal-range\022\017\n\rorganism-name\022\013\n\tsubmitter\022\025\n\023assembly-level-enum\022\021\n\017assembly-status\022\016\n\014release-date\022\017\n\ris-ref-genome\022\017\n\ris-rep-genome\022\013\n\tis-refseq\022\014\n\nis-primary\022\014\n\nis-current\022\034\n\032paired-assembly-is-current\022!\n\037paired-assembly-annotation-name\022\020\n\016has-annotation\022\r\n\013is-atypical\022\026\n\024bioproject-accession\022\027\n\025total-sequence-length\022\021\n\017pair-is-current\022\035\n\033pair-has-genbank-annotation\022\034\n\032pair-has-refseq-annotation\022\025\n\023pair-has-annotation\022\014\n\ncontig-n50\022\016\n\014scaffold-n50\022\020\n\016busco-complete\022\023\n\021genbank-accession\022\022\n\020refseq-accession\022\021\n\017annotation-name\022\017\n\rwgs-accession\022\021\n\017sequencing-tech\022\025\n\023biosample-accession\022\014\n\ngene-count\022\017\n\rprotein-count\022\n\n\010modifier'
  _BIOSAMPLEDATASETREPORTSREQUEST.fields_by_name['accessions']._options = None
  _BIOSAMPLEDATASETREPORTSREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031(\022&\n\016SAMEA103927779\022\024BioSample Identifier'
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['taxons']._options = None
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['taxons']._serialized_options = b'\302\360\031p\022 \n\0049606\022\030NCBI Taxonomy Identifier\022+\n\nchimpanzee\n\017eastern gorilla\022\014Common Names\022\037\n\014Homo sapiens\022\017Scientific Name'
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['bioprojects']._options = None
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['bioprojects']._serialized_options = b'\302\360\031O\022)\n\013PRJNA489243\022\032Vertebrate Genomes Project\022\"\n\nPRJNA31257\022\024Human Genome Project'
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['accessions']._options = None
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human'
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['tax_exact_match']._options = None
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['tax_exact_match']._serialized_options = b'\302\360\031\007\n\005false'
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['chromosomes']._options = None
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['chromosomes']._serialized_options = b'\030\001'
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['table_fields']._options = None
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031f\022>\n\022assminfo-accession\n\rassminfo-name\022\031Genome Accession and Name\022$\n\022assminfo-accession\022\016Just accession'
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['page_size']._options = None
  _ASSEMBLYDATASETREPORTSREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['reference_only']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['reference_only']._serialized_options = b'\302\360\031r\n\005false\0223\n\004true\022+Limit to reference & representative genomes\0224\n\005false\022+Include reference and non-reference genomes'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_source']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_source']._serialized_options = b'\302\360\031a\n\003all\022!\n\006refseq\022\027Limit to RefSeq genomes\022#\n\007genbank\022\030Limit to GenBank genomes\022\022\n\003all\022\013All genomes'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['has_annotation']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['has_annotation']._serialized_options = b'\302\360\031N\n\005false\022%\n\004true\022\035Limit to annotated assemblies\022\036\n\005false\022\025Return all assemblies'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_paired_reports']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_paired_reports']._serialized_options = b'\302\360\031\\\n\005false\022\036\n\005false\022\025Return all assemblies\0223\n\004true\022+Only return single record for GCA/GCF pairs'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_atypical']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['exclude_atypical']._serialized_options = b'\302\360\031\226\001\n\005false\022F\n\005false\022=Return all selected genomes, including ones that are atypical\022E\n\004true\022=Only return genomes that have not been classified as atypical'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_version']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_version']._serialized_options = b'\302\360\031\200\001\n\007current\022)\n\007current\022\036Only return current assemblies\022J\n\016all_assemblies\0228Return all assemblies, including suppressed and replaced'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_level']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['assembly_level']._serialized_options = b'\302\360\031}\022=\n\nchromosome\n\017complete_genome\022\036Chromosome and complete genome\022<\n\010scaffold\n\nchromosome\n\017complete_genome\022\023Scaffold and higher'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['first_release_date']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['first_release_date']._serialized_options = b'\302\360\031\034\022\032\n\n01/10/2015\022\014Jan 10, 2015'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['last_release_date']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['last_release_date']._serialized_options = b'\302\360\031\034\022\032\n\n01/10/2021\022\014Jan 10, 2021'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['search_text']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['search_text']._serialized_options = b'\302\360\031d\022:\n\033Genome Reference Consortium\022\033Genome Reference Consortium\022\020\n\006GRCh38\022\006GRCh38\022\024\n\010C57BL/6J\022\010C57BL/6J'
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['is_metagenome_derived']._options = None
  _ASSEMBLYDATASETDESCRIPTORSFILTER.fields_by_name['is_metagenome_derived']._serialized_options = b'\302\360\031\213\001\022D\n\027metagenome_derived_only\022)Return only metagenome derived assemblies\022C\n\032metagenome_derived_exclude\022%Exclude metagenome derived assemblies'
  _ACCESSIONS.fields_by_name['accessions']._options = None
  _ACCESSIONS.fields_by_name['accessions']._serialized_options = b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human'
  _ASSEMBLYREVISIONHISTORYREQUEST.fields_by_name['accession']._options = None
  _ASSEMBLYREVISIONHISTORYREQUEST.fields_by_name['accession']._serialized_options = b'\302\360\031\033\022\031\n\020GCF_000001405.40\022\005Human'
  _ASSEMBLYCHECKMHISTOGRAMREQUEST.fields_by_name['species_taxon']._options = None
  _ASSEMBLYCHECKMHISTOGRAMREQUEST.fields_by_name['species_taxon']._serialized_options = b'\302\360\031!\022\037\n\006202956\022\025Acinetobacter towneri'
  _ASSEMBLYCHECKMHISTOGRAMREPLY.fields_by_name['species_taxid']._options = None
  _ASSEMBLYCHECKMHISTOGRAMREPLY.fields_by_name['species_taxid']._serialized_options = b'\302\360\031!\022\037\n\006202956\022\025Acinetobacter towneri'
  _ANNOTATIONFORASSEMBLY_TYPE._options = None
  _ANNOTATIONFORASSEMBLY_TYPE._serialized_options = b'\020\001'
  _ANNOTATIONFORASSEMBLY_TYPE.values_by_name["GENOME_GB"]._options = None
  _ANNOTATIONFORASSEMBLY_TYPE.values_by_name["GENOME_GB"]._serialized_options = b'\010\001'
  _ANNOTATIONFORASSEMBLY_TYPE.values_by_name["ASM_STRUCT"]._options = None
  _ANNOTATIONFORASSEMBLY_TYPE.values_by_name["ASM_STRUCT"]._serialized_options = b'\302\360\031\002\010\001'
  _ASSEMBLYDATASETREQUEST.fields_by_name['accessions']._options = None
  _ASSEMBLYDATASETREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human'
  _ASSEMBLYDATASETREQUEST.fields_by_name['chromosomes']._options = None
  _ASSEMBLYDATASETREQUEST.fields_by_name['chromosomes']._serialized_options = b'\302\360\031M\022,\n\0011\n\0012\n\0013\n\001X\n\001Y\n\002MT\022\027Selected chromosomes #1\022\035\n\001X\n\001Y\022\025Human sex chromosomes'
  _ASSEMBLYDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
  _ASSEMBLYDATASETREQUEST.fields_by_name['include_annotation_type']._serialized_options = b'\332\360\031\002\010\002\302\360\031\256\001\022M\n\014GENOME_FASTA\n\nPROT_FASTA\n\tRNA_FASTA\022&Genomic, Transcript & Protein Sequence\022]\n\nGENOME_GFF\n\013GENOME_GBFF\n\nGENOME_GTF\n\nPROT_FASTA\n\tRNA_FASTA\n\032CDS_FASTA, SEQUENCE_REPORT\022\003All'
  _ASSEMBLYDATASETREQUEST.fields_by_name['include_tsv']._options = None
  _ASSEMBLYDATASETREQUEST.fields_by_name['include_tsv']._serialized_options = b'\332\360\031\002\010\002\302\360\031\007\n\005false'
  _ASSEMBLYDATASETREQUEST._options = None
  _ASSEMBLYDATASETREQUEST._serialized_options = b'\302\360\031 \n\036accessions:\n- GCF_000001405.40'
  _ORGANISM.fields_by_name['assembly_count']._options = None
  _ORGANISM.fields_by_name['assembly_count']._serialized_options = b'\030\001'
  _ORGANISMQUERYREQUEST.fields_by_name['organism_query']._options = None
  _ORGANISMQUERYREQUEST.fields_by_name['organism_query']._serialized_options = b'\030\001'
  _ORGANISMQUERYREQUEST.fields_by_name['taxon_query']._options = None
  _ORGANISMQUERYREQUEST.fields_by_name['taxon_query']._serialized_options = b'\302\360\031\014\022\n\n\003hum\022\003hum'
  _ORGANISMQUERYREQUEST.fields_by_name['tax_rank_filter']._options = None
  _ORGANISMQUERYREQUEST.fields_by_name['tax_rank_filter']._serialized_options = b'\302\360\031\t\n\007species'
  _ORGANISMQUERYREQUEST.fields_by_name['taxon_resource_filter']._options = None
  _ORGANISMQUERYREQUEST.fields_by_name['taxon_resource_filter']._serialized_options = b'\302\360\031\033\n\031TAXON_RESOURCE_FILTER_ALL'
  _ORGANISMQUERYREQUEST.fields_by_name['exact_match']._options = None
  _ORGANISMQUERYREQUEST.fields_by_name['exact_match']._serialized_options = b'\302\360\031\007\n\005false'
  _ORTHOLOGREQUEST.fields_by_name['gene_id']._options = None
  _ORTHOLOGREQUEST.fields_by_name['gene_id']._serialized_options = b'\302\360\031\030\022\014\n\0042778\022\004GNAS\022\010\n\0012\022\003A2M'
  _ORTHOLOGREQUEST.fields_by_name['taxon_filter']._options = None
  _ORTHOLOGREQUEST.fields_by_name['taxon_filter']._serialized_options = b'\302\360\031{\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022!\n\005human\n\013house mouse\022\013Common Name\022-\n\014homo sapiens\n\014mus musculus\022\017Scientific Name'
  _ORTHOLOGREQUEST.fields_by_name['page_size']._options = None
  _ORTHOLOGREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _GENEDATASETREQUEST.fields_by_name['gene_ids']._options = None
  _GENEDATASETREQUEST.fields_by_name['gene_ids']._serialized_options = b'\302\360\031<\022\024\n\00559067\022\013IL21 GeneID\022$\n\00559067\n\00550615\022\024IL21 & IL21R GeneIDs'
  _GENEDATASETREQUEST.fields_by_name['include_annotation_type']._options = None
  _GENEDATASETREQUEST.fields_by_name['include_annotation_type']._serialized_options = b'\332\360\031\002\010\002'
  _GENEDATASETREQUEST.fields_by_name['aux_report']._options = None
  _GENEDATASETREQUEST.fields_by_name['aux_report']._serialized_options = b'\302\360\031z\022x\n\016PRODUCT_REPORT\022fSelect additional report types to include with download. The base data report will always be included.'
  _GENEDATASETREQUEST.fields_by_name['table_fields']._options = None
  _GENEDATASETREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031Y\022?\n\007gene-id\n\tgene-type\n\013description\022\034GeneID, type and description\022\026\n\007gene-id\022\013Just GeneID'
  _GENEDATASETREQUEST._options = None
  _GENEDATASETREQUEST._serialized_options = b'\302\360\031\033\n\031gene_ids:\n- 59067\n- 50615'
  _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON.fields_by_name['symbols']._options = None
  _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON.fields_by_name['symbols']._serialized_options = b'\302\360\031\016\022\014\n\004GNAS\022\004GNAS'
  _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON.fields_by_name['taxon']._options = None
  _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON.fields_by_name['taxon']._serialized_options = b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name'
  _GENEDATASETREPORTSREQUEST.fields_by_name['gene_ids']._options = None
  _GENEDATASETREPORTSREQUEST.fields_by_name['gene_ids']._serialized_options = b'\302\360\031<\022\024\n\00559067\022\013IL21 GeneID\022$\n\00559067\n\00550615\022\024IL21 & IL21R GeneIDs'
  _GENEDATASETREPORTSREQUEST.fields_by_name['accessions']._options = None
  _GENEDATASETREPORTSREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031]\022.\n\013NM_021803.4\n\013NM_181078.3\022\022RefSeq transcripts\022+\n\013NP_068575.1\n\013NP_851564.1\022\017RefSeq proteins'
  _GENEDATASETREPORTSREQUEST.fields_by_name['taxon']._options = None
  _GENEDATASETREPORTSREQUEST.fields_by_name['taxon']._serialized_options = b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name'
  _GENEDATASETREPORTSREQUEST.fields_by_name['table_fields']._options = None
  _GENEDATASETREPORTSREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031Y\022?\n\007gene-id\n\tgene-type\n\013description\022\034GeneID, type and description\022\026\n\007gene-id\022\013Just GeneID'
  _GENEDATASETREPORTSREQUEST.fields_by_name['page_size']._options = None
  _GENEDATASETREPORTSREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _GENEDATASETREPORTSREQUEST.fields_by_name['query']._options = None
  _GENEDATASETREPORTSREQUEST.fields_by_name['query']._serialized_options = b'\302\360\031\026\022\024\n\022A2M immunoglobulin'
  _GENEDATASETREPORTSREQUEST.fields_by_name['types']._options = None
  _GENEDATASETREPORTSREQUEST.fields_by_name['types']._serialized_options = b'\302\360\031-\022+\n\016PROTEIN_CODING\022\031Only protein coding genes'
  _GENEDATASETREPORTSREQUEST._options = None
  _GENEDATASETREPORTSREQUEST._serialized_options = b'\302\360\031\033\n\031gene_ids:\n- 59067\n- 50615'
  _GENELINKSREQUEST.fields_by_name['gene_ids']._options = None
  _GENELINKSREQUEST.fields_by_name['gene_ids']._serialized_options = b'\302\360\031<\022\024\n\00559067\022\013IL21 GeneID\022$\n\00559067\n\00550615\022\024IL21 & IL21R GeneIDs'
  _GENELINKSREQUEST._options = None
  _GENELINKSREQUEST._serialized_options = b'\302\360\031\033\n\031gene_ids:\n- 59067\n- 50615'
  _GENECHROMOSOMESUMMARYREQUEST.fields_by_name['taxon']._options = None
  _GENECHROMOSOMESUMMARYREQUEST.fields_by_name['taxon']._serialized_options = b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name'
  _GENECHROMOSOMESUMMARYREQUEST.fields_by_name['annotation_name']._options = None
  _GENECHROMOSOMESUMMARYREQUEST.fields_by_name['annotation_name']._serialized_options = b'\302\360\031=\022;\n\037Annotation Release 109.20211119\022\030Human annotation release'
  _ASSEMBLYLINKSREQUEST.fields_by_name['accessions']._options = None
  _ASSEMBLYLINKSREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031R\0225\n\020GCF_000001405.40\n\020GCF_000001635.27\022\017Human and Mouse\022\031\n\020GCF_000001405.40\022\005Human'
  _ASSEMBLYLINKSREQUEST._options = None
  _ASSEMBLYLINKSREQUEST._serialized_options = b'\302\360\0310\n.accessions:\n  accessions:\n  - GCF_000001405.40'
  _VIRUSDATASETREQUEST.fields_by_name['accessions']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031?\022=\n\013NC_045512.2\022.SARS-COV-2 isolate Wuhan-Hu-1, complete genome'
  _VIRUSDATASETREQUEST.fields_by_name['taxon']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['taxon']._serialized_options = b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS'
  _VIRUSDATASETREQUEST.fields_by_name['refseq_only']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['refseq_only']._serialized_options = b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes'
  _VIRUSDATASETREQUEST.fields_by_name['annotated_only']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['annotated_only']._serialized_options = b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes'
  _VIRUSDATASETREQUEST.fields_by_name['released_since']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['released_since']._serialized_options = b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020'
  _VIRUSDATASETREQUEST.fields_by_name['updated_since']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['updated_since']._serialized_options = b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021'
  _VIRUSDATASETREQUEST.fields_by_name['host']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['host']._serialized_options = b'\302\360\031\020\022\016\n\005human\022\005human'
  _VIRUSDATASETREQUEST.fields_by_name['geo_location']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['geo_location']._serialized_options = b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas'
  _VIRUSDATASETREQUEST.fields_by_name['complete_only']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['complete_only']._serialized_options = b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes'
  _VIRUSDATASETREQUEST.fields_by_name['table_fields']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031N\022L\n\024nucleotide_accession\n\021nucleotide_length\n\020nuc_completeness\022\017Assembly Fields'
  _VIRUSDATASETREQUEST.fields_by_name['include_sequence']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['include_sequence']._serialized_options = b'\302\360\031;\0229\n\006GENOME\n\003CDS\n\007PROTEIN\022!Select viral sequences to include'
  _VIRUSDATASETREQUEST.fields_by_name['aux_report']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['aux_report']._serialized_options = b'\302\360\031v\022t\n\nANNOTATION\022fSelect additional report types to include with download. The base data report will always be included.'
  _VIRUSDATASETREQUEST.fields_by_name['format']._options = None
  _VIRUSDATASETREQUEST.fields_by_name['format']._serialized_options = b'\302\360\0312\n\003tsv\022\n\n\003tsv\022\003TSV\022\n\n\003csv\022\003CSV\022\023\n\005jsonl\022\nJSON Lines'
  _VIRUSDATASETFILTER.fields_by_name['accessions']._options = None
  _VIRUSDATASETFILTER.fields_by_name['accessions']._serialized_options = b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes'
  _VIRUSDATASETFILTER.fields_by_name['taxon']._options = None
  _VIRUSDATASETFILTER.fields_by_name['taxon']._serialized_options = b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS'
  _VIRUSDATASETFILTER.fields_by_name['refseq_only']._options = None
  _VIRUSDATASETFILTER.fields_by_name['refseq_only']._serialized_options = b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes'
  _VIRUSDATASETFILTER.fields_by_name['annotated_only']._options = None
  _VIRUSDATASETFILTER.fields_by_name['annotated_only']._serialized_options = b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes'
  _VIRUSDATASETFILTER.fields_by_name['released_since']._options = None
  _VIRUSDATASETFILTER.fields_by_name['released_since']._serialized_options = b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020'
  _VIRUSDATASETFILTER.fields_by_name['updated_since']._options = None
  _VIRUSDATASETFILTER.fields_by_name['updated_since']._serialized_options = b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021'
  _VIRUSDATASETFILTER.fields_by_name['host']._options = None
  _VIRUSDATASETFILTER.fields_by_name['host']._serialized_options = b'\302\360\031\020\022\016\n\005human\022\005human'
  _VIRUSDATASETFILTER.fields_by_name['geo_location']._options = None
  _VIRUSDATASETFILTER.fields_by_name['geo_location']._serialized_options = b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas'
  _VIRUSDATASETFILTER.fields_by_name['complete_only']._options = None
  _VIRUSDATASETFILTER.fields_by_name['complete_only']._serialized_options = b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes'
  _VIRUSDATAREPORTREQUEST.fields_by_name['table_fields']._options = None
  _VIRUSDATAREPORTREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031B\022@\n\taccession\n\013is-complete\n\014is-annotated\022\030Virus Data Report Fields'
  _VIRUSDATAREPORTREQUEST.fields_by_name['page_size']._options = None
  _VIRUSDATAREPORTREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _VIRUSDATAREPORTREQUEST._options = None
  _VIRUSDATAREPORTREQUEST._serialized_options = b'\302\360\031\"\n taxon: 2697049\nrefseq_only: true'
  _VIRUSANNOTATIONFILTER.fields_by_name['accessions']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['accessions']._serialized_options = b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes'
  _VIRUSANNOTATIONFILTER.fields_by_name['taxon']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['taxon']._serialized_options = b'\302\360\031@\022\025\n\0072697049\022\nSARS-COV-2\022\026\n\00511118\022\rCoronaviridae\022\017\n\0071335626\022\004MERS'
  _VIRUSANNOTATIONFILTER.fields_by_name['refseq_only']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['refseq_only']._serialized_options = b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes'
  _VIRUSANNOTATIONFILTER.fields_by_name['released_since']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['released_since']._serialized_options = b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020'
  _VIRUSANNOTATIONFILTER.fields_by_name['updated_since']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['updated_since']._serialized_options = b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021'
  _VIRUSANNOTATIONFILTER.fields_by_name['host']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['host']._serialized_options = b'\302\360\031\020\022\016\n\005human\022\005human'
  _VIRUSANNOTATIONFILTER.fields_by_name['geo_location']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['geo_location']._serialized_options = b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas'
  _VIRUSANNOTATIONFILTER.fields_by_name['complete_only']._options = None
  _VIRUSANNOTATIONFILTER.fields_by_name['complete_only']._serialized_options = b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes'
  _VIRUSANNOTATIONREPORTREQUEST.fields_by_name['table_fields']._options = None
  _VIRUSANNOTATIONREPORTREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031;\0229\n\taccession\n\014isolate-name\022\036Virus Annotation Report Fields'
  _VIRUSANNOTATIONREPORTREQUEST.fields_by_name['page_size']._options = None
  _VIRUSANNOTATIONREPORTREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _VIRUSANNOTATIONREPORTREQUEST._options = None
  _VIRUSANNOTATIONREPORTREQUEST._serialized_options = b'\302\360\031\"\n taxon: 2697049\nrefseq_only: true'
  _VIRUSAVAILABILITYREQUEST.fields_by_name['accessions']._options = None
  _VIRUSAVAILABILITYREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031\206\001\022\203\001\n\027NC_045512.2,NC_038294.1\022hSARS-COV-2 isolate Wuhan-Hu-1 and Middle East respiratory syndrome-related coronavirus, complete genomes'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['proteins']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['proteins']._serialized_options = b'\302\360\031^\022\036\n\rspike protein\022\rSpike Protein\022<\n\rspike protein\n\020envelope protein\n\004RdRp\022\023SARS-CoV-2 proteins'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['refseq_only']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['refseq_only']._serialized_options = b'\302\360\031P\n\005false\022\037\n\004true\022\027Limit to RefSeq genomes\022&\n\005false\022\035Both GenBank & RefSeq genomes'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['annotated_only']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['annotated_only']._serialized_options = b'\302\360\031A\n\005false\022\"\n\004true\022\032Limit to annotated genomes\022\024\n\005false\022\013All genomes'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['released_since']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['released_since']._serialized_options = b'\302\360\031\036\022\034\n\n2020-08-01\022\016August 1, 2020'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['updated_since']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['updated_since']._serialized_options = b'\302\360\031\035\022\033\n\n2021-07-18\022\rJuly 18, 2021'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['host']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['host']._serialized_options = b'\302\360\031\020\022\016\n\005human\022\005human'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['geo_location']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['geo_location']._serialized_options = b'\302\360\031\031\022\n\n\003USA\022\003USA\022\013\n\002TX\022\005Texas'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['complete_only']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['complete_only']._serialized_options = b'\302\360\031@\n\005false\022!\n\004true\022\031Limit to complete genomes\022\024\n\005false\022\013All genomes'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['table_fields']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['table_fields']._serialized_options = b'\302\360\031N\022L\n\024nucleotide_accession\n\021nucleotide_length\n\020nuc_completeness\022\017Assembly Fields'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['include_sequence']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['include_sequence']._serialized_options = b'\302\360\0313\0221\n\003CDS\n\007PROTEIN\022!Select viral sequences to include'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['aux_report']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['aux_report']._serialized_options = b'\302\360\031v\022t\n\nANNOTATION\022fSelect additional report types to include with download. The base data report will always be included.'
  _SARS2PROTEINDATASETREQUEST.fields_by_name['format']._options = None
  _SARS2PROTEINDATASETREQUEST.fields_by_name['format']._serialized_options = b'\302\360\0312\n\003tsv\022\n\n\003tsv\022\003TSV\022\n\n\003csv\022\003CSV\022\023\n\005jsonl\022\nJSON Lines'
  _DOWNLOADSUMMARY.fields_by_name['assembly_count']._options = None
  _DOWNLOADSUMMARY.fields_by_name['assembly_count']._serialized_options = b'\030\001'
  _DOWNLOADSUMMARY.fields_by_name['errors']._options = None
  _DOWNLOADSUMMARY.fields_by_name['errors']._serialized_options = b'\030\001'
  _MICROBIGGEDATASETREQUEST.fields_by_name['files']._options = None
  _MICROBIGGEDATASETREQUEST.fields_by_name['files']._serialized_options = b'\332\360\031\002\010\002'
  _PROKARYOTEGENEREQUEST.fields_by_name['accessions']._options = None
  _PROKARYOTEGENEREQUEST.fields_by_name['accessions']._serialized_options = b'\302\360\031o\022.\n\016WP_015878339.1\022\034DNA-deoxyinosine glycosylase\022=\n\016WP_001040085.1\022+type II CRISPR RNA-guided endonuclease Cas9'
  _PROKARYOTEGENEREQUEST.fields_by_name['include_annotation_type']._options = None
  _PROKARYOTEGENEREQUEST.fields_by_name['include_annotation_type']._serialized_options = b'\332\360\031\002\010\002\302\360\031\231\001\022\036\n\nFASTA_GENE\022\020Gene range FASTA\022?\n\tFASTA_RNA\n\rFASTA_PROTEIN\022#Transcript & protein sequence FASTA\0226\n\nFASTA_GENE\n\tFASTA_RNA\n\rFASTA_PROTEIN\022\016All annotation'
  _PROKARYOTEGENEREQUEST._options = None
  _PROKARYOTEGENEREQUEST._serialized_options = b'\302\360\031/\n-accessions:\n- WP_000000001.1\n- WP_000000002.1'
  _TAXONOMYMETADATAREQUEST.fields_by_name['taxons']._options = None
  _TAXONOMYMETADATAREQUEST.fields_by_name['taxons']._serialized_options = b'\302\360\031u\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022\033\n\005human\n\005mouse\022\013Common Name\022-\n\014Homo sapiens\n\014Mus musculus\022\017Scientific Name'
  _TAXONOMYMETADATAREQUEST.fields_by_name['page_size']._options = None
  _TAXONOMYMETADATAREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _TAXONOMYMETADATAREQUEST._options = None
  _TAXONOMYMETADATAREQUEST._serialized_options = b'\302\360\031\030\n\026taxons:\n- 9606\n- mouse'
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['tax_id']._options = None
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['tax_id']._serialized_options = b'\302\360\031\"\022 \n\0049606\022\030NCBI Taxonomy Identifier'
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['include_lineage']._options = None
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['include_lineage']._serialized_options = b'\302\360\031\007\n\005false'
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['include_subtree']._options = None
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['include_subtree']._serialized_options = b'\302\360\031\007\n\005false'
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['page_size']._options = None
  _TAXONOMYRELATEDIDREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _TAXONOMYRELATEDIDREQUEST._options = None
  _TAXONOMYRELATEDIDREQUEST._serialized_options = b'\302\360\031\016\n\014tax_id: 9606'
  _TAXONOMYDATASETREQUEST.fields_by_name['tax_ids']._options = None
  _TAXONOMYDATASETREQUEST.fields_by_name['tax_ids']._serialized_options = b'\302\360\031)\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier'
  _TAXONOMYDATASETREQUEST.fields_by_name['aux_reports']._options = None
  _TAXONOMYDATASETREQUEST.fields_by_name['aux_reports']._serialized_options = b'\302\360\031\213\001\022\210\001\n\014NAMES_REPORT\n\020TAXONOMY_SUMMARY\022fSelect additional report types to include with download. The base data report will always be included.'
  _TAXONOMYDATASETREQUEST._options = None
  _TAXONOMYDATASETREQUEST._serialized_options = b'\302\360\031\031\n\027tax_ids:\n- 9606\n- 10090'
  _TAXONOMYIMAGEREQUEST.fields_by_name['taxon']._options = None
  _TAXONOMYIMAGEREQUEST.fields_by_name['taxon']._serialized_options = b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name'
  _TAXONOMYIMAGEREQUEST._options = None
  _TAXONOMYIMAGEREQUEST._serialized_options = b'\302\360\031\017\n\rtaxon:\n- 9606'
  _TAXONOMYIMAGEMETADATAREQUEST.fields_by_name['taxon']._options = None
  _TAXONOMYIMAGEMETADATAREQUEST.fields_by_name['taxon']._serialized_options = b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name'
  _TAXONOMYIMAGEMETADATAREQUEST._options = None
  _TAXONOMYIMAGEMETADATAREQUEST._serialized_options = b'\302\360\031\017\n\rtaxon:\n- 9606'
  _TAXONOMYFILTEREDSUBTREEREQUEST.fields_by_name['taxons']._options = None
  _TAXONOMYFILTEREDSUBTREEREQUEST.fields_by_name['taxons']._serialized_options = b'\302\360\031u\022\'\n\0049606\n\00510090\022\030NCBI Taxonomy Identifier\022\033\n\005human\n\005mouse\022\013Common Name\022-\n\014Homo sapiens\n\014Mus musculus\022\017Scientific Name'
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY._options = None
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY._serialized_options = b'8\001'
  _TAXONOMYLINKSREQUEST.fields_by_name['taxon']._options = None
  _TAXONOMYLINKSREQUEST.fields_by_name['taxon']._serialized_options = b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name'
  _TAXONOMYLINKSREQUEST._options = None
  _TAXONOMYLINKSREQUEST._serialized_options = b'\302\360\031\017\n\rtaxon:\n- 9606'
  _GENECOUNTSBYTAXONREQUEST.fields_by_name['taxon']._options = None
  _GENECOUNTSBYTAXONREQUEST.fields_by_name['taxon']._serialized_options = b'\302\360\031Y\022 \n\0049606\022\030NCBI Taxonomy Identifier\022\024\n\005human\022\013Common Name\022\037\n\014Homo sapiens\022\017Scientific Name'
  _GENOMEANNOTATIONREQUEST.fields_by_name['page_size']._options = None
  _GENOMEANNOTATIONREQUEST.fields_by_name['page_size']._serialized_options = b'\302\360\031\004\n\00220'
  _INCLUDETABULARHEADER._serialized_start=30687
  _INCLUDETABULARHEADER._serialized_end=30822
  _SORTDIRECTION._serialized_start=30824
  _SORTDIRECTION._serialized_end=30932
  _FASTA._serialized_start=30935
  _FASTA._serialized_end=31088
  _GENETYPE._serialized_start=31091
  _GENETYPE._serialized_end=31275
  _VIRUSTABLEFIELD._serialized_start=31278
  _VIRUSTABLEFIELD._serialized_end=31695
  _TABLEFORMAT._serialized_start=31697
  _TABLEFORMAT._serialized_end=31739
  _VIRALSEQUENCETYPE._serialized_start=31741
  _VIRALSEQUENCETYPE._serialized_end=31819
  _VIRUSDATASETREPORTTYPE._serialized_start=31821
  _VIRUSDATASETREPORTTYPE._serialized_end=31903
  _IMAGESIZE._serialized_start=31905
  _IMAGESIZE._serialized_end=31956
  _METHODPAYLOADREQUEST._serialized_start=304
  _METHODPAYLOADREQUEST._serialized_end=376
  _ASSEMBLYSEQUENCEREPORTSREQUEST._serialized_start=379
  _ASSEMBLYSEQUENCEREPORTSREQUEST._serialized_end=1197
  _SEQUENCEREPORTPAGE._serialized_start=1200
  _SEQUENCEREPORTPAGE._serialized_end=1505
  _SORTFIELD._serialized_start=1508
  _SORTFIELD._serialized_end=2320
  _BIOSAMPLEDATASETREPORTSREQUEST._serialized_start=2322
  _BIOSAMPLEDATASETREPORTSREQUEST._serialized_end=2432
  _ASSEMBLYDATASETREPORTSREQUEST._serialized_start=2435
  _ASSEMBLYDATASETREPORTSREQUEST._serialized_end=3572
  _ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE._serialized_start=3515
  _ASSEMBLYDATASETREPORTSREQUEST_CONTENTTYPE._serialized_end=3572
  _ASSEMBLYDATASETDESCRIPTORSFILTER._serialized_start=3575
  _ASSEMBLYDATASETDESCRIPTORSFILTER._serialized_end=5772
  _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE._serialized_start=5552
  _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYSOURCE._serialized_end=5602
  _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION._serialized_start=5604
  _ASSEMBLYDATASETDESCRIPTORSFILTER_ASSEMBLYVERSION._serialized_end=5654
  _ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER._serialized_start=5656
  _ASSEMBLYDATASETDESCRIPTORSFILTER_METAGENOMEDERIVEDFILTER._serialized_end=5772
  _ACCESSIONS._serialized_start=5775
  _ACCESSIONS._serialized_end=5907
  _ASSEMBLYREVISIONHISTORY._serialized_start=5910
  _ASSEMBLYREVISIONHISTORY._serialized_end=6059
  _ASSEMBLYREVISIONHISTORYREQUEST._serialized_start=6061
  _ASSEMBLYREVISIONHISTORYREQUEST._serialized_end=6156
  _ASSEMBLYCHECKMHISTOGRAMREQUEST._serialized_start=6158
  _ASSEMBLYCHECKMHISTOGRAMREQUEST._serialized_end=6266
  _ASSEMBLYCHECKMHISTOGRAMREPLY._serialized_start=6269
  _ASSEMBLYCHECKMHISTOGRAMREPLY._serialized_end=6589
  _ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL._serialized_start=6492
  _ASSEMBLYCHECKMHISTOGRAMREPLY_HISTOGRAMINTERVAL._serialized_end=6589
  _BIOPROJECT._serialized_start=6592
  _BIOPROJECT._serialized_end=6744
  _BIOPROJECTLINEAGE._serialized_start=6746
  _BIOPROJECTLINEAGE._serialized_end=6829
  _BUSCOSTAT._serialized_start=6832
  _BUSCOSTAT._serialized_end=7093
  _ANNOTATIONFORASSEMBLY._serialized_start=7096
  _ANNOTATIONFORASSEMBLY._serialized_end=7756
  _ANNOTATIONFORASSEMBLY_FILE._serialized_start=7442
  _ANNOTATIONFORASSEMBLY_FILE._serialized_end=7553
  _ANNOTATIONFORASSEMBLY_TYPE._serialized_start=7556
  _ANNOTATIONFORASSEMBLY_TYPE._serialized_end=7756
  _FEATURECOUNTS._serialized_start=7758
  _FEATURECOUNTS._serialized_end=7836
  _GENECOUNTS._serialized_start=7838
  _GENECOUNTS._serialized_end=7911
  _ASSEMBLYDATASETREQUEST._serialized_start=7914
  _ASSEMBLYDATASETREQUEST._serialized_end=8693
  _ASSEMBLYDATASETREQUEST_RESOLUTION._serialized_start=8601
  _ASSEMBLYDATASETREQUEST_RESOLUTION._serialized_end=8655
  _ASSEMBLYDATASETAVAILABILITY._serialized_start=8696
  _ASSEMBLYDATASETAVAILABILITY._serialized_end=8839
  _ORGANISM._serialized_start=8842
  _ORGANISM._serialized_end=9886
  _ORGANISM_COUNTS._serialized_start=9700
  _ORGANISM_COUNTS._serialized_end=9754
  _ORGANISM_COUNTBYTYPE._serialized_start=9757
  _ORGANISM_COUNTBYTYPE._serialized_end=9886
  _ORGANISMQUERYREQUEST._serialized_start=9889
  _ORGANISMQUERYREQUEST._serialized_end=10484
  _ORGANISMQUERYREQUEST_TAXRANKFILTER._serialized_start=10312
  _ORGANISMQUERYREQUEST_TAXRANKFILTER._serialized_end=10358
  _ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER._serialized_start=10360
  _ORGANISMQUERYREQUEST_TAXONRESOURCEFILTER._serialized_end=10478
  _SCINAMEANDIDS._serialized_start=10487
  _SCINAMEANDIDS._serialized_end=10811
  _SCINAMEANDIDS_SCINAMEANDID._serialized_start=10592
  _SCINAMEANDIDS_SCINAMEANDID._serialized_end=10811
  _ORTHOLOGREQUEST._serialized_start=10814
  _ORTHOLOGREQUEST._serialized_end=11254
  _ORTHOLOGREQUEST_CONTENTTYPE._serialized_start=11213
  _ORTHOLOGREQUEST_CONTENTTYPE._serialized_end=11254
  _GENEDATASETREQUEST._serialized_start=11257
  _GENEDATASETREQUEST._serialized_end=12183
  _GENEDATASETREQUEST_CONTENTTYPE._serialized_start=11213
  _GENEDATASETREQUEST_CONTENTTYPE._serialized_end=11254
  _GENEDATASETREQUEST_GENEDATASETREPORTTYPE._serialized_start=12087
  _GENEDATASETREQUEST_GENEDATASETREPORTTYPE._serialized_end=12150
  _GENEDATASETREPORTSREQUEST._serialized_start=12186
  _GENEDATASETREPORTSREQUEST._serialized_end=13436
  _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON._serialized_start=13150
  _GENEDATASETREPORTSREQUEST_SYMBOLSFORTAXON._serialized_end=13330
  _GENEDATASETREPORTSREQUEST_CONTENTTYPE._serialized_start=13332
  _GENEDATASETREPORTSREQUEST_CONTENTTYPE._serialized_end=13403
  _GENELINKSREQUEST._serialized_start=13439
  _GENELINKSREQUEST._serialized_end=13583
  _GENELINKSREPLY._serialized_start=13586
  _GENELINKSREPLY._serialized_end=13946
  _GENELINKSREPLY_GENELINK._serialized_start=13679
  _GENELINKSREPLY_GENELINK._serialized_end=13869
  _GENELINKSREPLY_GENELINKTYPE._serialized_start=13871
  _GENELINKSREPLY_GENELINKTYPE._serialized_end=13946
  _GENECHROMOSOMESUMMARYREQUEST._serialized_start=13949
  _GENECHROMOSOMESUMMARYREQUEST._serialized_end=14204
  _GENECHROMOSOMESUMMARYREPLY._serialized_start=14207
  _GENECHROMOSOMESUMMARYREPLY._serialized_end=14460
  _GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY._serialized_start=14365
  _GENECHROMOSOMESUMMARYREPLY_GENECHROMOSOMESUMMARY._serialized_end=14460
  _ASSEMBLYLINKSREQUEST._serialized_start=14463
  _ASSEMBLYLINKSREQUEST._serialized_end=14659
  _ASSEMBLYLINKSREPLY._serialized_start=14662
  _ASSEMBLYLINKSREPLY._serialized_end=15102
  _ASSEMBLYLINKSREPLY_ASSEMBLYLINK._serialized_start=14775
  _ASSEMBLYLINKSREPLY_ASSEMBLYLINK._serialized_end=15004
  _ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE._serialized_start=15006
  _ASSEMBLYLINKSREPLY_ASSEMBLYLINKTYPE._serialized_end=15102
  _VIRUSDATASETREQUEST._serialized_start=15105
  _VIRUSDATASETREQUEST._serialized_end=16658
  _VIRUSDATASETFILTER._serialized_start=16661
  _VIRUSDATASETFILTER._serialized_end=17652
  _VIRUSDATAREPORTREQUEST._serialized_start=17655
  _VIRUSDATAREPORTREQUEST._serialized_end=18140
  _VIRUSDATAREPORTREQUEST_CONTENTTYPE._serialized_start=18052
  _VIRUSDATAREPORTREQUEST_CONTENTTYPE._serialized_end=18100
  _VIRUSANNOTATIONFILTER._serialized_start=18143
  _VIRUSANNOTATIONFILTER._serialized_end=19027
  _VIRUSANNOTATIONREPORTREQUEST._serialized_start=19030
  _VIRUSANNOTATIONREPORTREQUEST._serialized_end=19370
  _VIRUSAVAILABILITYREQUEST._serialized_start=19373
  _VIRUSAVAILABILITYREQUEST._serialized_end=19574
  _VIRUSAVAILABILITY._serialized_start=19577
  _VIRUSAVAILABILITY._serialized_end=19712
  _SARS2PROTEINDATASETREQUEST._serialized_start=19715
  _SARS2PROTEINDATASETREQUEST._serialized_end=21120
  _DOWNLOADSUMMARY._serialized_start=21123
  _DOWNLOADSUMMARY._serialized_end=22835
  _DOWNLOADSUMMARY_HYDRATED._serialized_start=21662
  _DOWNLOADSUMMARY_HYDRATED._serialized_end=21802
  _DOWNLOADSUMMARY_DEHYDRATED._serialized_start=21805
  _DOWNLOADSUMMARY_DEHYDRATED._serialized_end=22008
  _DOWNLOADSUMMARY_FILESUMMARY._serialized_start=22010
  _DOWNLOADSUMMARY_FILESUMMARY._serialized_end=22079
  _DOWNLOADSUMMARY_AVAILABLEFILES._serialized_start=22082
  _DOWNLOADSUMMARY_AVAILABLEFILES._serialized_end=22835
  _TABULAROUTPUT._serialized_start=22837
  _TABULAROUTPUT._serialized_end=22872
  _MICROBIGGEDATASETREQUEST._serialized_start=22875
  _MICROBIGGEDATASETREQUEST._serialized_end=23318
  _MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG._serialized_start=23145
  _MICROBIGGEDATASETREQUEST_ELEMENTFLANKCONFIG._serialized_end=23189
  _MICROBIGGEDATASETREQUEST_FILETYPE._serialized_start=23191
  _MICROBIGGEDATASETREQUEST_FILETYPE._serialized_end=23309
  _VERSIONREPLY._serialized_start=23320
  _VERSIONREPLY._serialized_end=23447
  _PROKARYOTEGENEREQUEST._serialized_start=23450
  _PROKARYOTEGENEREQUEST._serialized_end=24091
  _PROKARYOTEGENEREQUEST_GENEFLANKCONFIG._serialized_start=23997
  _PROKARYOTEGENEREQUEST_GENEFLANKCONFIG._serialized_end=24038
  _SLEEPREQUEST._serialized_start=24093
  _SLEEPREQUEST._serialized_end=24169
  _SLEEPREPLY._serialized_start=24171
  _SLEEPREPLY._serialized_end=24205
  _TAXONOMYNODE._serialized_start=24208
  _TAXONOMYNODE._serialized_end=25048
  _TAXONOMYNODE_COUNTBYTYPE._serialized_start=24956
  _TAXONOMYNODE_COUNTBYTYPE._serialized_end=25048
  _TAXONOMYMETADATAREQUEST._serialized_start=25051
  _TAXONOMYMETADATAREQUEST._serialized_end=25689
  _TAXONOMYMETADATAREQUEST_CONTENTTYPE._serialized_start=25578
  _TAXONOMYMETADATAREQUEST_CONTENTTYPE._serialized_end=25631
  _TAXONOMYMETADATAREQUEST_TABLEFORMAT._serialized_start=25633
  _TAXONOMYMETADATAREQUEST_TABLEFORMAT._serialized_end=25659
  _TAXONOMYRELATEDIDREQUEST._serialized_start=25692
  _TAXONOMYRELATEDIDREQUEST._serialized_end=25979
  _TAXONOMYTAXIDSPAGE._serialized_start=25981
  _TAXONOMYTAXIDSPAGE._serialized_end=26066
  _TAXONOMYDATASETREQUEST._serialized_start=26069
  _TAXONOMYDATASETREQUEST._serialized_end=26500
  _TAXONOMYDATASETREQUEST_TAXONOMYREPORTTYPE._serialized_start=26409
  _TAXONOMYDATASETREQUEST_TAXONOMYREPORTTYPE._serialized_end=26469
  _TAXONOMYMATCH._serialized_start=26503
  _TAXONOMYMATCH._serialized_end=26720
  _TAXONOMYMETADATARESPONSE._serialized_start=26723
  _TAXONOMYMETADATARESPONSE._serialized_end=26884
  _TAXONOMYIMAGEREQUEST._serialized_start=26887
  _TAXONOMYIMAGEREQUEST._serialized_end=27107
  _TAXONOMYIMAGEMETADATAREQUEST._serialized_start=27110
  _TAXONOMYIMAGEMETADATAREQUEST._serialized_end=27278
  _TAXONOMYIMAGEMETADATARESPONSE._serialized_start=27281
  _TAXONOMYIMAGEMETADATARESPONSE._serialized_end=27523
  _TAXONOMYFILTEREDSUBTREEREQUEST._serialized_start=27526
  _TAXONOMYFILTEREDSUBTREEREQUEST._serialized_end=27816
  _TAXONOMYFILTEREDSUBTREERESPONSE._serialized_start=27819
  _TAXONOMYFILTEREDSUBTREERESPONSE._serialized_end=28442
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE._serialized_start=28090
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE._serialized_end=28328
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS._serialized_start=28250
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGE_CHILDSTATUS._serialized_end=28328
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY._serialized_start=28330
  _TAXONOMYFILTEREDSUBTREERESPONSE_EDGESENTRY._serialized_end=28442
  _DATASETREQUEST._serialized_start=28445
  _DATASETREQUEST._serialized_end=28678
  _TAXONOMYLINKSREQUEST._serialized_start=28681
  _TAXONOMYLINKSREQUEST._serialized_end=28841
  _TAXONOMYLINKSRESPONSE._serialized_start=28844
  _TAXONOMYLINKSRESPONSE._serialized_end=29284
  _TAXONOMYLINKSRESPONSE_GENERICLINK._serialized_start=29215
  _TAXONOMYLINKSRESPONSE_GENERICLINK._serialized_end=29284
  _GENECOUNTSBYTAXONREQUEST._serialized_start=29287
  _GENECOUNTSBYTAXONREQUEST._serialized_end=29430
  _GENECOUNTSBYTAXONREPLY._serialized_start=29433
  _GENECOUNTSBYTAXONREPLY._serialized_end=29611
  _GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT._serialized_start=29542
  _GENECOUNTSBYTAXONREPLY_GENETYPEANDCOUNT._serialized_end=29611
  _GENOMEANNOTATIONREQUEST._serialized_start=29614
  _GENOMEANNOTATIONREQUEST._serialized_end=30447
  _GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE._serialized_start=30298
  _GENOMEANNOTATIONREQUEST_ANNOTATIONTYPE._serialized_end=30376
  _GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT._serialized_start=30378
  _GENOMEANNOTATIONREQUEST_GENOMEANNOTATIONTABLEFORMAT._serialized_end=30447
  _GENOMEANNOTATIONTABLESUMMARYREPLY._serialized_start=30450
  _GENOMEANNOTATIONTABLESUMMARYREPLY._serialized_end=30617
  _HTTPBODY._serialized_start=30619
  _HTTPBODY._serialized_end=30684
# @@protoc_insertion_point(module_scope)
