# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/options/report.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from google.protobuf import descriptor_pb2 as google_dot_protobuf_dot_descriptor__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\"ncbi/datasets/options/report.proto\x12\x15ncbi.datasets.options\x1a google/protobuf/descriptor.proto\"\xbf\x01\n\rDocumentation\x12\x1f\n\x0breport_name\x18\x01 \x01(\tR\nreportName\x12\'\n\x0fprimary_message\x18\x02 \x01(\tR\x0eprimaryMessage\x12\x18\n\x07\x63ontent\x18\x03 \x03(\tR\x07\x63ontent\x12\'\n\x0f\x65lided_messages\x18\x04 \x03(\tR\x0e\x65lidedMessages\x12!\n\x0c\x65xample_json\x18\x05 \x01(\tR\x0b\x65xampleJson\"\xcb\x01\n\x0eTabularOptions\x12\x1a\n\x08mnemonic\x18\x01 \x01(\tR\x08mnemonic\x12\x1f\n\x0b\x63olumn_name\x18\x02 \x01(\tR\ncolumnName\x12 \n\x0b\x64\x65scription\x18\x03 \x01(\tR\x0b\x64\x65scription\x12\x1f\n\x0bgolang_type\x18\x04 \x01(\tR\ngolangType\x12\x1d\n\nexcel_type\x18\x05 \x01(\tR\texcelType\x12\x1a\n\x08\x65xamples\x18\x06 \x03(\tR\x08\x65xamples\"\x9a\x01\n\x0c\x46\x61staOptions\x12\x45\n\nfield_name\x18\x01 \x01(\x0e\x32&.ncbi.datasets.options.FastaFieldNamesR\tfieldName\x12 \n\x0b\x64\x65scription\x18\x02 \x01(\tR\x0b\x64\x65scription\x12!\n\x0cleader_field\x18\x03 \x01(\x08R\x0bleaderField\"N\n\x0bSolrOptions\x12\x1d\n\nfield_name\x18\x01 \x01(\tR\tfieldName\x12 \n\x0b\x64\x65scription\x18\x02 \x01(\tR\x0b\x64\x65scription*\xb1\x01\n\x0f\x46\x61staFieldNames\x12\x0c\n\x08ORGANISM\x10\x00\x12\x11\n\rELEMENT_RANGE\x10\x01\x12\x10\n\x0c\x45LEMENT_NAME\x10\x02\x12\x12\n\x0e\x45LEMENT_SYMBOL\x10\x03\x12\n\n\x06\x43ONTIG\x10\x04\x12\x08\n\x04GENE\x10\x05\x12\x15\n\x11PROTEIN_ACCESSION\x10\x06\x12\x0e\n\nCHROMOSOME\x10\x07\x12\x08\n\x04NAME\x10\x08\x12\x10\n\x0c\x43OMPLETENESS\x10\t:e\n\x0breport_docs\x12\x1c.google.protobuf.FileOptions\x18\xb8\x8e\x03 \x01(\x0b\x32$.ncbi.datasets.options.DocumentationR\nreportDocs:`\n\x07tabular\x12\x1d.google.protobuf.FieldOptions\x18\xb8\x8e\x03 \x01(\x0b\x32%.ncbi.datasets.options.TabularOptionsR\x07tabular:Z\n\x05\x66\x61sta\x12\x1d.google.protobuf.FieldOptions\x18\xb9\x8e\x03 \x01(\x0b\x32#.ncbi.datasets.options.FastaOptionsR\x05\x66\x61sta:W\n\x04solr\x12\x1d.google.protobuf.FieldOptions\x18\xba\x8e\x03 \x01(\x0b\x32\".ncbi.datasets.options.SolrOptionsR\x04solrB\x1aZ\x15ncbi/datasets/options\xf8\x01\x01\x62\x06proto3')

_FASTAFIELDNAMES = DESCRIPTOR.enum_types_by_name['FastaFieldNames']
FastaFieldNames = enum_type_wrapper.EnumTypeWrapper(_FASTAFIELDNAMES)
ORGANISM = 0
ELEMENT_RANGE = 1
ELEMENT_NAME = 2
ELEMENT_SYMBOL = 3
CONTIG = 4
GENE = 5
PROTEIN_ACCESSION = 6
CHROMOSOME = 7
NAME = 8
COMPLETENESS = 9

REPORT_DOCS_FIELD_NUMBER = 51000
report_docs = DESCRIPTOR.extensions_by_name['report_docs']
TABULAR_FIELD_NUMBER = 51000
tabular = DESCRIPTOR.extensions_by_name['tabular']
FASTA_FIELD_NUMBER = 51001
fasta = DESCRIPTOR.extensions_by_name['fasta']
SOLR_FIELD_NUMBER = 51002
solr = DESCRIPTOR.extensions_by_name['solr']

_DOCUMENTATION = DESCRIPTOR.message_types_by_name['Documentation']
_TABULAROPTIONS = DESCRIPTOR.message_types_by_name['TabularOptions']
_FASTAOPTIONS = DESCRIPTOR.message_types_by_name['FastaOptions']
_SOLROPTIONS = DESCRIPTOR.message_types_by_name['SolrOptions']
Documentation = _reflection.GeneratedProtocolMessageType('Documentation', (_message.Message,), {
  'DESCRIPTOR' : _DOCUMENTATION,
  '__module__' : 'ncbi.datasets.options.report_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.options.Documentation)
  })
_sym_db.RegisterMessage(Documentation)

TabularOptions = _reflection.GeneratedProtocolMessageType('TabularOptions', (_message.Message,), {
  'DESCRIPTOR' : _TABULAROPTIONS,
  '__module__' : 'ncbi.datasets.options.report_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.options.TabularOptions)
  })
_sym_db.RegisterMessage(TabularOptions)

FastaOptions = _reflection.GeneratedProtocolMessageType('FastaOptions', (_message.Message,), {
  'DESCRIPTOR' : _FASTAOPTIONS,
  '__module__' : 'ncbi.datasets.options.report_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.options.FastaOptions)
  })
_sym_db.RegisterMessage(FastaOptions)

SolrOptions = _reflection.GeneratedProtocolMessageType('SolrOptions', (_message.Message,), {
  'DESCRIPTOR' : _SOLROPTIONS,
  '__module__' : 'ncbi.datasets.options.report_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.options.SolrOptions)
  })
_sym_db.RegisterMessage(SolrOptions)

if _descriptor._USE_C_DESCRIPTORS == False:
  google_dot_protobuf_dot_descriptor__pb2.FileOptions.RegisterExtension(report_docs)
  google_dot_protobuf_dot_descriptor__pb2.FieldOptions.RegisterExtension(tabular)
  google_dot_protobuf_dot_descriptor__pb2.FieldOptions.RegisterExtension(fasta)
  google_dot_protobuf_dot_descriptor__pb2.FieldOptions.RegisterExtension(solr)

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'Z\025ncbi/datasets/options\370\001\001'
  _FASTAFIELDNAMES._serialized_start=733
  _FASTAFIELDNAMES._serialized_end=910
  _DOCUMENTATION._serialized_start=96
  _DOCUMENTATION._serialized_end=287
  _TABULAROPTIONS._serialized_start=290
  _TABULAROPTIONS._serialized_end=493
  _FASTAOPTIONS._serialized_start=496
  _FASTAOPTIONS._serialized_end=650
  _SOLROPTIONS._serialized_start=652
  _SOLROPTIONS._serialized_end=730
# @@protoc_insertion_point(module_scope)
