# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v2/reports/assembly.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from ncbi.datasets.v2.reports import biosample_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_biosample__pb2
from ncbi.datasets.v2.reports import common_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2
from ncbi.datasets.options import report_pb2 as ncbi_dot_datasets_dot_options_dot_report__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\'ncbi/datasets/v2/reports/assembly.proto\x12\x18ncbi.datasets.v2.reports\x1a(ncbi/datasets/v2/reports/biosample.proto\x1a%ncbi/datasets/v2/reports/common.proto\x1a\"ncbi/datasets/options/report.proto\"\xc4\x03\n\x16\x41ssemblyDataReportPage\x12\x46\n\x07reports\x18\x01 \x03(\x0b\x32,.ncbi.datasets.v2.reports.AssemblyDataReportR\x07reports\x12H\n\x0c\x63ontent_type\x18\x02 \x01(\x0e\x32%.ncbi.datasets.v2.reports.ContentTypeR\x0b\x63ontentType\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\x12&\n\x0fnext_page_token\x18\x04 \x01(\tR\rnextPageToken\x12=\n\x08messages\x18\x06 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12!\n\x0c_report_type\x18\xf4\x03 \x01(\tR\nReportType\x12%\n\x0e_report_fields\x18\xf5\x03 \x03(\tR\x0cReportFields\x12\x1f\n\x0b_first_page\x18\xf6\x03 \x01(\x08R\tFirstPage\x12%\n\x0e_report_format\x18\xf7\x03 \x01(\tR\x0cReportFormat\"\xce\n\n\x12\x41ssemblyDataReport\x12S\n\taccession\x18\x1f \x01(\tB5\xc2\xf3\x18\x31\n\taccession\x12\x12\x41ssembly Accession2\x10GCF_000001405.40R\taccession\x12i\n\x11\x63urrent_accession\x18\x0f \x01(\tB<\xc2\xf3\x18\x38\n\x11\x63urrent-accession\x12\x11\x43urrent Accession2\x10GCF_000001405.40R\x10\x63urrentAccession\x12-\n\x10paired_accession\x18  \x01(\tB\x02\x18\x01R\x0fpairedAccession\x12\x8a\x01\n\x0fsource_database\x18! \x01(\x0e\x32(.ncbi.datasets.v2.reports.SourceDatabaseB7\xc2\xf3\x18\x33\n\x0fsource_database\x12\x0fSource Database2\x06REFSEQ2\x07GENBANKR\x0esourceDatabase\x12Z\n\x08organism\x18\" \x01(\x0b\x32\".ncbi.datasets.v2.reports.OrganismB\x1a\xc2\xf3\x18\x16\n\torganism-\x12\tOrganism R\x08organism\x12g\n\rassembly_info\x18\n \x01(\x0b\x32&.ncbi.datasets.v2.reports.AssemblyInfoB\x1a\xc2\xf3\x18\x16\n\tassminfo-\x12\tAssembly R\x0c\x61ssemblyInfo\x12q\n\x0e\x61ssembly_stats\x18\x0b \x01(\x0b\x32\'.ncbi.datasets.v2.reports.AssemblyStatsB!\xc2\xf3\x18\x1d\n\nassmstats-\x12\x0f\x41ssembly Stats R\rassemblyStats\x12l\n\x0eorganelle_info\x18\x0c \x03(\x0b\x32\'.ncbi.datasets.v2.reports.OrganelleInfoB\x1c\xc2\xf3\x18\x18\n\norganelle-\x12\nOrganelle R\rorganelleInfo\x12p\n\x0f\x61nnotation_info\x18\r \x01(\x0b\x32(.ncbi.datasets.v2.reports.AnnotationInfoB\x1d\xc2\xf3\x18\x19\n\nannotinfo-\x12\x0b\x41nnotation R\x0e\x61nnotationInfo\x12N\n\x08wgs_info\x18\x0e \x01(\x0b\x32!.ncbi.datasets.v2.reports.WGSInfoB\x10\xc2\xf3\x18\x0c\n\x04wgs-\x12\x04WGS R\x07wgsInfo\x12q\n\rtype_material\x18\x10 \x01(\x0b\x32&.ncbi.datasets.v2.reports.TypeMaterialB$\xc2\xf3\x18 \n\x0etype_material-\x12\x0eType Material R\x0ctypeMaterial\x12Y\n\x0b\x63heckm_info\x18\x11 \x01(\x0b\x32 .ncbi.datasets.v2.reports.CheckMB\x16\xc2\xf3\x18\x12\n\x07\x63heckm-\x12\x07\x43heckM R\ncheckmInfo\x12\x85\x01\n\x1b\x61verage_nucleotide_identity\x18\x12 \x01(\x0b\x32\x33.ncbi.datasets.v2.reports.AverageNucleotideIdentityB\x10\xc2\xf3\x18\x0c\n\x04\x61ni-\x12\x04\x41NI R\x19\x61verageNucleotideIdentity\"\x8f\x01\n\x0cTypeMaterial\x12\x31\n\ntype_label\x18\x01 \x01(\tB\x12\xc2\xf3\x18\x0e\n\x05label\x12\x05LabelR\ttypeLabel\x12L\n\x11type_display_text\x18\x02 \x01(\tB \xc2\xf3\x18\x1c\n\x0c\x64isplay_text\x12\x0c\x44isplay TextR\x0ftypeDisplayText\"\xcb\x01\n\x0eLinkedAssembly\x12T\n\x0flinked_assembly\x18\x01 \x01(\tB+\xc2\xf3\x18\'\n\taccession\x12\tAccession2\x0fGCA_000212995.1R\x0elinkedAssembly\x12\x63\n\rassembly_type\x18\x02 \x01(\x0e\x32,.ncbi.datasets.v2.reports.LinkedAssemblyTypeB\x10\xc2\xf3\x18\x0c\n\x04type\x12\x04TypeR\x0c\x61ssemblyType\"\xf1\x03\n\x0ePairedAssembly\x12J\n\taccession\x18\x01 \x01(\tB,\xc2\xf3\x18(\n\taccession\x12\tAccession2\x10GCF_000001405.40R\taccession\x12_\n\x06status\x18\x02 \x01(\x0e\x32(.ncbi.datasets.v2.reports.AssemblyStatusB\x1d\xc2\xf3\x18\x19\n\x06status\x12\x06Status2\x07\x63urrentR\x06status\x12\x39\n\x0f\x61nnotation_name\x18\x03 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x0e\x61nnotationName\x12\x43\n\x0conly_genbank\x18\x04 \x01(\tB \xc2\xf3\x18\x1c\n\x0conly-genbank\x12\x0cOnly GenbankR\x0bonlyGenbank\x12?\n\x0bonly_refseq\x18\x05 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x0bonly-refseq\x12\x0bOnly RefSeqR\nonlyRefseq\x12\x30\n\x07\x63hanged\x18\x06 \x01(\tB\x16\xc2\xf3\x18\x12\n\x07\x63hanged\x12\x07\x43hangedR\x07\x63hanged\x12?\n\x0bmanual_diff\x18\x07 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x0bmanual-diff\x12\x0bManual DiffR\nmanualDiff\"\x85\x01\n\x0c\x41typicalInfo\x12?\n\x0bis_atypical\x18\x01 \x01(\x08\x42\x1e\xc2\xf3\x18\x1a\n\x0bis-atypical\x12\x0bIs AtypicalR\nisAtypical\x12\x34\n\x08warnings\x18\x03 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08warnings\x12\x08WarningsR\x08warnings\"\xc1\x0f\n\x0c\x41ssemblyInfo\x12W\n\x0e\x61ssembly_level\x18\x01 \x01(\tB0\xc2\xf3\x18,\n\x05level\x12\x05Level2\nchromosome2\x08scaffold2\x06\x63ontigR\rassemblyLevel\x12p\n\x0f\x61ssembly_status\x18\x16 \x01(\x0e\x32(.ncbi.datasets.v2.reports.AssemblyStatusB\x1d\xc2\xf3\x18\x19\n\x06status\x12\x06Status2\x07\x63urrentR\x0e\x61ssemblyStatus\x12w\n\x0fpaired_assembly\x18\x17 \x01(\x0b\x32(.ncbi.datasets.v2.reports.PairedAssemblyB$\xc2\xf3\x18 \n\x0cpaired-assm-\x12\x10Paired Assembly R\x0epairedAssembly\x12N\n\rassembly_name\x18\x02 \x01(\tB)\xc2\xf3\x18%\n\x04name\x12\x04Name2\nGRCh38.p142\x0b\x41SM985889v3R\x0c\x61ssemblyName\x12U\n\rassembly_type\x18\x03 \x01(\tB0\xc2\xf3\x18,\n\x04type\x12\x04Type2\x15haploid-with-alt-loci2\x07haploidR\x0c\x61ssemblyType\x12z\n\x12\x62ioproject_lineage\x18\r \x03(\x0b\x32+.ncbi.datasets.v2.reports.BioProjectLineageB\x1e\xc2\xf3\x18\x1a\n\x0b\x62ioproject-\x12\x0b\x42ioProject R\x11\x62ioprojectLineage\x12Y\n\x14\x62ioproject_accession\x18\x1b \x01(\tB&\xc2\xf3\x18\"\n\nbioproject\x12\x14\x42ioProject AccessionR\x13\x62ioprojectAccession\x12+\n\x0fsubmission_date\x18\x05 \x01(\tB\x02\x18\x01R\x0esubmissionDate\x12\x43\n\x0crelease_date\x18\x1d \x01(\tB \xc2\xf3\x18\x1c\n\x0crelease-date\x12\x0cRelease DateR\x0breleaseDate\x12@\n\x0b\x64\x65scription\x18\x06 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x0b\x64\x65scription\x12\x0b\x44\x65scriptionR\x0b\x64\x65scription\x12\x38\n\tsubmitter\x18\x08 \x01(\tB\x1a\xc2\xf3\x18\x16\n\tsubmitter\x12\tSubmitterR\tsubmitter\x12x\n\x0frefseq_category\x18\t \x01(\tBO\xc2\xf3\x18K\n\x0frefseq-category\x12\x0fRefseq Category2\x10reference genome2\x15representative genomeR\x0erefseqCategory\x12\x36\n\x07synonym\x18\x0b \x01(\tB\x1c\xc2\xf3\x18\x18\n\x07synonym\x12\x07Synonym2\x04hg38R\x07synonym\x12+\n\x0flinked_assembly\x18\x0c \x01(\tB\x02\x18\x01R\x0elinkedAssembly\x12{\n\x11linked_assemblies\x18\x15 \x03(\x0b\x32(.ncbi.datasets.v2.reports.LinkedAssemblyB$\xc2\xf3\x18 \n\x0clinked-assm-\x12\x10Linked Assembly R\x10linkedAssemblies\x12]\n\x08\x61typical\x18\x18 \x01(\x0b\x32&.ncbi.datasets.v2.reports.AtypicalInfoB\x19\xc2\xf3\x18\x15\n\x08\x61typical\x12\tAtypical R\x08\x61typical\x12\x36\n\x0cgenome_notes\x18\x1a \x03(\tB\x13\xc2\xf3\x18\x0f\n\x05notes\x12\x06Notes R\x0bgenomeNotes\x12O\n\x0fsequencing_tech\x18\x0e \x01(\tB&\xc2\xf3\x18\"\n\x0fsequencing-tech\x12\x0fSequencing TechR\x0esequencingTech\x12O\n\x0f\x61ssembly_method\x18\x1c \x01(\tB&\xc2\xf3\x18\"\n\x0f\x61ssembly-method\x12\x0f\x41ssembly MethodR\x0e\x61ssemblyMethod\x12i\n\tbiosample\x18\x13 \x01(\x0b\x32-.ncbi.datasets.v2.reports.BioSampleDescriptorB\x1c\xc2\xf3\x18\x18\n\nbiosample-\x12\nBioSample R\tbiosample\x12\x37\n\tblast_url\x18\x12 \x01(\tB\x1a\xc2\xf3\x18\x16\n\tblast-url\x12\tBlast URLR\x08\x62lastUrl\x12\x1a\n\x08\x63omments\x18\x14 \x01(\tR\x08\x63omments\x12[\n\x12suppression_reason\x18\x1e \x01(\tB,\xc2\xf3\x18(\n\x12suppression-reason\x12\x12Suppression ReasonR\x11suppressionReason\x12O\n\x0c\x64iploid_role\x18\x19 \x01(\x0e\x32,.ncbi.datasets.v2.reports.LinkedAssemblyTypeR\x0b\x64iploidRole\"\xc9\n\n\rAssemblyStats\x12}\n\x1btotal_number_of_chromosomes\x18\x01 \x01(\rB>\xc2\xf3\x18:\n\x1btotal-number-of-chromosomes\x12\x1bTotal Number of ChromosomesR\x18totalNumberOfChromosomes\x12\x63\n\x15total_sequence_length\x18\x02 \x01(\x04\x42/\xc2\xf3\x18+\n\x12total-sequence-len\x12\x15Total Sequence LengthR\x13totalSequenceLength\x12\x63\n\x15total_ungapped_length\x18\x03 \x01(\x04\x42/\xc2\xf3\x18+\n\x12total-ungapped-len\x12\x15Total Ungapped LengthR\x13totalUngappedLength\x12V\n\x11number_of_contigs\x18\x04 \x01(\rB*\xc2\xf3\x18&\n\x11number-of-contigs\x12\x11Number of ContigsR\x0fnumberOfContigs\x12;\n\ncontig_n50\x18\x05 \x01(\rB\x1c\xc2\xf3\x18\x18\n\ncontig-n50\x12\nContig N50R\tcontigN50\x12;\n\ncontig_l50\x18\x06 \x01(\rB\x1c\xc2\xf3\x18\x18\n\ncontig-l50\x12\nContig L50R\tcontigL50\x12^\n\x13number_of_scaffolds\x18\x07 \x01(\rB.\xc2\xf3\x18*\n\x13number-of-scaffolds\x12\x13Number of ScaffoldsR\x11numberOfScaffolds\x12\x43\n\x0cscaffold_n50\x18\x08 \x01(\rB \xc2\xf3\x18\x1c\n\x0cscaffold-n50\x12\x0cScaffold N50R\x0bscaffoldN50\x12\x43\n\x0cscaffold_l50\x18\t \x01(\rB \xc2\xf3\x18\x1c\n\x0cscaffold-l50\x12\x0cScaffold L50R\x0bscaffoldL50\x12\x81\x01\n\x1cgaps_between_scaffolds_count\x18\n \x01(\rB@\xc2\xf3\x18<\n\x1cgaps-between-scaffolds-count\x12\x1cGaps Between Scaffolds CountR\x19gapsBetweenScaffoldsCount\x12\x85\x01\n\x1dnumber_of_component_sequences\x18\x0b \x01(\rBB\xc2\xf3\x18>\n\x1dnumber-of-component-sequences\x12\x1dNumber of Component SequencesR\x1anumberOfComponentSequences\x12\x35\n\x08gc_count\x18\x0c \x01(\x04\x42\x1a\x18\x01\xc2\xf3\x18\x14\n\x08gc-count\x12\x08GC CountR\x07gcCount\x12;\n\ngc_percent\x18\r \x01(\x02\x42\x1c\xc2\xf3\x18\x18\n\ngc-percent\x12\nGC PercentR\tgcPercent\x12O\n\x0fgenome_coverage\x18\x0e \x01(\tB&\xc2\xf3\x18\"\n\x0fgenome-coverage\x12\x0fGenome CoverageR\x0egenomeCoverage\x12\x62\n\x14number_of_organelles\x18\x0f \x01(\rB0\xc2\xf3\x18,\n\x14number-of-organelles\x12\x14Number of OrganellesR\x12numberOfOrganelles\"\xdb\x03\n\rOrganelleInfo\x12I\n\rassembly_name\x18\x01 \x01(\tB$\x18\x01\xc2\xf3\x18\x1e\n\rassembly-name\x12\rAssembly NameR\x0c\x61ssemblyName\x12[\n\x12infraspecific_name\x18\x02 \x01(\tB,\xc2\xf3\x18(\n\x12infraspecific-name\x12\x12Infraspecific NameR\x11infraspecificName\x12R\n\nbioproject\x18\x03 \x03(\tB2\xc2\xf3\x18.\n\x15\x62ioproject-accessions\x12\x15\x42ioProject AccessionsR\nbioproject\x12@\n\x0b\x64\x65scription\x18\x04 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x0b\x64\x65scription\x12\x0b\x44\x65scriptionR\x0b\x64\x65scription\x12R\n\x10total_seq_length\x18\x05 \x01(\x04\x42(\xc2\xf3\x18$\n\x10total-seq-length\x12\x10Total Seq LengthR\x0etotalSeqLength\x12\x38\n\tsubmitter\x18\x06 \x01(\tB\x1a\xc2\xf3\x18\x16\n\tsubmitter\x12\tSubmitterR\tsubmitter\"\xd0\x05\n\x0e\x41nnotationInfo\x12$\n\x04name\x18\x01 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\x12\x34\n\x08provider\x18\x02 \x01(\tB\x18\xc2\xf3\x18\x14\n\x08provider\x12\x08ProviderR\x08provider\x12\x43\n\x0crelease_date\x18\x03 \x01(\tB \xc2\xf3\x18\x1c\n\x0crelease-date\x12\x0cRelease DateR\x0breleaseDate\x12;\n\nreport_url\x18\x04 \x01(\tB\x1c\xc2\xf3\x18\x18\n\nreport-url\x12\nReport URLR\treportUrl\x12W\n\x05stats\x18\x05 \x01(\x0b\x32\'.ncbi.datasets.v2.reports.FeatureCountsB\x18\xc2\xf3\x18\x14\n\nfeatcount-\x12\x06\x43ount R\x05stats\x12O\n\x05\x62usco\x18\x06 \x01(\x0b\x32#.ncbi.datasets.v2.reports.BuscoStatB\x14\xc2\xf3\x18\x10\n\x06\x62usco-\x12\x06\x42USCO R\x05\x62usco\x12,\n\x06method\x18\x07 \x01(\tB\x14\xc2\xf3\x18\x10\n\x06method\x12\x06MethodR\x06method\x12\x34\n\x08pipeline\x18\x08 \x01(\tB\x18\xc2\xf3\x18\x14\n\x08pipeline\x12\x08PipelineR\x08pipeline\x12S\n\x10software_version\x18\t \x01(\tB(\xc2\xf3\x18$\n\x10software-version\x12\x10Software VersionR\x0fsoftwareVersion\x12,\n\x06status\x18\n \x01(\tB\x14\xc2\xf3\x18\x10\n\x06status\x12\x06StatusR\x06status\x12O\n\x0frelease_version\x18\x0b \x01(\tB&\xc2\xf3\x18\"\n\x0frelease-version\x12\x0fRelease VersionR\x0ereleaseVersion\"j\n\rFeatureCounts\x12Y\n\x0bgene_counts\x18\x01 \x01(\x0b\x32$.ncbi.datasets.v2.reports.GeneCountsB\x12\xc2\xf3\x18\x0e\n\x05gene-\x12\x05Gene R\ngeneCounts\"\xe5\x03\n\tBuscoStat\x12<\n\rbusco_lineage\x18\x01 \x01(\tB\x17\xc2\xf3\x18\x13\n\x07lineage\x12\x08Lineage R\x0c\x62uscoLineage\x12\x30\n\tbusco_ver\x18\x02 \x01(\tB\x13\xc2\xf3\x18\x0f\n\x03ver\x12\x08Version R\x08\x62uscoVer\x12\x35\n\x08\x63omplete\x18\x03 \x01(\x02\x42\x19\xc2\xf3\x18\x15\n\x08\x63omplete\x12\tComplete R\x08\x63omplete\x12?\n\x0bsingle_copy\x18\x04 \x01(\x02\x42\x1e\xc2\xf3\x18\x1a\n\nsinglecopy\x12\x0cSingle Copy R\nsingleCopy\x12=\n\nduplicated\x18\x05 \x01(\x02\x42\x1d\xc2\xf3\x18\x19\n\nduplicated\x12\x0b\x44uplicated R\nduplicated\x12=\n\nfragmented\x18\x06 \x01(\x02\x42\x1d\xc2\xf3\x18\x19\n\nfragmented\x12\x0b\x46ragmented R\nfragmented\x12\x31\n\x07missing\x18\x07 \x01(\x02\x42\x17\xc2\xf3\x18\x13\n\x07missing\x12\x08Missing R\x07missing\x12?\n\x0btotal_count\x18\x08 \x01(\x04\x42\x1e\xc2\xf3\x18\x1a\n\ntotalcount\x12\x0cTotal Count R\ntotalCount\"\xa8\x02\n\nGeneCounts\x12(\n\x05total\x18\x01 \x01(\rB\x12\xc2\xf3\x18\x0e\n\x05total\x12\x05TotalR\x05total\x12K\n\x0eprotein_coding\x18\x02 \x01(\rB$\xc2\xf3\x18 \n\x0eprotein-coding\x12\x0eProtein-codingR\rproteinCoding\x12;\n\nnon_coding\x18\x03 \x01(\rB\x1c\xc2\xf3\x18\x18\n\nnon-coding\x12\nNon-codingR\tnonCoding\x12<\n\npseudogene\x18\x04 \x01(\rB\x1c\xc2\xf3\x18\x18\n\npseudogene\x12\nPseudogeneR\npseudogene\x12(\n\x05other\x18\x05 \x01(\rB\x12\xc2\xf3\x18\x0e\n\x05other\x12\x05OtherR\x05other\"\xde\x02\n\x07WGSInfo\x12p\n\x15wgs_project_accession\x18\x01 \x01(\tB<\xc2\xf3\x18\x38\n\x11project-accession\x12\x11project accession2\x06\x41\x41\x45X032\x08\x43\x41\x42HLF01R\x13wgsProjectAccession\x12i\n\x0emaster_wgs_url\x18\x02 \x01(\tBC\xc2\xf3\x18?\n\x03url\x12\x03URL23https://www.ncbi.nlm.nih.gov/nuccore/AAEX00000000.3R\x0cmasterWgsUrl\x12v\n\x0fwgs_contigs_url\x18\x03 \x01(\tBN\xc2\xf3\x18J\n\x0b\x63ontigs-url\x12\x0b\x63ontigs URL2.https://www.ncbi.nlm.nih.gov/Traces/wgs/AAEX03R\rwgsContigsUrl\"\x8c\x05\n\x06\x43heckM\x12]\n\x11\x63heckm_marker_set\x18\x01 \x01(\tB1\xc2\xf3\x18-\n\nmarker-set\x12\nmarker set2\x13Mycobacterium aviumR\x0f\x63heckmMarkerSet\x12]\n\x15\x63heckm_species_tax_id\x18\x07 \x01(\rB*\xc2\xf3\x18&\n\x0especies-tax-id\x12\x0especies tax id2\x04\x31\x37\x36\x34R\x12\x63heckmSpeciesTaxId\x12k\n\x16\x63heckm_marker_set_rank\x18\x02 \x01(\tB6\xc2\xf3\x18\x32\n\x0fmarker-set-rank\x12\x0fmarker set rank2\x07species2\x05genusR\x13\x63heckmMarkerSetRank\x12\x45\n\x0e\x63heckm_version\x18\x03 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x07version\x12\x07version2\x06v1.2.0R\rcheckmVersion\x12K\n\x0c\x63ompleteness\x18\x04 \x01(\x02\x42\'\xc2\xf3\x18#\n\x0c\x63ompleteness\x12\x0c\x63ompleteness2\x05\x38\x36.83R\x0c\x63ompleteness\x12N\n\rcontamination\x18\x05 \x01(\x02\x42(\xc2\xf3\x18$\n\rcontamination\x12\rcontamination2\x04\x35.18R\rcontamination\x12s\n\x17\x63ompleteness_percentile\x18\x06 \x01(\x02\x42:\xc2\xf3\x18\x36\n\x17\x63ompleteness-percentile\x12\x17\x63ompleteness percentile2\x02\x37\x39R\x16\x63ompletenessPercentile\"\xb1\x0b\n\x19\x41verageNucleotideIdentity\x12\xb7\x01\n\x15taxonomy_check_status\x18\x01 \x01(\x0e\x32G.ncbi.datasets.v2.reports.AverageNucleotideIdentity.TaxonomyCheckStatusB:\xc2\xf3\x18\x36\n\x0c\x63heck-status\x12\x0c\x43heck status2\x02ok2\x06\x66\x61iled2\x0cinconclusiveR\x13taxonomyCheckStatus\x12\xa5\x01\n\x0cmatch_status\x18\x02 \x01(\x0e\x32?.ncbi.datasets.v2.reports.AverageNucleotideIdentity.MatchStatusBA\xc2\xf3\x18=\n\x11\x62\x65st-match-status\x12\x11\x42\x65st match status2\x15\x64\x65rived-species-matchR\x0bmatchStatus\x12\xa2\x01\n\x12submitted_organism\x18\x03 \x01(\tBs\xc2\xf3\x18o\n\x12submitted-organism\x12\x12Submitted organism2ESalmonella enterica subsp. enterica serovar Tennessee str. CDC07-0191R\x11submittedOrganism\x12l\n\x11submitted_species\x18\x04 \x01(\tB?\xc2\xf3\x18;\n\x11submitted-species\x12\x11Submitted species2\x13Salmonella entericaR\x10submittedSpecies\x12h\n\x08\x63\x61tegory\x18\x05 \x01(\x0e\x32).ncbi.datasets.v2.reports.ANITypeCategoryB!\xc2\xf3\x18\x1d\n\x08\x63\x61tegory\x12\x08\x43\x61tegory2\x07syntypeR\x08\x63\x61tegory\x12\x83\x01\n\x13submitted_ani_match\x18\x06 \x01(\x0b\x32\".ncbi.datasets.v2.reports.ANIMatchB/\xc2\xf3\x18+\n\x14submitted-ani-match-\x12\x13\x44\x65\x63lared ANI match R\x11submittedAniMatch\x12p\n\x0e\x62\x65st_ani_match\x18\x07 \x01(\x0b\x32\".ncbi.datasets.v2.reports.ANIMatchB&\xc2\xf3\x18\"\n\x0f\x62\x65st-ani-match-\x12\x0f\x42\x65st ANI match R\x0c\x62\x65stAniMatch\x12\x30\n\x07\x63omment\x18\x08 \x01(\tB\x16\xc2\xf3\x18\x12\n\x07\x63omment\x12\x07\x43ommentR\x07\x63omment\"^\n\x13TaxonomyCheckStatus\x12!\n\x1dTAXONOMY_CHECK_STATUS_UNKNOWN\x10\x00\x12\x06\n\x02OK\x10\x01\x12\n\n\x06\x46\x61iled\x10\x02\x12\x10\n\x0cInconclusive\x10\x03\"\xaa\x02\n\x0bMatchStatus\x12\x1d\n\x19\x42\x45ST_MATCH_STATUS_UNKNOWN\x10\x00\x12\x15\n\x11\x61pproved_mismatch\x10\x01\x12\x19\n\x15\x62\x65low_threshold_match\x10\x02\x12\x1c\n\x18\x62\x65low_threshold_mismatch\x10\x03\x12\x15\n\x11\x62\x65st_match_status\x10\x04\x12\x19\n\x15\x64\x65rived_species_match\x10\x05\x12\x0f\n\x0bgenus_match\x10\x06\x12\x10\n\x0clow_coverage\x10\x07\x12\x0c\n\x08mismatch\x10\x08\x12\r\n\tstatus_na\x10\t\x12\x11\n\rspecies_match\x10\n\x12\x14\n\x10subspecies_match\x10\x0b\x12\x11\n\rsynonym_match\x10\x0c\"\xb9\x04\n\x08\x41NIMatch\x12\x45\n\x08\x61ssembly\x18\x02 \x01(\tB)\xc2\xf3\x18%\n\x08\x61ssembly\x12\x08\x41ssembly2\x0fGCA_010191885.1R\x08\x61ssembly\x12v\n\rorganism_name\x18\x03 \x01(\tBQ\xc2\xf3\x18M\n\x08organism\x12\x08Organism27Salmonella enterica subsp. enterica serovar TyphimuriumR\x0corganismName\x12s\n\x08\x63\x61tegory\x18\x04 \x01(\x0e\x32).ncbi.datasets.v2.reports.ANITypeCategoryB,\xc2\xf3\x18(\n\x08\x63\x61tegory\x12\rType Category2\rType materialR\x08\x63\x61tegory\x12&\n\x03\x61ni\x18\x05 \x01(\x02\x42\x14\xc2\xf3\x18\x10\n\x03\x61ni\x12\x03\x41NI2\x04\x39\x38.5R\x03\x61ni\x12^\n\x11\x61ssembly_coverage\x18\x06 \x01(\x02\x42\x31\xc2\xf3\x18-\n\x11\x61ssembly_coverage\x12\x11\x41ssembly Coverage2\x05\x39\x30.75R\x10\x61ssemblyCoverage\x12q\n\x16type_assembly_coverage\x18\x07 \x01(\x02\x42;\xc2\xf3\x18\x37\n\x16type_assembly_coverage\x12\x16Type Assembly Coverage2\x05\x38\x39.60R\x14typeAssemblyCoverage*j\n\x0eSourceDatabase\x12\x1f\n\x1bSOURCE_DATABASE_UNSPECIFIED\x10\x00\x12\x1b\n\x17SOURCE_DATABASE_GENBANK\x10\x01\x12\x1a\n\x16SOURCE_DATABASE_REFSEQ\x10\x02*9\n\x0b\x43ontentType\x12\x0c\n\x08\x43OMPLETE\x10\x00\x12\x0c\n\x08\x41SSM_ACC\x10\x01\x12\x0e\n\nPAIRED_ACC\x10\x02*\xa1\x02\n\x12LinkedAssemblyType\x12 \n\x1cLINKED_ASSEMBLY_TYPE_UNKNOWN\x10\x00\x12(\n$alternate_pseudohaplotype_of_diploid\x10\x01\x12(\n$principal_pseudohaplotype_of_diploid\x10\x02\x12!\n\x1dmaternal_haplotype_of_diploid\x10\x03\x12!\n\x1dpaternal_haplotype_of_diploid\x10\x04\x12\x0f\n\x0bhaplotype_1\x10\x06\x12\x0f\n\x0bhaplotype_2\x10\x07\x12\x0f\n\x0bhaplotype_3\x10\x08\x12\x0f\n\x0bhaplotype_4\x10\t\x12\x0b\n\x07haploid\x10\n*\xaa\x01\n\x0f\x41NITypeCategory\x12\x18\n\x14\x41NI_CATEGORY_UNKNOWN\x10\x00\x12\x0c\n\x08\x63laderef\x10\x01\x12\x0f\n\x0b\x63\x61tegory_na\x10\x02\x12\x0b\n\x07neotype\x10\x03\x12\x0b\n\x07no_type\x10\x04\x12\x0c\n\x08pathovar\x10\x05\x12\x0b\n\x07reftype\x10\x06\x12\x12\n\x0esuspected_type\x10\x07\x12\x0b\n\x07syntype\x10\x08\x12\x08\n\x04type\x10\tB\x82\x0bZ\x18ncbi/datasets/v2/reports\xf8\x01\x01\xc2\xf3\x18\xe0\n\n\x1bGenome Assembly Data Report\x12\x12\x41ssemblyDataReport\x1a;<p>The downloaded genome package contains a genome assembly\x1aIdata report in <a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\x1aYformat in the file: <br><br><em>ncbi_dataset/data/assembly_data_report.jsonl</em><br><br>\x1axEach line of the genome assembly data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\x1ayobject that represents a single genome assembly record. The schema of the genome assembly record is defined in the tables\x1apbelow where each row describes a single field in the report or a sub-structure, which is a collection of fields.\x1aJThe outermost structure of the report is <em>AssemblyDataReport</em>.</p> \x1aQ<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\x1a\x90\x01<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option\x1a\x8e\x01Refer to the <a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this\x1aPtool to transform assembly data reports from JSON Lines to tabular formats.</p> \"\x0bOrientation\"\x0bSeqRangeSet\"\x05Range\"\x0fLineageOrganism\"\x08Organism*\x91\x01{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/genome-report.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}b\x06proto3')

_SOURCEDATABASE = DESCRIPTOR.enum_types_by_name['SourceDatabase']
SourceDatabase = enum_type_wrapper.EnumTypeWrapper(_SOURCEDATABASE)
_CONTENTTYPE = DESCRIPTOR.enum_types_by_name['ContentType']
ContentType = enum_type_wrapper.EnumTypeWrapper(_CONTENTTYPE)
_LINKEDASSEMBLYTYPE = DESCRIPTOR.enum_types_by_name['LinkedAssemblyType']
LinkedAssemblyType = enum_type_wrapper.EnumTypeWrapper(_LINKEDASSEMBLYTYPE)
_ANITYPECATEGORY = DESCRIPTOR.enum_types_by_name['ANITypeCategory']
ANITypeCategory = enum_type_wrapper.EnumTypeWrapper(_ANITYPECATEGORY)
SOURCE_DATABASE_UNSPECIFIED = 0
SOURCE_DATABASE_GENBANK = 1
SOURCE_DATABASE_REFSEQ = 2
COMPLETE = 0
ASSM_ACC = 1
PAIRED_ACC = 2
LINKED_ASSEMBLY_TYPE_UNKNOWN = 0
alternate_pseudohaplotype_of_diploid = 1
principal_pseudohaplotype_of_diploid = 2
maternal_haplotype_of_diploid = 3
paternal_haplotype_of_diploid = 4
haplotype_1 = 6
haplotype_2 = 7
haplotype_3 = 8
haplotype_4 = 9
haploid = 10
ANI_CATEGORY_UNKNOWN = 0
claderef = 1
category_na = 2
neotype = 3
no_type = 4
pathovar = 5
reftype = 6
suspected_type = 7
syntype = 8
type = 9


_ASSEMBLYDATAREPORTPAGE = DESCRIPTOR.message_types_by_name['AssemblyDataReportPage']
_ASSEMBLYDATAREPORT = DESCRIPTOR.message_types_by_name['AssemblyDataReport']
_TYPEMATERIAL = DESCRIPTOR.message_types_by_name['TypeMaterial']
_LINKEDASSEMBLY = DESCRIPTOR.message_types_by_name['LinkedAssembly']
_PAIREDASSEMBLY = DESCRIPTOR.message_types_by_name['PairedAssembly']
_ATYPICALINFO = DESCRIPTOR.message_types_by_name['AtypicalInfo']
_ASSEMBLYINFO = DESCRIPTOR.message_types_by_name['AssemblyInfo']
_ASSEMBLYSTATS = DESCRIPTOR.message_types_by_name['AssemblyStats']
_ORGANELLEINFO = DESCRIPTOR.message_types_by_name['OrganelleInfo']
_ANNOTATIONINFO = DESCRIPTOR.message_types_by_name['AnnotationInfo']
_FEATURECOUNTS = DESCRIPTOR.message_types_by_name['FeatureCounts']
_BUSCOSTAT = DESCRIPTOR.message_types_by_name['BuscoStat']
_GENECOUNTS = DESCRIPTOR.message_types_by_name['GeneCounts']
_WGSINFO = DESCRIPTOR.message_types_by_name['WGSInfo']
_CHECKM = DESCRIPTOR.message_types_by_name['CheckM']
_AVERAGENUCLEOTIDEIDENTITY = DESCRIPTOR.message_types_by_name['AverageNucleotideIdentity']
_ANIMATCH = DESCRIPTOR.message_types_by_name['ANIMatch']
_AVERAGENUCLEOTIDEIDENTITY_TAXONOMYCHECKSTATUS = _AVERAGENUCLEOTIDEIDENTITY.enum_types_by_name['TaxonomyCheckStatus']
_AVERAGENUCLEOTIDEIDENTITY_MATCHSTATUS = _AVERAGENUCLEOTIDEIDENTITY.enum_types_by_name['MatchStatus']
AssemblyDataReportPage = _reflection.GeneratedProtocolMessageType('AssemblyDataReportPage', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATAREPORTPAGE,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.AssemblyDataReportPage)
  })
_sym_db.RegisterMessage(AssemblyDataReportPage)

AssemblyDataReport = _reflection.GeneratedProtocolMessageType('AssemblyDataReport', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYDATAREPORT,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.AssemblyDataReport)
  })
_sym_db.RegisterMessage(AssemblyDataReport)

TypeMaterial = _reflection.GeneratedProtocolMessageType('TypeMaterial', (_message.Message,), {
  'DESCRIPTOR' : _TYPEMATERIAL,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.TypeMaterial)
  })
_sym_db.RegisterMessage(TypeMaterial)

LinkedAssembly = _reflection.GeneratedProtocolMessageType('LinkedAssembly', (_message.Message,), {
  'DESCRIPTOR' : _LINKEDASSEMBLY,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.LinkedAssembly)
  })
_sym_db.RegisterMessage(LinkedAssembly)

PairedAssembly = _reflection.GeneratedProtocolMessageType('PairedAssembly', (_message.Message,), {
  'DESCRIPTOR' : _PAIREDASSEMBLY,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.PairedAssembly)
  })
_sym_db.RegisterMessage(PairedAssembly)

AtypicalInfo = _reflection.GeneratedProtocolMessageType('AtypicalInfo', (_message.Message,), {
  'DESCRIPTOR' : _ATYPICALINFO,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.AtypicalInfo)
  })
_sym_db.RegisterMessage(AtypicalInfo)

AssemblyInfo = _reflection.GeneratedProtocolMessageType('AssemblyInfo', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYINFO,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.AssemblyInfo)
  })
_sym_db.RegisterMessage(AssemblyInfo)

AssemblyStats = _reflection.GeneratedProtocolMessageType('AssemblyStats', (_message.Message,), {
  'DESCRIPTOR' : _ASSEMBLYSTATS,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.AssemblyStats)
  })
_sym_db.RegisterMessage(AssemblyStats)

OrganelleInfo = _reflection.GeneratedProtocolMessageType('OrganelleInfo', (_message.Message,), {
  'DESCRIPTOR' : _ORGANELLEINFO,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.OrganelleInfo)
  })
_sym_db.RegisterMessage(OrganelleInfo)

AnnotationInfo = _reflection.GeneratedProtocolMessageType('AnnotationInfo', (_message.Message,), {
  'DESCRIPTOR' : _ANNOTATIONINFO,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.AnnotationInfo)
  })
_sym_db.RegisterMessage(AnnotationInfo)

FeatureCounts = _reflection.GeneratedProtocolMessageType('FeatureCounts', (_message.Message,), {
  'DESCRIPTOR' : _FEATURECOUNTS,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.FeatureCounts)
  })
_sym_db.RegisterMessage(FeatureCounts)

BuscoStat = _reflection.GeneratedProtocolMessageType('BuscoStat', (_message.Message,), {
  'DESCRIPTOR' : _BUSCOSTAT,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.BuscoStat)
  })
_sym_db.RegisterMessage(BuscoStat)

GeneCounts = _reflection.GeneratedProtocolMessageType('GeneCounts', (_message.Message,), {
  'DESCRIPTOR' : _GENECOUNTS,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GeneCounts)
  })
_sym_db.RegisterMessage(GeneCounts)

WGSInfo = _reflection.GeneratedProtocolMessageType('WGSInfo', (_message.Message,), {
  'DESCRIPTOR' : _WGSINFO,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.WGSInfo)
  })
_sym_db.RegisterMessage(WGSInfo)

CheckM = _reflection.GeneratedProtocolMessageType('CheckM', (_message.Message,), {
  'DESCRIPTOR' : _CHECKM,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.CheckM)
  })
_sym_db.RegisterMessage(CheckM)

AverageNucleotideIdentity = _reflection.GeneratedProtocolMessageType('AverageNucleotideIdentity', (_message.Message,), {
  'DESCRIPTOR' : _AVERAGENUCLEOTIDEIDENTITY,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.AverageNucleotideIdentity)
  })
_sym_db.RegisterMessage(AverageNucleotideIdentity)

ANIMatch = _reflection.GeneratedProtocolMessageType('ANIMatch', (_message.Message,), {
  'DESCRIPTOR' : _ANIMATCH,
  '__module__' : 'ncbi.datasets.v2.reports.assembly_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.ANIMatch)
  })
_sym_db.RegisterMessage(ANIMatch)

if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'Z\030ncbi/datasets/v2/reports\370\001\001\302\363\030\340\n\n\033Genome Assembly Data Report\022\022AssemblyDataReport\032;<p>The downloaded genome package contains a genome assembly\032Idata report in <a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\032Yformat in the file: <br><br><em>ncbi_dataset/data/assembly_data_report.jsonl</em><br><br>\032xEach line of the genome assembly data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\032yobject that represents a single genome assembly record. The schema of the genome assembly record is defined in the tables\032pbelow where each row describes a single field in the report or a sub-structure, which is a collection of fields.\032JThe outermost structure of the report is <em>AssemblyDataReport</em>.</p> \032Q<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\032\220\001<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option\032\216\001Refer to the <a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this\032Ptool to transform assembly data reports from JSON Lines to tabular formats.</p> \"\013Orientation\"\013SeqRangeSet\"\005Range\"\017LineageOrganism\"\010Organism*\221\001{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/genome-report.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}'
  _ASSEMBLYDATAREPORT.fields_by_name['accession']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['accession']._serialized_options = b'\302\363\0301\n\taccession\022\022Assembly Accession2\020GCF_000001405.40'
  _ASSEMBLYDATAREPORT.fields_by_name['current_accession']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['current_accession']._serialized_options = b'\302\363\0308\n\021current-accession\022\021Current Accession2\020GCF_000001405.40'
  _ASSEMBLYDATAREPORT.fields_by_name['paired_accession']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['paired_accession']._serialized_options = b'\030\001'
  _ASSEMBLYDATAREPORT.fields_by_name['source_database']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['source_database']._serialized_options = b'\302\363\0303\n\017source_database\022\017Source Database2\006REFSEQ2\007GENBANK'
  _ASSEMBLYDATAREPORT.fields_by_name['organism']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['organism']._serialized_options = b'\302\363\030\026\n\torganism-\022\tOrganism '
  _ASSEMBLYDATAREPORT.fields_by_name['assembly_info']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['assembly_info']._serialized_options = b'\302\363\030\026\n\tassminfo-\022\tAssembly '
  _ASSEMBLYDATAREPORT.fields_by_name['assembly_stats']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['assembly_stats']._serialized_options = b'\302\363\030\035\n\nassmstats-\022\017Assembly Stats '
  _ASSEMBLYDATAREPORT.fields_by_name['organelle_info']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['organelle_info']._serialized_options = b'\302\363\030\030\n\norganelle-\022\nOrganelle '
  _ASSEMBLYDATAREPORT.fields_by_name['annotation_info']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['annotation_info']._serialized_options = b'\302\363\030\031\n\nannotinfo-\022\013Annotation '
  _ASSEMBLYDATAREPORT.fields_by_name['wgs_info']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['wgs_info']._serialized_options = b'\302\363\030\014\n\004wgs-\022\004WGS '
  _ASSEMBLYDATAREPORT.fields_by_name['type_material']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['type_material']._serialized_options = b'\302\363\030 \n\016type_material-\022\016Type Material '
  _ASSEMBLYDATAREPORT.fields_by_name['checkm_info']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['checkm_info']._serialized_options = b'\302\363\030\022\n\007checkm-\022\007CheckM '
  _ASSEMBLYDATAREPORT.fields_by_name['average_nucleotide_identity']._options = None
  _ASSEMBLYDATAREPORT.fields_by_name['average_nucleotide_identity']._serialized_options = b'\302\363\030\014\n\004ani-\022\004ANI '
  _TYPEMATERIAL.fields_by_name['type_label']._options = None
  _TYPEMATERIAL.fields_by_name['type_label']._serialized_options = b'\302\363\030\016\n\005label\022\005Label'
  _TYPEMATERIAL.fields_by_name['type_display_text']._options = None
  _TYPEMATERIAL.fields_by_name['type_display_text']._serialized_options = b'\302\363\030\034\n\014display_text\022\014Display Text'
  _LINKEDASSEMBLY.fields_by_name['linked_assembly']._options = None
  _LINKEDASSEMBLY.fields_by_name['linked_assembly']._serialized_options = b'\302\363\030\'\n\taccession\022\tAccession2\017GCA_000212995.1'
  _LINKEDASSEMBLY.fields_by_name['assembly_type']._options = None
  _LINKEDASSEMBLY.fields_by_name['assembly_type']._serialized_options = b'\302\363\030\014\n\004type\022\004Type'
  _PAIREDASSEMBLY.fields_by_name['accession']._options = None
  _PAIREDASSEMBLY.fields_by_name['accession']._serialized_options = b'\302\363\030(\n\taccession\022\tAccession2\020GCF_000001405.40'
  _PAIREDASSEMBLY.fields_by_name['status']._options = None
  _PAIREDASSEMBLY.fields_by_name['status']._serialized_options = b'\302\363\030\031\n\006status\022\006Status2\007current'
  _PAIREDASSEMBLY.fields_by_name['annotation_name']._options = None
  _PAIREDASSEMBLY.fields_by_name['annotation_name']._serialized_options = b'\302\363\030\014\n\004name\022\004Name'
  _PAIREDASSEMBLY.fields_by_name['only_genbank']._options = None
  _PAIREDASSEMBLY.fields_by_name['only_genbank']._serialized_options = b'\302\363\030\034\n\014only-genbank\022\014Only Genbank'
  _PAIREDASSEMBLY.fields_by_name['only_refseq']._options = None
  _PAIREDASSEMBLY.fields_by_name['only_refseq']._serialized_options = b'\302\363\030\032\n\013only-refseq\022\013Only RefSeq'
  _PAIREDASSEMBLY.fields_by_name['changed']._options = None
  _PAIREDASSEMBLY.fields_by_name['changed']._serialized_options = b'\302\363\030\022\n\007changed\022\007Changed'
  _PAIREDASSEMBLY.fields_by_name['manual_diff']._options = None
  _PAIREDASSEMBLY.fields_by_name['manual_diff']._serialized_options = b'\302\363\030\032\n\013manual-diff\022\013Manual Diff'
  _ATYPICALINFO.fields_by_name['is_atypical']._options = None
  _ATYPICALINFO.fields_by_name['is_atypical']._serialized_options = b'\302\363\030\032\n\013is-atypical\022\013Is Atypical'
  _ATYPICALINFO.fields_by_name['warnings']._options = None
  _ATYPICALINFO.fields_by_name['warnings']._serialized_options = b'\302\363\030\024\n\010warnings\022\010Warnings'
  _ASSEMBLYINFO.fields_by_name['assembly_level']._options = None
  _ASSEMBLYINFO.fields_by_name['assembly_level']._serialized_options = b'\302\363\030,\n\005level\022\005Level2\nchromosome2\010scaffold2\006contig'
  _ASSEMBLYINFO.fields_by_name['assembly_status']._options = None
  _ASSEMBLYINFO.fields_by_name['assembly_status']._serialized_options = b'\302\363\030\031\n\006status\022\006Status2\007current'
  _ASSEMBLYINFO.fields_by_name['paired_assembly']._options = None
  _ASSEMBLYINFO.fields_by_name['paired_assembly']._serialized_options = b'\302\363\030 \n\014paired-assm-\022\020Paired Assembly '
  _ASSEMBLYINFO.fields_by_name['assembly_name']._options = None
  _ASSEMBLYINFO.fields_by_name['assembly_name']._serialized_options = b'\302\363\030%\n\004name\022\004Name2\nGRCh38.p142\013ASM985889v3'
  _ASSEMBLYINFO.fields_by_name['assembly_type']._options = None
  _ASSEMBLYINFO.fields_by_name['assembly_type']._serialized_options = b'\302\363\030,\n\004type\022\004Type2\025haploid-with-alt-loci2\007haploid'
  _ASSEMBLYINFO.fields_by_name['bioproject_lineage']._options = None
  _ASSEMBLYINFO.fields_by_name['bioproject_lineage']._serialized_options = b'\302\363\030\032\n\013bioproject-\022\013BioProject '
  _ASSEMBLYINFO.fields_by_name['bioproject_accession']._options = None
  _ASSEMBLYINFO.fields_by_name['bioproject_accession']._serialized_options = b'\302\363\030\"\n\nbioproject\022\024BioProject Accession'
  _ASSEMBLYINFO.fields_by_name['submission_date']._options = None
  _ASSEMBLYINFO.fields_by_name['submission_date']._serialized_options = b'\030\001'
  _ASSEMBLYINFO.fields_by_name['release_date']._options = None
  _ASSEMBLYINFO.fields_by_name['release_date']._serialized_options = b'\302\363\030\034\n\014release-date\022\014Release Date'
  _ASSEMBLYINFO.fields_by_name['description']._options = None
  _ASSEMBLYINFO.fields_by_name['description']._serialized_options = b'\302\363\030\032\n\013description\022\013Description'
  _ASSEMBLYINFO.fields_by_name['submitter']._options = None
  _ASSEMBLYINFO.fields_by_name['submitter']._serialized_options = b'\302\363\030\026\n\tsubmitter\022\tSubmitter'
  _ASSEMBLYINFO.fields_by_name['refseq_category']._options = None
  _ASSEMBLYINFO.fields_by_name['refseq_category']._serialized_options = b'\302\363\030K\n\017refseq-category\022\017Refseq Category2\020reference genome2\025representative genome'
  _ASSEMBLYINFO.fields_by_name['synonym']._options = None
  _ASSEMBLYINFO.fields_by_name['synonym']._serialized_options = b'\302\363\030\030\n\007synonym\022\007Synonym2\004hg38'
  _ASSEMBLYINFO.fields_by_name['linked_assembly']._options = None
  _ASSEMBLYINFO.fields_by_name['linked_assembly']._serialized_options = b'\030\001'
  _ASSEMBLYINFO.fields_by_name['linked_assemblies']._options = None
  _ASSEMBLYINFO.fields_by_name['linked_assemblies']._serialized_options = b'\302\363\030 \n\014linked-assm-\022\020Linked Assembly '
  _ASSEMBLYINFO.fields_by_name['atypical']._options = None
  _ASSEMBLYINFO.fields_by_name['atypical']._serialized_options = b'\302\363\030\025\n\010atypical\022\tAtypical '
  _ASSEMBLYINFO.fields_by_name['genome_notes']._options = None
  _ASSEMBLYINFO.fields_by_name['genome_notes']._serialized_options = b'\302\363\030\017\n\005notes\022\006Notes '
  _ASSEMBLYINFO.fields_by_name['sequencing_tech']._options = None
  _ASSEMBLYINFO.fields_by_name['sequencing_tech']._serialized_options = b'\302\363\030\"\n\017sequencing-tech\022\017Sequencing Tech'
  _ASSEMBLYINFO.fields_by_name['assembly_method']._options = None
  _ASSEMBLYINFO.fields_by_name['assembly_method']._serialized_options = b'\302\363\030\"\n\017assembly-method\022\017Assembly Method'
  _ASSEMBLYINFO.fields_by_name['biosample']._options = None
  _ASSEMBLYINFO.fields_by_name['biosample']._serialized_options = b'\302\363\030\030\n\nbiosample-\022\nBioSample '
  _ASSEMBLYINFO.fields_by_name['blast_url']._options = None
  _ASSEMBLYINFO.fields_by_name['blast_url']._serialized_options = b'\302\363\030\026\n\tblast-url\022\tBlast URL'
  _ASSEMBLYINFO.fields_by_name['suppression_reason']._options = None
  _ASSEMBLYINFO.fields_by_name['suppression_reason']._serialized_options = b'\302\363\030(\n\022suppression-reason\022\022Suppression Reason'
  _ASSEMBLYSTATS.fields_by_name['total_number_of_chromosomes']._options = None
  _ASSEMBLYSTATS.fields_by_name['total_number_of_chromosomes']._serialized_options = b'\302\363\030:\n\033total-number-of-chromosomes\022\033Total Number of Chromosomes'
  _ASSEMBLYSTATS.fields_by_name['total_sequence_length']._options = None
  _ASSEMBLYSTATS.fields_by_name['total_sequence_length']._serialized_options = b'\302\363\030+\n\022total-sequence-len\022\025Total Sequence Length'
  _ASSEMBLYSTATS.fields_by_name['total_ungapped_length']._options = None
  _ASSEMBLYSTATS.fields_by_name['total_ungapped_length']._serialized_options = b'\302\363\030+\n\022total-ungapped-len\022\025Total Ungapped Length'
  _ASSEMBLYSTATS.fields_by_name['number_of_contigs']._options = None
  _ASSEMBLYSTATS.fields_by_name['number_of_contigs']._serialized_options = b'\302\363\030&\n\021number-of-contigs\022\021Number of Contigs'
  _ASSEMBLYSTATS.fields_by_name['contig_n50']._options = None
  _ASSEMBLYSTATS.fields_by_name['contig_n50']._serialized_options = b'\302\363\030\030\n\ncontig-n50\022\nContig N50'
  _ASSEMBLYSTATS.fields_by_name['contig_l50']._options = None
  _ASSEMBLYSTATS.fields_by_name['contig_l50']._serialized_options = b'\302\363\030\030\n\ncontig-l50\022\nContig L50'
  _ASSEMBLYSTATS.fields_by_name['number_of_scaffolds']._options = None
  _ASSEMBLYSTATS.fields_by_name['number_of_scaffolds']._serialized_options = b'\302\363\030*\n\023number-of-scaffolds\022\023Number of Scaffolds'
  _ASSEMBLYSTATS.fields_by_name['scaffold_n50']._options = None
  _ASSEMBLYSTATS.fields_by_name['scaffold_n50']._serialized_options = b'\302\363\030\034\n\014scaffold-n50\022\014Scaffold N50'
  _ASSEMBLYSTATS.fields_by_name['scaffold_l50']._options = None
  _ASSEMBLYSTATS.fields_by_name['scaffold_l50']._serialized_options = b'\302\363\030\034\n\014scaffold-l50\022\014Scaffold L50'
  _ASSEMBLYSTATS.fields_by_name['gaps_between_scaffolds_count']._options = None
  _ASSEMBLYSTATS.fields_by_name['gaps_between_scaffolds_count']._serialized_options = b'\302\363\030<\n\034gaps-between-scaffolds-count\022\034Gaps Between Scaffolds Count'
  _ASSEMBLYSTATS.fields_by_name['number_of_component_sequences']._options = None
  _ASSEMBLYSTATS.fields_by_name['number_of_component_sequences']._serialized_options = b'\302\363\030>\n\035number-of-component-sequences\022\035Number of Component Sequences'
  _ASSEMBLYSTATS.fields_by_name['gc_count']._options = None
  _ASSEMBLYSTATS.fields_by_name['gc_count']._serialized_options = b'\030\001\302\363\030\024\n\010gc-count\022\010GC Count'
  _ASSEMBLYSTATS.fields_by_name['gc_percent']._options = None
  _ASSEMBLYSTATS.fields_by_name['gc_percent']._serialized_options = b'\302\363\030\030\n\ngc-percent\022\nGC Percent'
  _ASSEMBLYSTATS.fields_by_name['genome_coverage']._options = None
  _ASSEMBLYSTATS.fields_by_name['genome_coverage']._serialized_options = b'\302\363\030\"\n\017genome-coverage\022\017Genome Coverage'
  _ASSEMBLYSTATS.fields_by_name['number_of_organelles']._options = None
  _ASSEMBLYSTATS.fields_by_name['number_of_organelles']._serialized_options = b'\302\363\030,\n\024number-of-organelles\022\024Number of Organelles'
  _ORGANELLEINFO.fields_by_name['assembly_name']._options = None
  _ORGANELLEINFO.fields_by_name['assembly_name']._serialized_options = b'\030\001\302\363\030\036\n\rassembly-name\022\rAssembly Name'
  _ORGANELLEINFO.fields_by_name['infraspecific_name']._options = None
  _ORGANELLEINFO.fields_by_name['infraspecific_name']._serialized_options = b'\302\363\030(\n\022infraspecific-name\022\022Infraspecific Name'
  _ORGANELLEINFO.fields_by_name['bioproject']._options = None
  _ORGANELLEINFO.fields_by_name['bioproject']._serialized_options = b'\302\363\030.\n\025bioproject-accessions\022\025BioProject Accessions'
  _ORGANELLEINFO.fields_by_name['description']._options = None
  _ORGANELLEINFO.fields_by_name['description']._serialized_options = b'\302\363\030\032\n\013description\022\013Description'
  _ORGANELLEINFO.fields_by_name['total_seq_length']._options = None
  _ORGANELLEINFO.fields_by_name['total_seq_length']._serialized_options = b'\302\363\030$\n\020total-seq-length\022\020Total Seq Length'
  _ORGANELLEINFO.fields_by_name['submitter']._options = None
  _ORGANELLEINFO.fields_by_name['submitter']._serialized_options = b'\302\363\030\026\n\tsubmitter\022\tSubmitter'
  _ANNOTATIONINFO.fields_by_name['name']._options = None
  _ANNOTATIONINFO.fields_by_name['name']._serialized_options = b'\302\363\030\014\n\004name\022\004Name'
  _ANNOTATIONINFO.fields_by_name['provider']._options = None
  _ANNOTATIONINFO.fields_by_name['provider']._serialized_options = b'\302\363\030\024\n\010provider\022\010Provider'
  _ANNOTATIONINFO.fields_by_name['release_date']._options = None
  _ANNOTATIONINFO.fields_by_name['release_date']._serialized_options = b'\302\363\030\034\n\014release-date\022\014Release Date'
  _ANNOTATIONINFO.fields_by_name['report_url']._options = None
  _ANNOTATIONINFO.fields_by_name['report_url']._serialized_options = b'\302\363\030\030\n\nreport-url\022\nReport URL'
  _ANNOTATIONINFO.fields_by_name['stats']._options = None
  _ANNOTATIONINFO.fields_by_name['stats']._serialized_options = b'\302\363\030\024\n\nfeatcount-\022\006Count '
  _ANNOTATIONINFO.fields_by_name['busco']._options = None
  _ANNOTATIONINFO.fields_by_name['busco']._serialized_options = b'\302\363\030\020\n\006busco-\022\006BUSCO '
  _ANNOTATIONINFO.fields_by_name['method']._options = None
  _ANNOTATIONINFO.fields_by_name['method']._serialized_options = b'\302\363\030\020\n\006method\022\006Method'
  _ANNOTATIONINFO.fields_by_name['pipeline']._options = None
  _ANNOTATIONINFO.fields_by_name['pipeline']._serialized_options = b'\302\363\030\024\n\010pipeline\022\010Pipeline'
  _ANNOTATIONINFO.fields_by_name['software_version']._options = None
  _ANNOTATIONINFO.fields_by_name['software_version']._serialized_options = b'\302\363\030$\n\020software-version\022\020Software Version'
  _ANNOTATIONINFO.fields_by_name['status']._options = None
  _ANNOTATIONINFO.fields_by_name['status']._serialized_options = b'\302\363\030\020\n\006status\022\006Status'
  _ANNOTATIONINFO.fields_by_name['release_version']._options = None
  _ANNOTATIONINFO.fields_by_name['release_version']._serialized_options = b'\302\363\030\"\n\017release-version\022\017Release Version'
  _FEATURECOUNTS.fields_by_name['gene_counts']._options = None
  _FEATURECOUNTS.fields_by_name['gene_counts']._serialized_options = b'\302\363\030\016\n\005gene-\022\005Gene '
  _BUSCOSTAT.fields_by_name['busco_lineage']._options = None
  _BUSCOSTAT.fields_by_name['busco_lineage']._serialized_options = b'\302\363\030\023\n\007lineage\022\010Lineage '
  _BUSCOSTAT.fields_by_name['busco_ver']._options = None
  _BUSCOSTAT.fields_by_name['busco_ver']._serialized_options = b'\302\363\030\017\n\003ver\022\010Version '
  _BUSCOSTAT.fields_by_name['complete']._options = None
  _BUSCOSTAT.fields_by_name['complete']._serialized_options = b'\302\363\030\025\n\010complete\022\tComplete '
  _BUSCOSTAT.fields_by_name['single_copy']._options = None
  _BUSCOSTAT.fields_by_name['single_copy']._serialized_options = b'\302\363\030\032\n\nsinglecopy\022\014Single Copy '
  _BUSCOSTAT.fields_by_name['duplicated']._options = None
  _BUSCOSTAT.fields_by_name['duplicated']._serialized_options = b'\302\363\030\031\n\nduplicated\022\013Duplicated '
  _BUSCOSTAT.fields_by_name['fragmented']._options = None
  _BUSCOSTAT.fields_by_name['fragmented']._serialized_options = b'\302\363\030\031\n\nfragmented\022\013Fragmented '
  _BUSCOSTAT.fields_by_name['missing']._options = None
  _BUSCOSTAT.fields_by_name['missing']._serialized_options = b'\302\363\030\023\n\007missing\022\010Missing '
  _BUSCOSTAT.fields_by_name['total_count']._options = None
  _BUSCOSTAT.fields_by_name['total_count']._serialized_options = b'\302\363\030\032\n\ntotalcount\022\014Total Count '
  _GENECOUNTS.fields_by_name['total']._options = None
  _GENECOUNTS.fields_by_name['total']._serialized_options = b'\302\363\030\016\n\005total\022\005Total'
  _GENECOUNTS.fields_by_name['protein_coding']._options = None
  _GENECOUNTS.fields_by_name['protein_coding']._serialized_options = b'\302\363\030 \n\016protein-coding\022\016Protein-coding'
  _GENECOUNTS.fields_by_name['non_coding']._options = None
  _GENECOUNTS.fields_by_name['non_coding']._serialized_options = b'\302\363\030\030\n\nnon-coding\022\nNon-coding'
  _GENECOUNTS.fields_by_name['pseudogene']._options = None
  _GENECOUNTS.fields_by_name['pseudogene']._serialized_options = b'\302\363\030\030\n\npseudogene\022\nPseudogene'
  _GENECOUNTS.fields_by_name['other']._options = None
  _GENECOUNTS.fields_by_name['other']._serialized_options = b'\302\363\030\016\n\005other\022\005Other'
  _WGSINFO.fields_by_name['wgs_project_accession']._options = None
  _WGSINFO.fields_by_name['wgs_project_accession']._serialized_options = b'\302\363\0308\n\021project-accession\022\021project accession2\006AAEX032\010CABHLF01'
  _WGSINFO.fields_by_name['master_wgs_url']._options = None
  _WGSINFO.fields_by_name['master_wgs_url']._serialized_options = b'\302\363\030?\n\003url\022\003URL23https://www.ncbi.nlm.nih.gov/nuccore/AAEX00000000.3'
  _WGSINFO.fields_by_name['wgs_contigs_url']._options = None
  _WGSINFO.fields_by_name['wgs_contigs_url']._serialized_options = b'\302\363\030J\n\013contigs-url\022\013contigs URL2.https://www.ncbi.nlm.nih.gov/Traces/wgs/AAEX03'
  _CHECKM.fields_by_name['checkm_marker_set']._options = None
  _CHECKM.fields_by_name['checkm_marker_set']._serialized_options = b'\302\363\030-\n\nmarker-set\022\nmarker set2\023Mycobacterium avium'
  _CHECKM.fields_by_name['checkm_species_tax_id']._options = None
  _CHECKM.fields_by_name['checkm_species_tax_id']._serialized_options = b'\302\363\030&\n\016species-tax-id\022\016species tax id2\0041764'
  _CHECKM.fields_by_name['checkm_marker_set_rank']._options = None
  _CHECKM.fields_by_name['checkm_marker_set_rank']._serialized_options = b'\302\363\0302\n\017marker-set-rank\022\017marker set rank2\007species2\005genus'
  _CHECKM.fields_by_name['checkm_version']._options = None
  _CHECKM.fields_by_name['checkm_version']._serialized_options = b'\302\363\030\032\n\007version\022\007version2\006v1.2.0'
  _CHECKM.fields_by_name['completeness']._options = None
  _CHECKM.fields_by_name['completeness']._serialized_options = b'\302\363\030#\n\014completeness\022\014completeness2\00586.83'
  _CHECKM.fields_by_name['contamination']._options = None
  _CHECKM.fields_by_name['contamination']._serialized_options = b'\302\363\030$\n\rcontamination\022\rcontamination2\0045.18'
  _CHECKM.fields_by_name['completeness_percentile']._options = None
  _CHECKM.fields_by_name['completeness_percentile']._serialized_options = b'\302\363\0306\n\027completeness-percentile\022\027completeness percentile2\00279'
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['taxonomy_check_status']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['taxonomy_check_status']._serialized_options = b'\302\363\0306\n\014check-status\022\014Check status2\002ok2\006failed2\014inconclusive'
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['match_status']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['match_status']._serialized_options = b'\302\363\030=\n\021best-match-status\022\021Best match status2\025derived-species-match'
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['submitted_organism']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['submitted_organism']._serialized_options = b'\302\363\030o\n\022submitted-organism\022\022Submitted organism2ESalmonella enterica subsp. enterica serovar Tennessee str. CDC07-0191'
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['submitted_species']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['submitted_species']._serialized_options = b'\302\363\030;\n\021submitted-species\022\021Submitted species2\023Salmonella enterica'
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['category']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['category']._serialized_options = b'\302\363\030\035\n\010category\022\010Category2\007syntype'
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['submitted_ani_match']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['submitted_ani_match']._serialized_options = b'\302\363\030+\n\024submitted-ani-match-\022\023Declared ANI match '
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['best_ani_match']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['best_ani_match']._serialized_options = b'\302\363\030\"\n\017best-ani-match-\022\017Best ANI match '
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['comment']._options = None
  _AVERAGENUCLEOTIDEIDENTITY.fields_by_name['comment']._serialized_options = b'\302\363\030\022\n\007comment\022\007Comment'
  _ANIMATCH.fields_by_name['assembly']._options = None
  _ANIMATCH.fields_by_name['assembly']._serialized_options = b'\302\363\030%\n\010assembly\022\010Assembly2\017GCA_010191885.1'
  _ANIMATCH.fields_by_name['organism_name']._options = None
  _ANIMATCH.fields_by_name['organism_name']._serialized_options = b'\302\363\030M\n\010organism\022\010Organism27Salmonella enterica subsp. enterica serovar Typhimurium'
  _ANIMATCH.fields_by_name['category']._options = None
  _ANIMATCH.fields_by_name['category']._serialized_options = b'\302\363\030(\n\010category\022\rType Category2\rType material'
  _ANIMATCH.fields_by_name['ani']._options = None
  _ANIMATCH.fields_by_name['ani']._serialized_options = b'\302\363\030\020\n\003ani\022\003ANI2\00498.5'
  _ANIMATCH.fields_by_name['assembly_coverage']._options = None
  _ANIMATCH.fields_by_name['assembly_coverage']._serialized_options = b'\302\363\030-\n\021assembly_coverage\022\021Assembly Coverage2\00590.75'
  _ANIMATCH.fields_by_name['type_assembly_coverage']._options = None
  _ANIMATCH.fields_by_name['type_assembly_coverage']._serialized_options = b'\302\363\0307\n\026type_assembly_coverage\022\026Type Assembly Coverage2\00589.60'
  _SOURCEDATABASE._serialized_start=11470
  _SOURCEDATABASE._serialized_end=11576
  _CONTENTTYPE._serialized_start=11578
  _CONTENTTYPE._serialized_end=11635
  _LINKEDASSEMBLYTYPE._serialized_start=11638
  _LINKEDASSEMBLYTYPE._serialized_end=11927
  _ANITYPECATEGORY._serialized_start=11930
  _ANITYPECATEGORY._serialized_end=12100
  _ASSEMBLYDATAREPORTPAGE._serialized_start=187
  _ASSEMBLYDATAREPORTPAGE._serialized_end=639
  _ASSEMBLYDATAREPORT._serialized_start=642
  _ASSEMBLYDATAREPORT._serialized_end=2000
  _TYPEMATERIAL._serialized_start=2003
  _TYPEMATERIAL._serialized_end=2146
  _LINKEDASSEMBLY._serialized_start=2149
  _LINKEDASSEMBLY._serialized_end=2352
  _PAIREDASSEMBLY._serialized_start=2355
  _PAIREDASSEMBLY._serialized_end=2852
  _ATYPICALINFO._serialized_start=2855
  _ATYPICALINFO._serialized_end=2988
  _ASSEMBLYINFO._serialized_start=2991
  _ASSEMBLYINFO._serialized_end=4976
  _ASSEMBLYSTATS._serialized_start=4979
  _ASSEMBLYSTATS._serialized_end=6332
  _ORGANELLEINFO._serialized_start=6335
  _ORGANELLEINFO._serialized_end=6810
  _ANNOTATIONINFO._serialized_start=6813
  _ANNOTATIONINFO._serialized_end=7533
  _FEATURECOUNTS._serialized_start=7535
  _FEATURECOUNTS._serialized_end=7641
  _BUSCOSTAT._serialized_start=7644
  _BUSCOSTAT._serialized_end=8129
  _GENECOUNTS._serialized_start=8132
  _GENECOUNTS._serialized_end=8428
  _WGSINFO._serialized_start=8431
  _WGSINFO._serialized_end=8781
  _CHECKM._serialized_start=8784
  _CHECKM._serialized_end=9436
  _AVERAGENUCLEOTIDEIDENTITY._serialized_start=9439
  _AVERAGENUCLEOTIDEIDENTITY._serialized_end=10896
  _AVERAGENUCLEOTIDEIDENTITY_TAXONOMYCHECKSTATUS._serialized_start=10501
  _AVERAGENUCLEOTIDEIDENTITY_TAXONOMYCHECKSTATUS._serialized_end=10595
  _AVERAGENUCLEOTIDEIDENTITY_MATCHSTATUS._serialized_start=10598
  _AVERAGENUCLEOTIDEIDENTITY_MATCHSTATUS._serialized_end=10896
  _ANIMATCH._serialized_start=10899
  _ANIMATCH._serialized_end=11468
# @@protoc_insertion_point(module_scope)
