# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v2/reports/common.proto
"""Generated protocol buffer code."""
from google.protobuf.internal import enum_type_wrapper
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from ncbi.datasets.options import report_pb2 as ncbi_dot_datasets_dot_options_dot_report__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n%ncbi/datasets/v2/reports/common.proto\x12\x18ncbi.datasets.v2.reports\x1a\"ncbi/datasets/options/report.proto\"\x83\x07\n\x05\x45rror\x12\x63\n\x13\x61ssembly_error_code\x18\x06 \x01(\x0e\x32\x31.ncbi.datasets.v2.reports.Error.AssemblyErrorCodeH\x00R\x11\x61ssemblyErrorCode\x12W\n\x0fgene_error_code\x18\x07 \x01(\x0e\x32-.ncbi.datasets.v2.reports.Error.GeneErrorCodeH\x00R\rgeneErrorCode\x12Z\n\x10virus_error_code\x18\x08 \x01(\x0e\x32..ncbi.datasets.v2.reports.Error.VirusErrorCodeH\x00R\x0evirusErrorCode\x12\x63\n\x13taxonomy_error_code\x18\t \x01(\x0e\x32\x31.ncbi.datasets.v2.reports.Error.TaxonomyErrorCodeH\x00R\x11taxonomyErrorCode\x12\x16\n\x06reason\x18\x05 \x01(\tR\x06reason\x12\x18\n\x07message\x18\x02 \x01(\tR\x07message\x12/\n\x13invalid_identifiers\x18\x04 \x03(\tR\x12invalidIdentifiers\"\xa0\x01\n\x11\x41ssemblyErrorCode\x12\x1f\n\x1bUNKNOWN_ASSEMBLY_ERROR_CODE\x10\x00\x12\x1a\n\x16INVALID_BIOPROJECT_IDS\x10\x01\x12!\n\x1dNO_ASSEMBLIES_FOR_BIOPROJECTS\x10\x02\x12\x11\n\rINVALID_TAXON\x10\x03\x12\x18\n\x14MISSING_SEARCH_FIELD\x10\x04\"k\n\rGeneErrorCode\x12\x1b\n\x17UNKNOWN_GENE_ERROR_CODE\x10\x00\x12\x1c\n\x18INCOMPLETE_LOOKUP_SYMBOL\x10\x05\x12\x1f\n\x1bINVALID_TAXON_GENE_ARGUMENT\x10\x06\".\n\x0eVirusErrorCode\x12\x1c\n\x18UNKNOWN_VIRUS_ERROR_CODE\x10\x00\"P\n\x11TaxonomyErrorCode\x12\x1f\n\x1bUNKNOWN_TAXONOMY_ERROR_CODE\x10\x00\x12\x1a\n\x16INVALID_TAXONOMY_TAXON\x10\x01\x42\x06\n\x04\x63ode\"\x9e\x05\n\x07Warning\x12_\n\x11gene_warning_code\x18\x07 \x01(\x0e\x32\x31.ncbi.datasets.v2.reports.Warning.GeneWarningCodeH\x00R\x0fgeneWarningCode\x12\x16\n\x06reason\x18\x02 \x01(\tR\x06reason\x12\x18\n\x07message\x18\x03 \x01(\tR\x07message\x12M\n\x0breplaced_id\x18\x04 \x01(\x0b\x32,.ncbi.datasets.v2.reports.Warning.ReplacedIdR\nreplacedId\x12\x37\n\x17unrecognized_identifier\x18\x05 \x01(\tR\x16unrecognizedIdentifier\x1a\x46\n\nReplacedId\x12\x1c\n\trequested\x18\x01 \x01(\tR\trequested\x12\x1a\n\x08returned\x18\x02 \x01(\tR\x08returned\"\xa7\x02\n\x0fGeneWarningCode\x12\x1d\n\x19UNKNOWN_GENE_WARNING_CODE\x10\x00\x12\x1e\n\x1a\x41\x43\x43\x45SSION_VERSION_MISMATCH\x10\x01\x12\x14\n\x10REPLACED_GENE_ID\x10\x02\x12\x18\n\x14\x44ISCONTINUED_GENE_ID\x10\x03\x12\x18\n\x14UNRECOGNIZED_GENE_ID\x10\x04\x12\x1c\n\x18UNRECOGNIZED_GENE_SYMBOL\x10\x05\x12\x1a\n\x16UNRECOGNIZED_ACCESSION\x10\x06\x12\x1a\n\x16UNRECOGNIZED_TAX_TOKEN\x10\x07\x12\x1c\n\x18NO_GENE_ANNOTATION_FOUND\x10\x08\x12\x17\n\x13\x41\x42OVE_SPECIES_TAXON\x10\tB\x06\n\x04\x63ode\"\x8c\x01\n\x07Message\x12\x37\n\x05\x65rror\x18\x01 \x01(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorH\x00R\x05\x65rror\x12=\n\x07warning\x18\x02 \x01(\x0b\x32!.ncbi.datasets.v2.reports.WarningH\x00R\x07warningB\t\n\x07message\"^\n\x0fLineageOrganism\x12\"\n\x06tax_id\x18\x01 \x01(\rB\x0b\xc2\xf3\x18\x07\x32\x05\x31\x31\x31\x31\x38R\x05taxId\x12\'\n\x04name\x18\x02 \x01(\tB\x13\xc2\xf3\x18\x0f\x32\rCoronaviridaeR\x04name\"\xfd\x05\n\x08Organism\x12@\n\x06tax_id\x18\x01 \x01(\rB)\xc2\xf3\x18%\n\x06tax-id\x12\x0cTaxonomic ID2\x04\x39\x36\x30\x36\x32\x07\x32\x36\x39\x37\x30\x34\x39R\x05taxId\x12\x1d\n\x08sci_name\x18\x02 \x01(\tB\x02\x18\x01R\x07sciName\x12\xd8\x01\n\rorganism_name\x18\x07 \x01(\tB\xb2\x01\xc2\xf3\x18K\n\x04name\x12\x04Name2\x0cHomo sapiens2/Severe acute respiratory syndrome coronavirus 2\xca\xf3\x18%\x12#The scientific name of the organism\xd2\xf3\x18\x36\n\x0fscientific_name\x12#The scientific name of the organismR\x0corganismName\x12]\n\x0b\x63ommon_name\x18\x03 \x01(\tB<\xc2\xf3\x18\x38\n\x0b\x63ommon-name\x12\x0b\x43ommon Name2\x05human2\x08pangolin2\x04MERS2\x05SARS2R\ncommonName\x12\x43\n\x07lineage\x18\x05 \x03(\x0b\x32).ncbi.datasets.v2.reports.LineageOrganismR\x07lineage\x12\x1a\n\x06strain\x18\x06 \x01(\tB\x02\x18\x01R\x06strain\x12i\n\x17pangolin_classification\x18\x08 \x01(\tB0\xc2\xf3\x18,\n\x08pangolin\x12\x17Pangolin Classification2\x07\x42.1.1.7R\x16pangolinClassification\x12\x89\x01\n\x13infraspecific_names\x18\t \x01(\x0b\x32,.ncbi.datasets.v2.reports.InfraspecificNamesB*\xc2\xf3\x18&\n\x0einfraspecific-\x12\x14Infraspecific Names R\x12infraspecificNames\"\x8e\x03\n\x12InfraspecificNames\x12\x39\n\x05\x62reed\x18\x01 \x01(\tB#\xc2\xf3\x18\x1f\n\x05\x62reed\x12\x05\x42reed2\x08Hereford2\x05\x62oxerR\x05\x62reed\x12\x39\n\x08\x63ultivar\x18\x02 \x01(\tB\x1d\xc2\xf3\x18\x19\n\x08\x63ultivar\x12\x08\x43ultivar2\x03\x42\x37\x33R\x08\x63ultivar\x12\x38\n\x07\x65\x63otype\x18\x03 \x01(\tB\x1e\xc2\xf3\x18\x1a\n\x07\x65\x63otype\x12\x07\x45\x63otype2\x06\x41lpineR\x07\x65\x63otype\x12j\n\x07isolate\x18\x04 \x01(\tBP\xc2\xf3\x18L\n\x07isolate\x12\x07Isolate2/L1 Dominette 01449 registration number 421906802\x07Pmale09R\x07isolate\x12(\n\x03sex\x18\x05 \x01(\tB\x16\xc2\xf3\x18\x12\n\x03sex\x12\x03Sex2\x06\x66\x65maleR\x03sex\x12\x32\n\x06strain\x18\x06 \x01(\tB\x1a\xc2\xf3\x18\x16\n\x06strain\x12\x06Strain2\x04SE11R\x06strain\"\x97\x02\n\x05Range\x12(\n\x05\x62\x65gin\x18\x01 \x01(\x04\x42\x12\xc2\xf3\x18\x0e\n\x05start\x12\x05StartR\x05\x62\x65gin\x12\"\n\x03\x65nd\x18\x02 \x01(\x04\x42\x10\xc2\xf3\x18\x0c\n\x04stop\x12\x04StopR\x03\x65nd\x12g\n\x0borientation\x18\x03 \x01(\x0e\x32%.ncbi.datasets.v2.reports.OrientationB\x1e\xc2\xf3\x18\x1a\n\x0borientation\x12\x0bOrientationR\x0borientation\x12(\n\x05order\x18\x04 \x01(\rB\x12\xc2\xf3\x18\x0e\n\x05order\x12\x05OrderR\x05order\x12-\n\x12ribosomal_slippage\x18\x05 \x01(\x05R\x11ribosomalSlippage\"\xa4\x01\n\x0bSeqRangeSet\x12P\n\x11\x61\x63\x63\x65ssion_version\x18\x01 \x01(\tB#\xc2\xf3\x18\x1f\n\taccession\x12\x12Sequence AccessionR\x10\x61\x63\x63\x65ssionVersion\x12\x43\n\x05range\x18\x02 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.RangeB\x0c\xc2\xf3\x18\x08\n\x06range-R\x05range\"\x98\x03\n\nBioProject\x12\x44\n\taccession\x18\x01 \x01(\tB&\xc2\xf3\x18\"\n\taccession\x12\tAccession2\nPRJEB35387R\taccession\x12h\n\x05title\x18\x02 \x01(\tBR\xc2\xf3\x18N\n\x05title\x12\x05Title2>Sciurus carolinensis (grey squirrel) genome assembly, mSciCar1R\x05title\x12U\n\x10parent_accession\x18\x03 \x01(\tB*\x18\x01\xc2\xf3\x18$\n\x10parent-accession\x12\x10Parent AccessionR\x0fparentAccession\x12\x82\x01\n\x11parent_accessions\x18\x04 \x03(\tBU\xc2\xf3\x18Q\n\x11parent-accessions\x12\x11Parent Accessions2)[\"PRJNA489243\",\"PRJEB33226\",\"PRJEB40665\"]R\x10parentAccessions\"u\n\x11\x42ioProjectLineage\x12`\n\x0b\x62ioprojects\x18\x01 \x03(\x0b\x32$.ncbi.datasets.v2.reports.BioProjectB\x18\xc2\xf3\x18\x14\n\x08lineage-\x12\x08Lineage R\x0b\x62ioprojects*N\n\rAssemblyLevel\x12\x0e\n\nchromosome\x10\x00\x12\x0c\n\x08scaffold\x10\x01\x12\n\n\x06\x63ontig\x10\x02\x12\x13\n\x0f\x63omplete_genome\x10\x03*e\n\x0e\x41ssemblyStatus\x12\x1b\n\x17\x41SSEMBLY_STATUS_UNKNOWN\x10\x00\x12\x0b\n\x07\x63urrent\x10\x01\x12\x0c\n\x08previous\x10\x02\x12\x0e\n\nsuppressed\x10\x03\x12\x0b\n\x07retired\x10\x04*,\n\x0bOrientation\x12\x08\n\x04none\x10\x00\x12\x08\n\x04plus\x10\x01\x12\t\n\x05minus\x10\x02*\x9d\x05\n\x08RankType\x12\x0b\n\x07NO_RANK\x10\x00\x12\x10\n\x0cSUPERKINGDOM\x10\x01\x12\x0b\n\x07KINGDOM\x10\x02\x12\x0e\n\nSUBKINGDOM\x10\x03\x12\x0f\n\x0bSUPERPHYLUM\x10\x04\x12\r\n\tSUBPHYLUM\x10\x05\x12\n\n\x06PHYLUM\x10\x06\x12\t\n\x05\x43LADE\x10\x1f\x12\x0e\n\nSUPERCLASS\x10\x07\x12\t\n\x05\x43LASS\x10\x08\x12\x0c\n\x08SUBCLASS\x10\t\x12\x0e\n\nINFRACLASS\x10\n\x12\n\n\x06\x43OHORT\x10\x0b\x12\r\n\tSUBCOHORT\x10\x0c\x12\x0e\n\nSUPERORDER\x10\r\x12\t\n\x05ORDER\x10\x0e\x12\x0c\n\x08SUBORDER\x10\x0f\x12\x0e\n\nINFRAORDER\x10\x10\x12\r\n\tPARVORDER\x10\x11\x12\x0f\n\x0bSUPERFAMILY\x10\x12\x12\n\n\x06\x46\x41MILY\x10\x13\x12\r\n\tSUBFAMILY\x10\x14\x12\t\n\x05GENUS\x10\x15\x12\x0c\n\x08SUBGENUS\x10\x16\x12\x11\n\rSPECIES_GROUP\x10\x17\x12\x14\n\x10SPECIES_SUBGROUP\x10\x18\x12\x0b\n\x07SPECIES\x10\x19\x12\x0e\n\nSUBSPECIES\x10\x1a\x12\t\n\x05TRIBE\x10\x1b\x12\x0c\n\x08SUBTRIBE\x10\x1c\x12\t\n\x05\x46ORMA\x10\x1d\x12\x0c\n\x08VARIETAS\x10\x1e\x12\x0b\n\x06STRAIN\x10\xc0\x02\x12\x0c\n\x07SECTION\x10\xca\x02\x12\x0f\n\nSUBSECTION\x10\xd4\x02\x12\x0f\n\nPATHOGROUP\x10\xde\x02\x12\x0f\n\nSUBVARIETY\x10\xe8\x02\x12\r\n\x08GENOTYPE\x10\xf2\x02\x12\r\n\x08SEROTYPE\x10\xfc\x02\x12\x0c\n\x07ISOLATE\x10\x86\x03\x12\n\n\x05MORPH\x10\x90\x03\x12\x0b\n\x06SERIES\x10\x9a\x03\x12\x14\n\x0f\x46ORMA_SPECIALIS\x10\xa4\x03\x12\x0e\n\tSEROGROUP\x10\xae\x03\x12\x0c\n\x07\x42IOTYPE\x10\xb8\x03*\xf9\x02\n\tCountType\x12\x1a\n\x16\x43OUNT_TYPE_UNSPECIFIED\x10\x00\x12\x17\n\x13\x43OUNT_TYPE_ASSEMBLY\x10\x01\x12\x13\n\x0f\x43OUNT_TYPE_GENE\x10\x02\x12\x13\n\x0f\x43OUNT_TYPE_tRNA\x10\x03\x12\x13\n\x0f\x43OUNT_TYPE_rRNA\x10\x04\x12\x14\n\x10\x43OUNT_TYPE_snRNA\x10\x05\x12\x14\n\x10\x43OUNT_TYPE_scRNA\x10\x06\x12\x15\n\x11\x43OUNT_TYPE_snoRNA\x10\x07\x12\x1d\n\x19\x43OUNT_TYPE_PROTEIN_CODING\x10\x08\x12\x15\n\x11\x43OUNT_TYPE_PSEUDO\x10\t\x12\x19\n\x15\x43OUNT_TYPE_TRANSPOSON\x10\n\x12\x16\n\x12\x43OUNT_TYPE_miscRNA\x10\x0b\x12\x14\n\x10\x43OUNT_TYPE_ncRNA\x10\x0c\x12 \n\x1c\x43OUNT_TYPE_BIOLOGICAL_REGION\x10\r\x12\x14\n\x10\x43OUNT_TYPE_OTHER\x10\x0e\x42\x45Z\x18ncbi/datasets/v2/reports\xf8\x01\x01\xc2\xf3\x18$\n\x18\x43ommon Report Structures\x12\x08Organismb\x06proto3')

_ASSEMBLYLEVEL = DESCRIPTOR.enum_types_by_name['AssemblyLevel']
AssemblyLevel = enum_type_wrapper.EnumTypeWrapper(_ASSEMBLYLEVEL)
_ASSEMBLYSTATUS = DESCRIPTOR.enum_types_by_name['AssemblyStatus']
AssemblyStatus = enum_type_wrapper.EnumTypeWrapper(_ASSEMBLYSTATUS)
_ORIENTATION = DESCRIPTOR.enum_types_by_name['Orientation']
Orientation = enum_type_wrapper.EnumTypeWrapper(_ORIENTATION)
_RANKTYPE = DESCRIPTOR.enum_types_by_name['RankType']
RankType = enum_type_wrapper.EnumTypeWrapper(_RANKTYPE)
_COUNTTYPE = DESCRIPTOR.enum_types_by_name['CountType']
CountType = enum_type_wrapper.EnumTypeWrapper(_COUNTTYPE)
chromosome = 0
scaffold = 1
contig = 2
complete_genome = 3
ASSEMBLY_STATUS_UNKNOWN = 0
current = 1
previous = 2
suppressed = 3
retired = 4
none = 0
plus = 1
minus = 2
NO_RANK = 0
SUPERKINGDOM = 1
KINGDOM = 2
SUBKINGDOM = 3
SUPERPHYLUM = 4
SUBPHYLUM = 5
PHYLUM = 6
CLADE = 31
SUPERCLASS = 7
CLASS = 8
SUBCLASS = 9
INFRACLASS = 10
COHORT = 11
SUBCOHORT = 12
SUPERORDER = 13
ORDER = 14
SUBORDER = 15
INFRAORDER = 16
PARVORDER = 17
SUPERFAMILY = 18
FAMILY = 19
SUBFAMILY = 20
GENUS = 21
SUBGENUS = 22
SPECIES_GROUP = 23
SPECIES_SUBGROUP = 24
SPECIES = 25
SUBSPECIES = 26
TRIBE = 27
SUBTRIBE = 28
FORMA = 29
VARIETAS = 30
STRAIN = 320
SECTION = 330
SUBSECTION = 340
PATHOGROUP = 350
SUBVARIETY = 360
GENOTYPE = 370
SEROTYPE = 380
ISOLATE = 390
MORPH = 400
SERIES = 410
FORMA_SPECIALIS = 420
SEROGROUP = 430
BIOTYPE = 440
COUNT_TYPE_UNSPECIFIED = 0
COUNT_TYPE_ASSEMBLY = 1
COUNT_TYPE_GENE = 2
COUNT_TYPE_tRNA = 3
COUNT_TYPE_rRNA = 4
COUNT_TYPE_snRNA = 5
COUNT_TYPE_scRNA = 6
COUNT_TYPE_snoRNA = 7
COUNT_TYPE_PROTEIN_CODING = 8
COUNT_TYPE_PSEUDO = 9
COUNT_TYPE_TRANSPOSON = 10
COUNT_TYPE_miscRNA = 11
COUNT_TYPE_ncRNA = 12
COUNT_TYPE_BIOLOGICAL_REGION = 13
COUNT_TYPE_OTHER = 14


_ERROR = DESCRIPTOR.message_types_by_name['Error']
_WARNING = DESCRIPTOR.message_types_by_name['Warning']
_WARNING_REPLACEDID = _WARNING.nested_types_by_name['ReplacedId']
_MESSAGE = DESCRIPTOR.message_types_by_name['Message']
_LINEAGEORGANISM = DESCRIPTOR.message_types_by_name['LineageOrganism']
_ORGANISM = DESCRIPTOR.message_types_by_name['Organism']
_INFRASPECIFICNAMES = DESCRIPTOR.message_types_by_name['InfraspecificNames']
_RANGE = DESCRIPTOR.message_types_by_name['Range']
_SEQRANGESET = DESCRIPTOR.message_types_by_name['SeqRangeSet']
_BIOPROJECT = DESCRIPTOR.message_types_by_name['BioProject']
_BIOPROJECTLINEAGE = DESCRIPTOR.message_types_by_name['BioProjectLineage']
_ERROR_ASSEMBLYERRORCODE = _ERROR.enum_types_by_name['AssemblyErrorCode']
_ERROR_GENEERRORCODE = _ERROR.enum_types_by_name['GeneErrorCode']
_ERROR_VIRUSERRORCODE = _ERROR.enum_types_by_name['VirusErrorCode']
_ERROR_TAXONOMYERRORCODE = _ERROR.enum_types_by_name['TaxonomyErrorCode']
_WARNING_GENEWARNINGCODE = _WARNING.enum_types_by_name['GeneWarningCode']
Error = _reflection.GeneratedProtocolMessageType('Error', (_message.Message,), {
  'DESCRIPTOR' : _ERROR,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.Error)
  })
_sym_db.RegisterMessage(Error)

Warning = _reflection.GeneratedProtocolMessageType('Warning', (_message.Message,), {

  'ReplacedId' : _reflection.GeneratedProtocolMessageType('ReplacedId', (_message.Message,), {
    'DESCRIPTOR' : _WARNING_REPLACEDID,
    '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.Warning.ReplacedId)
    })
  ,
  'DESCRIPTOR' : _WARNING,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.Warning)
  })
_sym_db.RegisterMessage(Warning)
_sym_db.RegisterMessage(Warning.ReplacedId)

Message = _reflection.GeneratedProtocolMessageType('Message', (_message.Message,), {
  'DESCRIPTOR' : _MESSAGE,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.Message)
  })
_sym_db.RegisterMessage(Message)

LineageOrganism = _reflection.GeneratedProtocolMessageType('LineageOrganism', (_message.Message,), {
  'DESCRIPTOR' : _LINEAGEORGANISM,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.LineageOrganism)
  })
_sym_db.RegisterMessage(LineageOrganism)

Organism = _reflection.GeneratedProtocolMessageType('Organism', (_message.Message,), {
  'DESCRIPTOR' : _ORGANISM,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.Organism)
  })
_sym_db.RegisterMessage(Organism)

InfraspecificNames = _reflection.GeneratedProtocolMessageType('InfraspecificNames', (_message.Message,), {
  'DESCRIPTOR' : _INFRASPECIFICNAMES,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.InfraspecificNames)
  })
_sym_db.RegisterMessage(InfraspecificNames)

Range = _reflection.GeneratedProtocolMessageType('Range', (_message.Message,), {
  'DESCRIPTOR' : _RANGE,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.Range)
  })
_sym_db.RegisterMessage(Range)

SeqRangeSet = _reflection.GeneratedProtocolMessageType('SeqRangeSet', (_message.Message,), {
  'DESCRIPTOR' : _SEQRANGESET,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.SeqRangeSet)
  })
_sym_db.RegisterMessage(SeqRangeSet)

BioProject = _reflection.GeneratedProtocolMessageType('BioProject', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECT,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.BioProject)
  })
_sym_db.RegisterMessage(BioProject)

BioProjectLineage = _reflection.GeneratedProtocolMessageType('BioProjectLineage', (_message.Message,), {
  'DESCRIPTOR' : _BIOPROJECTLINEAGE,
  '__module__' : 'ncbi.datasets.v2.reports.common_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.BioProjectLineage)
  })
_sym_db.RegisterMessage(BioProjectLineage)

if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'Z\030ncbi/datasets/v2/reports\370\001\001\302\363\030$\n\030Common Report Structures\022\010Organism'
  _LINEAGEORGANISM.fields_by_name['tax_id']._options = None
  _LINEAGEORGANISM.fields_by_name['tax_id']._serialized_options = b'\302\363\030\0072\00511118'
  _LINEAGEORGANISM.fields_by_name['name']._options = None
  _LINEAGEORGANISM.fields_by_name['name']._serialized_options = b'\302\363\030\0172\rCoronaviridae'
  _ORGANISM.fields_by_name['tax_id']._options = None
  _ORGANISM.fields_by_name['tax_id']._serialized_options = b'\302\363\030%\n\006tax-id\022\014Taxonomic ID2\00496062\0072697049'
  _ORGANISM.fields_by_name['sci_name']._options = None
  _ORGANISM.fields_by_name['sci_name']._serialized_options = b'\030\001'
  _ORGANISM.fields_by_name['organism_name']._options = None
  _ORGANISM.fields_by_name['organism_name']._serialized_options = b'\302\363\030K\n\004name\022\004Name2\014Homo sapiens2/Severe acute respiratory syndrome coronavirus 2\312\363\030%\022#The scientific name of the organism\322\363\0306\n\017scientific_name\022#The scientific name of the organism'
  _ORGANISM.fields_by_name['common_name']._options = None
  _ORGANISM.fields_by_name['common_name']._serialized_options = b'\302\363\0308\n\013common-name\022\013Common Name2\005human2\010pangolin2\004MERS2\005SARS2'
  _ORGANISM.fields_by_name['strain']._options = None
  _ORGANISM.fields_by_name['strain']._serialized_options = b'\030\001'
  _ORGANISM.fields_by_name['pangolin_classification']._options = None
  _ORGANISM.fields_by_name['pangolin_classification']._serialized_options = b'\302\363\030,\n\010pangolin\022\027Pangolin Classification2\007B.1.1.7'
  _ORGANISM.fields_by_name['infraspecific_names']._options = None
  _ORGANISM.fields_by_name['infraspecific_names']._serialized_options = b'\302\363\030&\n\016infraspecific-\022\024Infraspecific Names '
  _INFRASPECIFICNAMES.fields_by_name['breed']._options = None
  _INFRASPECIFICNAMES.fields_by_name['breed']._serialized_options = b'\302\363\030\037\n\005breed\022\005Breed2\010Hereford2\005boxer'
  _INFRASPECIFICNAMES.fields_by_name['cultivar']._options = None
  _INFRASPECIFICNAMES.fields_by_name['cultivar']._serialized_options = b'\302\363\030\031\n\010cultivar\022\010Cultivar2\003B73'
  _INFRASPECIFICNAMES.fields_by_name['ecotype']._options = None
  _INFRASPECIFICNAMES.fields_by_name['ecotype']._serialized_options = b'\302\363\030\032\n\007ecotype\022\007Ecotype2\006Alpine'
  _INFRASPECIFICNAMES.fields_by_name['isolate']._options = None
  _INFRASPECIFICNAMES.fields_by_name['isolate']._serialized_options = b'\302\363\030L\n\007isolate\022\007Isolate2/L1 Dominette 01449 registration number 421906802\007Pmale09'
  _INFRASPECIFICNAMES.fields_by_name['sex']._options = None
  _INFRASPECIFICNAMES.fields_by_name['sex']._serialized_options = b'\302\363\030\022\n\003sex\022\003Sex2\006female'
  _INFRASPECIFICNAMES.fields_by_name['strain']._options = None
  _INFRASPECIFICNAMES.fields_by_name['strain']._serialized_options = b'\302\363\030\026\n\006strain\022\006Strain2\004SE11'
  _RANGE.fields_by_name['begin']._options = None
  _RANGE.fields_by_name['begin']._serialized_options = b'\302\363\030\016\n\005start\022\005Start'
  _RANGE.fields_by_name['end']._options = None
  _RANGE.fields_by_name['end']._serialized_options = b'\302\363\030\014\n\004stop\022\004Stop'
  _RANGE.fields_by_name['orientation']._options = None
  _RANGE.fields_by_name['orientation']._serialized_options = b'\302\363\030\032\n\013orientation\022\013Orientation'
  _RANGE.fields_by_name['order']._options = None
  _RANGE.fields_by_name['order']._serialized_options = b'\302\363\030\016\n\005order\022\005Order'
  _SEQRANGESET.fields_by_name['accession_version']._options = None
  _SEQRANGESET.fields_by_name['accession_version']._serialized_options = b'\302\363\030\037\n\taccession\022\022Sequence Accession'
  _SEQRANGESET.fields_by_name['range']._options = None
  _SEQRANGESET.fields_by_name['range']._serialized_options = b'\302\363\030\010\n\006range-'
  _BIOPROJECT.fields_by_name['accession']._options = None
  _BIOPROJECT.fields_by_name['accession']._serialized_options = b'\302\363\030\"\n\taccession\022\tAccession2\nPRJEB35387'
  _BIOPROJECT.fields_by_name['title']._options = None
  _BIOPROJECT.fields_by_name['title']._serialized_options = b'\302\363\030N\n\005title\022\005Title2>Sciurus carolinensis (grey squirrel) genome assembly, mSciCar1'
  _BIOPROJECT.fields_by_name['parent_accession']._options = None
  _BIOPROJECT.fields_by_name['parent_accession']._serialized_options = b'\030\001\302\363\030$\n\020parent-accession\022\020Parent Accession'
  _BIOPROJECT.fields_by_name['parent_accessions']._options = None
  _BIOPROJECT.fields_by_name['parent_accessions']._serialized_options = b'\302\363\030Q\n\021parent-accessions\022\021Parent Accessions2)[\"PRJNA489243\",\"PRJEB33226\",\"PRJEB40665\"]'
  _BIOPROJECTLINEAGE.fields_by_name['bioprojects']._options = None
  _BIOPROJECTLINEAGE.fields_by_name['bioprojects']._serialized_options = b'\302\363\030\024\n\010lineage-\022\010Lineage '
  _ASSEMBLYLEVEL._serialized_start=4065
  _ASSEMBLYLEVEL._serialized_end=4143
  _ASSEMBLYSTATUS._serialized_start=4145
  _ASSEMBLYSTATUS._serialized_end=4246
  _ORIENTATION._serialized_start=4248
  _ORIENTATION._serialized_end=4292
  _RANKTYPE._serialized_start=4295
  _RANKTYPE._serialized_end=4964
  _COUNTTYPE._serialized_start=4967
  _COUNTTYPE._serialized_end=5344
  _ERROR._serialized_start=104
  _ERROR._serialized_end=1003
  _ERROR_ASSEMBLYERRORCODE._serialized_start=596
  _ERROR_ASSEMBLYERRORCODE._serialized_end=756
  _ERROR_GENEERRORCODE._serialized_start=758
  _ERROR_GENEERRORCODE._serialized_end=865
  _ERROR_VIRUSERRORCODE._serialized_start=867
  _ERROR_VIRUSERRORCODE._serialized_end=913
  _ERROR_TAXONOMYERRORCODE._serialized_start=915
  _ERROR_TAXONOMYERRORCODE._serialized_end=995
  _WARNING._serialized_start=1006
  _WARNING._serialized_end=1676
  _WARNING_REPLACEDID._serialized_start=1300
  _WARNING_REPLACEDID._serialized_end=1370
  _WARNING_GENEWARNINGCODE._serialized_start=1373
  _WARNING_GENEWARNINGCODE._serialized_end=1668
  _MESSAGE._serialized_start=1679
  _MESSAGE._serialized_end=1819
  _LINEAGEORGANISM._serialized_start=1821
  _LINEAGEORGANISM._serialized_end=1915
  _ORGANISM._serialized_start=1918
  _ORGANISM._serialized_end=2683
  _INFRASPECIFICNAMES._serialized_start=2686
  _INFRASPECIFICNAMES._serialized_end=3084
  _RANGE._serialized_start=3087
  _RANGE._serialized_end=3366
  _SEQRANGESET._serialized_start=3369
  _SEQRANGESET._serialized_end=3533
  _BIOPROJECT._serialized_start=3536
  _BIOPROJECT._serialized_end=3944
  _BIOPROJECTLINEAGE._serialized_start=3946
  _BIOPROJECTLINEAGE._serialized_end=4063
# @@protoc_insertion_point(module_scope)
