# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v2/reports/feature.proto
"""Generated protocol buffer code."""
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from ncbi.datasets.v2.reports import common_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2
from ncbi.datasets.v2.reports import gene_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__pb2
from ncbi.datasets.v2.reports import gene_product_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__product__pb2
from ncbi.datasets.options import report_pb2 as ncbi_dot_datasets_dot_options_dot_report__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n&ncbi/datasets/v2/reports/feature.proto\x12\x18ncbi.datasets.v2.reports\x1a%ncbi/datasets/v2/reports/common.proto\x1a#ncbi/datasets/v2/reports/gene.proto\x1a+ncbi/datasets/v2/reports/gene_product.proto\x1a\"ncbi/datasets/options/report.proto\"\x87\x03\n\x1aGenomeAnnotationReportPage\x12O\n\x07reports\x18\x01 \x03(\x0b\x32\x35.ncbi.datasets.v2.reports.GenomeAnnotationReportMatchR\x07reports\x12=\n\x08messages\x18\x02 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\x12&\n\x0fnext_page_token\x18\x04 \x01(\tR\rnextPageToken\x12!\n\x0c_report_type\x18\xf4\x03 \x01(\tR\nReportType\x12%\n\x0e_report_fields\x18\xf5\x03 \x03(\tR\x0cReportFields\x12\x1f\n\x0b_first_page\x18\xf6\x03 \x01(\x08R\tFirstPage\x12%\n\x0e_report_format\x18\xf7\x03 \x01(\tR\x0cReportFormat\"\x8c\x02\n\x1bGenomeAnnotationReportMatch\x12J\n\nannotation\x18\x01 \x01(\x0b\x32*.ncbi.datasets.v2.reports.GenomeAnnotationR\nannotation\x12\x14\n\x05query\x18\x02 \x03(\tR\x05query\x12;\n\x07warning\x18\x03 \x01(\x0b\x32!.ncbi.datasets.v2.reports.WarningR\x07warning\x12\x37\n\x06\x65rrors\x18\x04 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorR\x06\x65rrors\x12\x15\n\x06row_id\x18\x05 \x01(\tR\x05rowId\"i\n\x19GenomeAnnotationContainer\x12L\n\x0b\x61nnotations\x18\x01 \x03(\x0b\x32*.ncbi.datasets.v2.reports.GenomeAnnotationR\x0b\x61nnotations\"\xf4\x0c\n\x10GenomeAnnotation\x12\x39\n\x07gene_id\x18\x01 \x01(\x04\x42 \xc2\xf3\x18\x1c\n\x07gene-id\x12\x0bNCBI GeneID2\x04\x32\x37\x37\x38R\x06geneId\x12\x32\n\x06symbol\x18\x02 \x01(\tB\x1a\xc2\xf3\x18\x16\n\x06symbol\x12\x06Symbol2\x04GNASR\x06symbol\x12$\n\x0b\x64\x65scription\x18\x03 \x01(\tB\x02\x18\x01R\x0b\x64\x65scription\x12\x38\n\x04name\x18\x18 \x01(\tB$\xc2\xf3\x18 \n\x04name\x12\x04Name2\x12GNAS complex locusR\x04name\x12\x37\n\x06tax_id\x18\x04 \x01(\x04\x42 \xc2\xf3\x18\x1c\n\x06tax-id\x12\x0cTaxonomic ID2\x04\x39\x36\x30\x36R\x05taxId\x12\x46\n\x07taxname\x18\x05 \x01(\tB,\xc2\xf3\x18(\n\x08tax-name\x12\x0eTaxonomic Name2\x0cHomo sapiensR\x07taxname\x12\x46\n\x0b\x63ommon_name\x18\x0c \x01(\tB%\xc2\xf3\x18!\n\x0b\x63ommon-name\x12\x0b\x43ommon Name2\x05humanR\ncommonName\x12:\n\x04type\x18\x06 \x01(\x0e\x32\".ncbi.datasets.v2.reports.GeneTypeB\x02\x18\x01R\x04type\x12\x37\n\tgene_type\x18\x19 \x01(\tB\x1a\xc2\xf3\x18\x16\n\tgene-type\x12\tGene TypeR\x08geneType\x12V\n\x08rna_type\x18\x0b \x01(\x0e\x32!.ncbi.datasets.v2.reports.RnaTypeB\x18\xc2\xf3\x18\x14\n\x08rna-type\x12\x08RNA TypeR\x07rnaType\x12g\n\x0borientation\x18\x07 \x01(\x0e\x32%.ncbi.datasets.v2.reports.OrientationB\x1e\xc2\xf3\x18\x1a\n\x0borientation\x12\x0bOrientationR\x0borientation\x12\x37\n\tlocus_tag\x18\x1a \x01(\tB\x1a\xc2\xf3\x18\x16\n\tlocus-tag\x12\tLocus TagR\x08locusTag\x12\x82\x01\n\x13reference_standards\x18\x15 \x03(\x0b\x32\'.ncbi.datasets.v2.reports.GenomicRegionB(\xc2\xf3\x18$\n\rref-standard-\x12\x13Reference Standard R\x12referenceStandards\x12x\n\x0fgenomic_regions\x18\x16 \x03(\x0b\x32\'.ncbi.datasets.v2.reports.GenomicRegionB&\xc2\xf3\x18\"\n\x0fgenomic-region-\x12\x0fGenomic Region R\x0egenomicRegions\x12\x46\n\x0btranscripts\x18\t \x03(\x0b\x32$.ncbi.datasets.v2.reports.TranscriptR\x0btranscripts\x12=\n\x08proteins\x18\n \x03(\x0b\x32!.ncbi.datasets.v2.reports.ProteinR\x08proteins\x12H\n\x0b\x63hromosomes\x18\x0e \x03(\tB&\xc2\xf3\x18\"\n\x0b\x63hromosomes\x12\x0b\x43hromosomes2\x01\x31\x32\x03X,YR\x0b\x63hromosomes\x12\x64\n\x15swiss_prot_accessions\x18\x10 \x03(\tB0\xc2\xf3\x18,\n\x14swissprot-accessions\x12\x14SwissProt AccessionsR\x13swissProtAccessions\x12P\n\x10\x65nsembl_gene_ids\x18\x11 \x03(\tB&\xc2\xf3\x18\"\n\x0f\x65nsembl-geneids\x12\x0f\x45nsembl GeneIDsR\x0e\x65nsemblGeneIds\x12\x33\n\x08omim_ids\x18\x12 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08omim-ids\x12\x08OMIM IDsR\x07omimIds\x12\x34\n\x08synonyms\x18\x13 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08synonyms\x12\x08SynonymsR\x08synonyms\x12\x66\n\x0b\x61nnotations\x18\x17 \x03(\x0b\x32$.ncbi.datasets.v2.reports.AnnotationB\x1e\xc2\xf3\x18\x1a\n\x0b\x61nnotation-\x12\x0b\x41nnotation R\x0b\x61nnotationsB\xf4\nZ\x18ncbi/datasets/v2/reports\xf8\x01\x01\xc2\xf3\x18\xd2\n\n\x18Genome Annotation Report\x12\x10GenomeAnnotation\x1aH<p>The downloaded genome annotation package contains a genome annotation\x1a\x44report in <a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\x1aVformat in the file: <br><br><em>ncbi_dataset/data/annotation_report.jsonl</em><br><br>\x1auEach line of the genome annotation report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\x1a}object that represents a single genome annotation record. The schema of the genome annotation record is defined in the tables\x1apbelow where each row describes a single field in the report or a sub-structure, which is a collection of fields.\x1aHThe outermost structure of the report is <em>GenomeAnnotation</em>.</p> \x1aQ<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\x1a\x90\x01<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option\x1a\x8e\x01Refer to the <a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this\x1aMtool to transform annotation reports from JSON Lines to tabular formats.</p> \"\x0fLineageOrganism\"\x08Organism*\xa8\x01{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/genome-annotation-report.cli.output\" begin=\"START annotation-report\" end=\"END annotation-report\">}}b\x06proto3')



_GENOMEANNOTATIONREPORTPAGE = DESCRIPTOR.message_types_by_name['GenomeAnnotationReportPage']
_GENOMEANNOTATIONREPORTMATCH = DESCRIPTOR.message_types_by_name['GenomeAnnotationReportMatch']
_GENOMEANNOTATIONCONTAINER = DESCRIPTOR.message_types_by_name['GenomeAnnotationContainer']
_GENOMEANNOTATION = DESCRIPTOR.message_types_by_name['GenomeAnnotation']
GenomeAnnotationReportPage = _reflection.GeneratedProtocolMessageType('GenomeAnnotationReportPage', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATIONREPORTPAGE,
  '__module__' : 'ncbi.datasets.v2.reports.feature_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GenomeAnnotationReportPage)
  })
_sym_db.RegisterMessage(GenomeAnnotationReportPage)

GenomeAnnotationReportMatch = _reflection.GeneratedProtocolMessageType('GenomeAnnotationReportMatch', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATIONREPORTMATCH,
  '__module__' : 'ncbi.datasets.v2.reports.feature_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GenomeAnnotationReportMatch)
  })
_sym_db.RegisterMessage(GenomeAnnotationReportMatch)

GenomeAnnotationContainer = _reflection.GeneratedProtocolMessageType('GenomeAnnotationContainer', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATIONCONTAINER,
  '__module__' : 'ncbi.datasets.v2.reports.feature_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GenomeAnnotationContainer)
  })
_sym_db.RegisterMessage(GenomeAnnotationContainer)

GenomeAnnotation = _reflection.GeneratedProtocolMessageType('GenomeAnnotation', (_message.Message,), {
  'DESCRIPTOR' : _GENOMEANNOTATION,
  '__module__' : 'ncbi.datasets.v2.reports.feature_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GenomeAnnotation)
  })
_sym_db.RegisterMessage(GenomeAnnotation)

if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'Z\030ncbi/datasets/v2/reports\370\001\001\302\363\030\322\n\n\030Genome Annotation Report\022\020GenomeAnnotation\032H<p>The downloaded genome annotation package contains a genome annotation\032Dreport in <a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\032Vformat in the file: <br><br><em>ncbi_dataset/data/annotation_report.jsonl</em><br><br>\032uEach line of the genome annotation report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\032}object that represents a single genome annotation record. The schema of the genome annotation record is defined in the tables\032pbelow where each row describes a single field in the report or a sub-structure, which is a collection of fields.\032HThe outermost structure of the report is <em>GenomeAnnotation</em>.</p> \032Q<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\032\220\001<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option\032\216\001Refer to the <a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this\032Mtool to transform annotation reports from JSON Lines to tabular formats.</p> \"\017LineageOrganism\"\010Organism*\250\001{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/genome-annotation-report.cli.output\" begin=\"START annotation-report\" end=\"END annotation-report\">}}'
  _GENOMEANNOTATION.fields_by_name['gene_id']._options = None
  _GENOMEANNOTATION.fields_by_name['gene_id']._serialized_options = b'\302\363\030\034\n\007gene-id\022\013NCBI GeneID2\0042778'
  _GENOMEANNOTATION.fields_by_name['symbol']._options = None
  _GENOMEANNOTATION.fields_by_name['symbol']._serialized_options = b'\302\363\030\026\n\006symbol\022\006Symbol2\004GNAS'
  _GENOMEANNOTATION.fields_by_name['description']._options = None
  _GENOMEANNOTATION.fields_by_name['description']._serialized_options = b'\030\001'
  _GENOMEANNOTATION.fields_by_name['name']._options = None
  _GENOMEANNOTATION.fields_by_name['name']._serialized_options = b'\302\363\030 \n\004name\022\004Name2\022GNAS complex locus'
  _GENOMEANNOTATION.fields_by_name['tax_id']._options = None
  _GENOMEANNOTATION.fields_by_name['tax_id']._serialized_options = b'\302\363\030\034\n\006tax-id\022\014Taxonomic ID2\0049606'
  _GENOMEANNOTATION.fields_by_name['taxname']._options = None
  _GENOMEANNOTATION.fields_by_name['taxname']._serialized_options = b'\302\363\030(\n\010tax-name\022\016Taxonomic Name2\014Homo sapiens'
  _GENOMEANNOTATION.fields_by_name['common_name']._options = None
  _GENOMEANNOTATION.fields_by_name['common_name']._serialized_options = b'\302\363\030!\n\013common-name\022\013Common Name2\005human'
  _GENOMEANNOTATION.fields_by_name['type']._options = None
  _GENOMEANNOTATION.fields_by_name['type']._serialized_options = b'\030\001'
  _GENOMEANNOTATION.fields_by_name['gene_type']._options = None
  _GENOMEANNOTATION.fields_by_name['gene_type']._serialized_options = b'\302\363\030\026\n\tgene-type\022\tGene Type'
  _GENOMEANNOTATION.fields_by_name['rna_type']._options = None
  _GENOMEANNOTATION.fields_by_name['rna_type']._serialized_options = b'\302\363\030\024\n\010rna-type\022\010RNA Type'
  _GENOMEANNOTATION.fields_by_name['orientation']._options = None
  _GENOMEANNOTATION.fields_by_name['orientation']._serialized_options = b'\302\363\030\032\n\013orientation\022\013Orientation'
  _GENOMEANNOTATION.fields_by_name['locus_tag']._options = None
  _GENOMEANNOTATION.fields_by_name['locus_tag']._serialized_options = b'\302\363\030\026\n\tlocus-tag\022\tLocus Tag'
  _GENOMEANNOTATION.fields_by_name['reference_standards']._options = None
  _GENOMEANNOTATION.fields_by_name['reference_standards']._serialized_options = b'\302\363\030$\n\rref-standard-\022\023Reference Standard '
  _GENOMEANNOTATION.fields_by_name['genomic_regions']._options = None
  _GENOMEANNOTATION.fields_by_name['genomic_regions']._serialized_options = b'\302\363\030\"\n\017genomic-region-\022\017Genomic Region '
  _GENOMEANNOTATION.fields_by_name['chromosomes']._options = None
  _GENOMEANNOTATION.fields_by_name['chromosomes']._serialized_options = b'\302\363\030\"\n\013chromosomes\022\013Chromosomes2\00112\003X,Y'
  _GENOMEANNOTATION.fields_by_name['swiss_prot_accessions']._options = None
  _GENOMEANNOTATION.fields_by_name['swiss_prot_accessions']._serialized_options = b'\302\363\030,\n\024swissprot-accessions\022\024SwissProt Accessions'
  _GENOMEANNOTATION.fields_by_name['ensembl_gene_ids']._options = None
  _GENOMEANNOTATION.fields_by_name['ensembl_gene_ids']._serialized_options = b'\302\363\030\"\n\017ensembl-geneids\022\017Ensembl GeneIDs'
  _GENOMEANNOTATION.fields_by_name['omim_ids']._options = None
  _GENOMEANNOTATION.fields_by_name['omim_ids']._serialized_options = b'\302\363\030\024\n\010omim-ids\022\010OMIM IDs'
  _GENOMEANNOTATION.fields_by_name['synonyms']._options = None
  _GENOMEANNOTATION.fields_by_name['synonyms']._serialized_options = b'\302\363\030\024\n\010synonyms\022\010Synonyms'
  _GENOMEANNOTATION.fields_by_name['annotations']._options = None
  _GENOMEANNOTATION.fields_by_name['annotations']._serialized_options = b'\302\363\030\032\n\013annotation-\022\013Annotation '
  _GENOMEANNOTATIONREPORTPAGE._serialized_start=226
  _GENOMEANNOTATIONREPORTPAGE._serialized_end=617
  _GENOMEANNOTATIONREPORTMATCH._serialized_start=620
  _GENOMEANNOTATIONREPORTMATCH._serialized_end=888
  _GENOMEANNOTATIONCONTAINER._serialized_start=890
  _GENOMEANNOTATIONCONTAINER._serialized_end=995
  _GENOMEANNOTATION._serialized_start=998
  _GENOMEANNOTATION._serialized_end=2650
# @@protoc_insertion_point(module_scope)
