# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v2/reports/gene.proto
"""Generated protocol buffer code."""
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from ncbi.datasets.v2.reports import common_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2
from ncbi.datasets.v2.reports import gene_product_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_gene__product__pb2
from ncbi.datasets.options import report_pb2 as ncbi_dot_datasets_dot_options_dot_report__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n#ncbi/datasets/v2/reports/gene.proto\x12\x18ncbi.datasets.v2.reports\x1a%ncbi/datasets/v2/reports/common.proto\x1a+ncbi/datasets/v2/reports/gene_product.proto\x1a\"ncbi/datasets/options/report.proto\"\xf3\x02\n\x12GeneDataReportPage\x12\x43\n\x07reports\x18\x01 \x03(\x0b\x32).ncbi.datasets.v2.reports.GeneReportMatchR\x07reports\x12=\n\x08messages\x18\x02 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\x12&\n\x0fnext_page_token\x18\x04 \x01(\tR\rnextPageToken\x12!\n\x0c_report_type\x18\xf4\x03 \x01(\tR\nReportType\x12%\n\x0e_report_fields\x18\xf5\x03 \x03(\tR\x0cReportFields\x12\x1f\n\x0b_first_page\x18\xf6\x03 \x01(\x08R\tFirstPage\x12%\n\x0e_report_format\x18\xf7\x03 \x01(\tR\x0cReportFormat\"\x91\x01\n\x0bOrthologSet\x12&\n\x0fortholog_set_id\x18\x01 \x01(\rR\rorthologSetId\x12\x16\n\x06method\x18\x02 \x01(\tR\x06method\x12\x42\n\x05genes\x18\x03 \x01(\x0b\x32,.ncbi.datasets.v2.reports.GeneReportMetadataR\x05genes\"\xb5\x01\n\x12GeneReportMetadata\x12=\n\x08messages\x18\x01 \x03(\x0b\x32!.ncbi.datasets.v2.reports.MessageR\x08messages\x12?\n\x05genes\x18\x02 \x03(\x0b\x32).ncbi.datasets.v2.reports.GeneReportMatchR\x05genes\x12\x1f\n\x0btotal_count\x18\x03 \x01(\rR\ntotalCount\"\xef\x02\n\x0fGeneReportMatch\x12>\n\x04gene\x18\x01 \x01(\x0b\x32(.ncbi.datasets.v2.reports.GeneDescriptorH\x00R\x04gene\x12G\n\x07product\x18\x05 \x01(\x0b\x32+.ncbi.datasets.v2.reports.ProductDescriptorH\x00R\x07product\x12\x14\n\x05query\x18\x02 \x03(\tR\x05query\x12=\n\x08warnings\x18\x06 \x03(\x0b\x32!.ncbi.datasets.v2.reports.WarningR\x08warnings\x12;\n\x07warning\x18\x03 \x01(\x0b\x32!.ncbi.datasets.v2.reports.WarningR\x07warning\x12\x37\n\x06\x65rrors\x18\x04 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.ErrorR\x06\x65rrorsB\x08\n\x06report\"\xd9\x02\n\rGenomicRegion\x12\x64\n\ngene_range\x18\x01 \x01(\x0b\x32%.ncbi.datasets.v2.reports.SeqRangeSetB\x1e\xc2\xf3\x18\x1a\n\x0bgene-range-\x12\x0bGene Range R\tgeneRange\x12}\n\x04type\x18\x02 \x01(\x0e\x32\x39.ncbi.datasets.v2.reports.GenomicRegion.GenomicRegionTypeB.\xc2\xf3\x18*\n\x13genomic-region-type\x12\x13Genomic Region TypeR\x04type\"c\n\x11GenomicRegionType\x12\x0b\n\x07UNKNOWN\x10\x00\x12\x0f\n\x0bREFSEQ_GENE\x10\x01\x12\x0e\n\nPSEUDOGENE\x10\x02\x12\x15\n\x11\x42IOLOGICAL_REGION\x10\x03\x12\t\n\x05OTHER\x10\x04\"\xd5\x03\n\nAnnotation\x12[\n\x12\x61ssembly_accession\x18\x04 \x01(\tB,\xc2\xf3\x18(\n\x12\x61ssembly-accession\x12\x12\x41ssembly AccessionR\x11\x61ssemblyAccession\x12G\n\rassembly_name\x18\x05 \x01(\tB\"\xc2\xf3\x18\x1e\n\rassembly-name\x12\rAssembly NameR\x0c\x61ssemblyName\x12I\n\x0f\x61nnotation_name\x18\x06 \x01(\tB \xc2\xf3\x18\x1c\n\x0crelease-name\x12\x0cRelease NameR\x0e\x61nnotationName\x12X\n\x17\x61nnotation_release_date\x18\x07 \x01(\tB \xc2\xf3\x18\x1c\n\x0crelease-date\x12\x0cRelease DateR\x15\x61nnotationReleaseDate\x12|\n\x11genomic_locations\x18\x08 \x03(\x0b\x32).ncbi.datasets.v2.reports.GenomicLocationB$\xc2\xf3\x18 \n\x0egenomic-range-\x12\x0eGenomic Range R\x10genomicLocations\"\xbe\x0f\n\x0eGeneDescriptor\x12\x39\n\x07gene_id\x18\x01 \x01(\x04\x42 \xc2\xf3\x18\x1c\n\x07gene-id\x12\x0bNCBI GeneID2\x04\x32\x37\x37\x38R\x06geneId\x12\x32\n\x06symbol\x18\x02 \x01(\tB\x1a\xc2\xf3\x18\x16\n\x06symbol\x12\x06Symbol2\x04GNASR\x06symbol\x12T\n\x0b\x64\x65scription\x18\x03 \x01(\tB2\xc2\xf3\x18.\n\x0b\x64\x65scription\x12\x0b\x44\x65scription2\x12GNAS complex locusR\x0b\x64\x65scription\x12\x37\n\x06tax_id\x18\x04 \x01(\x04\x42 \xc2\xf3\x18\x1c\n\x06tax-id\x12\x0cTaxonomic ID2\x04\x39\x36\x30\x36R\x05taxId\x12\x46\n\x07taxname\x18\x05 \x01(\tB,\xc2\xf3\x18(\n\x08tax-name\x12\x0eTaxonomic Name2\x0cHomo sapiensR\x07taxname\x12\x46\n\x0b\x63ommon_name\x18\x0c \x01(\tB%\xc2\xf3\x18!\n\x0b\x63ommon-name\x12\x0b\x43ommon Name2\x05humanR\ncommonName\x12R\n\x04type\x18\x06 \x01(\x0e\x32\".ncbi.datasets.v2.reports.GeneTypeB\x1a\xc2\xf3\x18\x16\n\tgene-type\x12\tGene TypeR\x04type\x12V\n\x08rna_type\x18\x0b \x01(\x0e\x32!.ncbi.datasets.v2.reports.RnaTypeB\x18\xc2\xf3\x18\x14\n\x08rna-type\x12\x08RNA TypeR\x07rnaType\x12g\n\x0borientation\x18\x07 \x01(\x0e\x32%.ncbi.datasets.v2.reports.OrientationB\x1e\xc2\xf3\x18\x1a\n\x0borientation\x12\x0bOrientationR\x0borientation\x12\x82\x01\n\x13reference_standards\x18\x15 \x03(\x0b\x32\'.ncbi.datasets.v2.reports.GenomicRegionB(\xc2\xf3\x18$\n\rref-standard-\x12\x13Reference Standard R\x12referenceStandards\x12x\n\x0fgenomic_regions\x18\x16 \x03(\x0b\x32\'.ncbi.datasets.v2.reports.GenomicRegionB&\xc2\xf3\x18\"\n\x0fgenomic-region-\x12\x0fGenomic Region R\x0egenomicRegions\x12H\n\x0b\x63hromosomes\x18\x0e \x03(\tB&\xc2\xf3\x18\"\n\x0b\x63hromosomes\x12\x0b\x43hromosomes2\x01\x31\x32\x03X,YR\x0b\x63hromosomes\x12\x82\x01\n\x16nomenclature_authority\x18\x0f \x01(\x0b\x32/.ncbi.datasets.v2.reports.NomenclatureAuthorityB\x1a\xc2\xf3\x18\x16\n\x05name-\x12\rNomenclature R\x15nomenclatureAuthority\x12\x64\n\x15swiss_prot_accessions\x18\x10 \x03(\tB0\xc2\xf3\x18,\n\x14swissprot-accessions\x12\x14SwissProt AccessionsR\x13swissProtAccessions\x12P\n\x10\x65nsembl_gene_ids\x18\x11 \x03(\tB&\xc2\xf3\x18\"\n\x0f\x65nsembl-geneids\x12\x0f\x45nsembl GeneIDsR\x0e\x65nsemblGeneIds\x12\x33\n\x08omim_ids\x18\x12 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08omim-ids\x12\x08OMIM IDsR\x07omimIds\x12\x34\n\x08synonyms\x18\x13 \x03(\tB\x18\xc2\xf3\x18\x14\n\x08synonyms\x12\x08SynonymsR\x08synonyms\x12V\n\x10replaced_gene_id\x18\x14 \x01(\x04\x42,\xc2\xf3\x18(\n\x10replaced-gene-id\x12\x14Replaced NCBI GeneIDR\x0ereplacedGeneId\x12\x66\n\x0b\x61nnotations\x18\x17 \x03(\x0b\x32$.ncbi.datasets.v2.reports.AnnotationB\x1e\xc2\xf3\x18\x1a\n\x0b\x61nnotation-\x12\x0b\x41nnotation R\x0b\x61nnotations\x12N\n\x10transcript_count\x18\x1a \x01(\rB#\xc2\xf3\x18\x1f\n\x10transcript-count\x12\x0bTranscriptsR\x0ftranscriptCount\x12\x42\n\rprotein_count\x18\x1b \x01(\rB\x1d\xc2\xf3\x18\x19\n\rprotein-count\x12\x08ProteinsR\x0cproteinCount\x12\x63\n\x16transcript_type_counts\x18\x1c \x03(\x0b\x32-.ncbi.datasets.v2.reports.TranscriptTypeCountR\x14transcriptTypeCounts\x12_\n\x0bgene_groups\x18\x1d \x03(\x0b\x32#.ncbi.datasets.v2.reports.GeneGroupB\x19\xc2\xf3\x18\x15\n\x06group-\x12\x0bGene Group R\ngeneGroups\"_\n\tGeneGroup\x12$\n\x02id\x18\x01 \x01(\tB\x14\xc2\xf3\x18\x10\n\x02id\x12\nIdentifierR\x02id\x12,\n\x06method\x18\x02 \x01(\tB\x14\xc2\xf3\x18\x10\n\x06method\x12\x06MethodR\x06method\"\x90\x01\n\x15NomenclatureAuthority\x12>\n\tauthority\x18\x01 \x01(\tB \xc2\xf3\x18\x1c\n\tauthority\x12\tAuthority2\x04HGNCR\tauthority\x12\x37\n\nidentifier\x18\x02 \x01(\tB\x17\xc2\xf3\x18\x13\n\x02id\x12\x02ID2\tHGNC:4392R\nidentifier\"Q\n\x0fGeneDescriptors\x12>\n\x05genes\x18\x01 \x03(\x0b\x32(.ncbi.datasets.v2.reports.GeneDescriptorR\x05genesB\x9d\nZ\x18ncbi/datasets/v2/reports\xf8\x01\x01\xc2\xf3\x18\xfb\t\n\x0bGene Report\x12\x0eGeneDescriptor\x1a=<p>The downloaded gene package contains a gene data report in\x1a:<a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\x1aOformat in the file:<br><br><em>ncbi_dataset/data/data_report.jsonl</em><br><br>\x1amEach line of the gene data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\x1aiobject that represents a single gene record. The schema of the gene record is defined in the tables below\x1ajwhere each row describes a single field in the report or a sub-structure, which is a collection of fields.\x1a\x45The outermost structure of the report is <em>GeneDescriptor</em>.</p>\x1aQ<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\x1a\x91\x01<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option.\x1a\x93\x01Refer to the <a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this tool\x1a\x46to transform gene data reports from JSON Lines to tabular formats.</p>\"\x0fLineageOrganism\"\x08Organism\"\x0fGeneDescriptors*\x91\x01{{< readfile-code-block lang=\"json\" file=\"/docs_samples/dataformat-cli/gene-to-tsv.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}b\x06proto3')



_GENEDATAREPORTPAGE = DESCRIPTOR.message_types_by_name['GeneDataReportPage']
_ORTHOLOGSET = DESCRIPTOR.message_types_by_name['OrthologSet']
_GENEREPORTMETADATA = DESCRIPTOR.message_types_by_name['GeneReportMetadata']
_GENEREPORTMATCH = DESCRIPTOR.message_types_by_name['GeneReportMatch']
_GENOMICREGION = DESCRIPTOR.message_types_by_name['GenomicRegion']
_ANNOTATION = DESCRIPTOR.message_types_by_name['Annotation']
_GENEDESCRIPTOR = DESCRIPTOR.message_types_by_name['GeneDescriptor']
_GENEGROUP = DESCRIPTOR.message_types_by_name['GeneGroup']
_NOMENCLATUREAUTHORITY = DESCRIPTOR.message_types_by_name['NomenclatureAuthority']
_GENEDESCRIPTORS = DESCRIPTOR.message_types_by_name['GeneDescriptors']
_GENOMICREGION_GENOMICREGIONTYPE = _GENOMICREGION.enum_types_by_name['GenomicRegionType']
GeneDataReportPage = _reflection.GeneratedProtocolMessageType('GeneDataReportPage', (_message.Message,), {
  'DESCRIPTOR' : _GENEDATAREPORTPAGE,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GeneDataReportPage)
  })
_sym_db.RegisterMessage(GeneDataReportPage)

OrthologSet = _reflection.GeneratedProtocolMessageType('OrthologSet', (_message.Message,), {
  'DESCRIPTOR' : _ORTHOLOGSET,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.OrthologSet)
  })
_sym_db.RegisterMessage(OrthologSet)

GeneReportMetadata = _reflection.GeneratedProtocolMessageType('GeneReportMetadata', (_message.Message,), {
  'DESCRIPTOR' : _GENEREPORTMETADATA,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GeneReportMetadata)
  })
_sym_db.RegisterMessage(GeneReportMetadata)

GeneReportMatch = _reflection.GeneratedProtocolMessageType('GeneReportMatch', (_message.Message,), {
  'DESCRIPTOR' : _GENEREPORTMATCH,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GeneReportMatch)
  })
_sym_db.RegisterMessage(GeneReportMatch)

GenomicRegion = _reflection.GeneratedProtocolMessageType('GenomicRegion', (_message.Message,), {
  'DESCRIPTOR' : _GENOMICREGION,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GenomicRegion)
  })
_sym_db.RegisterMessage(GenomicRegion)

Annotation = _reflection.GeneratedProtocolMessageType('Annotation', (_message.Message,), {
  'DESCRIPTOR' : _ANNOTATION,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.Annotation)
  })
_sym_db.RegisterMessage(Annotation)

GeneDescriptor = _reflection.GeneratedProtocolMessageType('GeneDescriptor', (_message.Message,), {
  'DESCRIPTOR' : _GENEDESCRIPTOR,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GeneDescriptor)
  })
_sym_db.RegisterMessage(GeneDescriptor)

GeneGroup = _reflection.GeneratedProtocolMessageType('GeneGroup', (_message.Message,), {
  'DESCRIPTOR' : _GENEGROUP,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GeneGroup)
  })
_sym_db.RegisterMessage(GeneGroup)

NomenclatureAuthority = _reflection.GeneratedProtocolMessageType('NomenclatureAuthority', (_message.Message,), {
  'DESCRIPTOR' : _NOMENCLATUREAUTHORITY,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.NomenclatureAuthority)
  })
_sym_db.RegisterMessage(NomenclatureAuthority)

GeneDescriptors = _reflection.GeneratedProtocolMessageType('GeneDescriptors', (_message.Message,), {
  'DESCRIPTOR' : _GENEDESCRIPTORS,
  '__module__' : 'ncbi.datasets.v2.reports.gene_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.GeneDescriptors)
  })
_sym_db.RegisterMessage(GeneDescriptors)

if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'Z\030ncbi/datasets/v2/reports\370\001\001\302\363\030\373\t\n\013Gene Report\022\016GeneDescriptor\032=<p>The downloaded gene package contains a gene data report in\032:<a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\032Oformat in the file:<br><br><em>ncbi_dataset/data/data_report.jsonl</em><br><br>\032mEach line of the gene data report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\032iobject that represents a single gene record. The schema of the gene record is defined in the tables below\032jwhere each row describes a single field in the report or a sub-structure, which is a collection of fields.\032EThe outermost structure of the report is <em>GeneDescriptor</em>.</p>\032Q<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\032\221\001<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s <nobr><code>--fields</code></nobr> option.\032\223\001Refer to the <a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see how you can use this tool\032Fto transform gene data reports from JSON Lines to tabular formats.</p>\"\017LineageOrganism\"\010Organism\"\017GeneDescriptors*\221\001{{< readfile-code-block lang=\"json\" file=\"/docs_samples/dataformat-cli/gene-to-tsv.cli.output\" begin=\"START data-report\" end=\"END data-report\">}}'
  _GENOMICREGION.fields_by_name['gene_range']._options = None
  _GENOMICREGION.fields_by_name['gene_range']._serialized_options = b'\302\363\030\032\n\013gene-range-\022\013Gene Range '
  _GENOMICREGION.fields_by_name['type']._options = None
  _GENOMICREGION.fields_by_name['type']._serialized_options = b'\302\363\030*\n\023genomic-region-type\022\023Genomic Region Type'
  _ANNOTATION.fields_by_name['assembly_accession']._options = None
  _ANNOTATION.fields_by_name['assembly_accession']._serialized_options = b'\302\363\030(\n\022assembly-accession\022\022Assembly Accession'
  _ANNOTATION.fields_by_name['assembly_name']._options = None
  _ANNOTATION.fields_by_name['assembly_name']._serialized_options = b'\302\363\030\036\n\rassembly-name\022\rAssembly Name'
  _ANNOTATION.fields_by_name['annotation_name']._options = None
  _ANNOTATION.fields_by_name['annotation_name']._serialized_options = b'\302\363\030\034\n\014release-name\022\014Release Name'
  _ANNOTATION.fields_by_name['annotation_release_date']._options = None
  _ANNOTATION.fields_by_name['annotation_release_date']._serialized_options = b'\302\363\030\034\n\014release-date\022\014Release Date'
  _ANNOTATION.fields_by_name['genomic_locations']._options = None
  _ANNOTATION.fields_by_name['genomic_locations']._serialized_options = b'\302\363\030 \n\016genomic-range-\022\016Genomic Range '
  _GENEDESCRIPTOR.fields_by_name['gene_id']._options = None
  _GENEDESCRIPTOR.fields_by_name['gene_id']._serialized_options = b'\302\363\030\034\n\007gene-id\022\013NCBI GeneID2\0042778'
  _GENEDESCRIPTOR.fields_by_name['symbol']._options = None
  _GENEDESCRIPTOR.fields_by_name['symbol']._serialized_options = b'\302\363\030\026\n\006symbol\022\006Symbol2\004GNAS'
  _GENEDESCRIPTOR.fields_by_name['description']._options = None
  _GENEDESCRIPTOR.fields_by_name['description']._serialized_options = b'\302\363\030.\n\013description\022\013Description2\022GNAS complex locus'
  _GENEDESCRIPTOR.fields_by_name['tax_id']._options = None
  _GENEDESCRIPTOR.fields_by_name['tax_id']._serialized_options = b'\302\363\030\034\n\006tax-id\022\014Taxonomic ID2\0049606'
  _GENEDESCRIPTOR.fields_by_name['taxname']._options = None
  _GENEDESCRIPTOR.fields_by_name['taxname']._serialized_options = b'\302\363\030(\n\010tax-name\022\016Taxonomic Name2\014Homo sapiens'
  _GENEDESCRIPTOR.fields_by_name['common_name']._options = None
  _GENEDESCRIPTOR.fields_by_name['common_name']._serialized_options = b'\302\363\030!\n\013common-name\022\013Common Name2\005human'
  _GENEDESCRIPTOR.fields_by_name['type']._options = None
  _GENEDESCRIPTOR.fields_by_name['type']._serialized_options = b'\302\363\030\026\n\tgene-type\022\tGene Type'
  _GENEDESCRIPTOR.fields_by_name['rna_type']._options = None
  _GENEDESCRIPTOR.fields_by_name['rna_type']._serialized_options = b'\302\363\030\024\n\010rna-type\022\010RNA Type'
  _GENEDESCRIPTOR.fields_by_name['orientation']._options = None
  _GENEDESCRIPTOR.fields_by_name['orientation']._serialized_options = b'\302\363\030\032\n\013orientation\022\013Orientation'
  _GENEDESCRIPTOR.fields_by_name['reference_standards']._options = None
  _GENEDESCRIPTOR.fields_by_name['reference_standards']._serialized_options = b'\302\363\030$\n\rref-standard-\022\023Reference Standard '
  _GENEDESCRIPTOR.fields_by_name['genomic_regions']._options = None
  _GENEDESCRIPTOR.fields_by_name['genomic_regions']._serialized_options = b'\302\363\030\"\n\017genomic-region-\022\017Genomic Region '
  _GENEDESCRIPTOR.fields_by_name['chromosomes']._options = None
  _GENEDESCRIPTOR.fields_by_name['chromosomes']._serialized_options = b'\302\363\030\"\n\013chromosomes\022\013Chromosomes2\00112\003X,Y'
  _GENEDESCRIPTOR.fields_by_name['nomenclature_authority']._options = None
  _GENEDESCRIPTOR.fields_by_name['nomenclature_authority']._serialized_options = b'\302\363\030\026\n\005name-\022\rNomenclature '
  _GENEDESCRIPTOR.fields_by_name['swiss_prot_accessions']._options = None
  _GENEDESCRIPTOR.fields_by_name['swiss_prot_accessions']._serialized_options = b'\302\363\030,\n\024swissprot-accessions\022\024SwissProt Accessions'
  _GENEDESCRIPTOR.fields_by_name['ensembl_gene_ids']._options = None
  _GENEDESCRIPTOR.fields_by_name['ensembl_gene_ids']._serialized_options = b'\302\363\030\"\n\017ensembl-geneids\022\017Ensembl GeneIDs'
  _GENEDESCRIPTOR.fields_by_name['omim_ids']._options = None
  _GENEDESCRIPTOR.fields_by_name['omim_ids']._serialized_options = b'\302\363\030\024\n\010omim-ids\022\010OMIM IDs'
  _GENEDESCRIPTOR.fields_by_name['synonyms']._options = None
  _GENEDESCRIPTOR.fields_by_name['synonyms']._serialized_options = b'\302\363\030\024\n\010synonyms\022\010Synonyms'
  _GENEDESCRIPTOR.fields_by_name['replaced_gene_id']._options = None
  _GENEDESCRIPTOR.fields_by_name['replaced_gene_id']._serialized_options = b'\302\363\030(\n\020replaced-gene-id\022\024Replaced NCBI GeneID'
  _GENEDESCRIPTOR.fields_by_name['annotations']._options = None
  _GENEDESCRIPTOR.fields_by_name['annotations']._serialized_options = b'\302\363\030\032\n\013annotation-\022\013Annotation '
  _GENEDESCRIPTOR.fields_by_name['transcript_count']._options = None
  _GENEDESCRIPTOR.fields_by_name['transcript_count']._serialized_options = b'\302\363\030\037\n\020transcript-count\022\013Transcripts'
  _GENEDESCRIPTOR.fields_by_name['protein_count']._options = None
  _GENEDESCRIPTOR.fields_by_name['protein_count']._serialized_options = b'\302\363\030\031\n\rprotein-count\022\010Proteins'
  _GENEDESCRIPTOR.fields_by_name['gene_groups']._options = None
  _GENEDESCRIPTOR.fields_by_name['gene_groups']._serialized_options = b'\302\363\030\025\n\006group-\022\013Gene Group '
  _GENEGROUP.fields_by_name['id']._options = None
  _GENEGROUP.fields_by_name['id']._serialized_options = b'\302\363\030\020\n\002id\022\nIdentifier'
  _GENEGROUP.fields_by_name['method']._options = None
  _GENEGROUP.fields_by_name['method']._serialized_options = b'\302\363\030\020\n\006method\022\006Method'
  _NOMENCLATUREAUTHORITY.fields_by_name['authority']._options = None
  _NOMENCLATUREAUTHORITY.fields_by_name['authority']._serialized_options = b'\302\363\030\034\n\tauthority\022\tAuthority2\004HGNC'
  _NOMENCLATUREAUTHORITY.fields_by_name['identifier']._options = None
  _NOMENCLATUREAUTHORITY.fields_by_name['identifier']._serialized_options = b'\302\363\030\023\n\002id\022\002ID2\tHGNC:4392'
  _GENEDATAREPORTPAGE._serialized_start=186
  _GENEDATAREPORTPAGE._serialized_end=557
  _ORTHOLOGSET._serialized_start=560
  _ORTHOLOGSET._serialized_end=705
  _GENEREPORTMETADATA._serialized_start=708
  _GENEREPORTMETADATA._serialized_end=889
  _GENEREPORTMATCH._serialized_start=892
  _GENEREPORTMATCH._serialized_end=1259
  _GENOMICREGION._serialized_start=1262
  _GENOMICREGION._serialized_end=1607
  _GENOMICREGION_GENOMICREGIONTYPE._serialized_start=1508
  _GENOMICREGION_GENOMICREGIONTYPE._serialized_end=1607
  _ANNOTATION._serialized_start=1610
  _ANNOTATION._serialized_end=2079
  _GENEDESCRIPTOR._serialized_start=2082
  _GENEDESCRIPTOR._serialized_end=4064
  _GENEGROUP._serialized_start=4066
  _GENEGROUP._serialized_end=4161
  _NOMENCLATUREAUTHORITY._serialized_start=4164
  _NOMENCLATUREAUTHORITY._serialized_end=4308
  _GENEDESCRIPTORS._serialized_start=4310
  _GENEDESCRIPTORS._serialized_end=4391
# @@protoc_insertion_point(module_scope)
