# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: ncbi/datasets/v2/reports/virus_annotation.proto
"""Generated protocol buffer code."""
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import message as _message
from google.protobuf import reflection as _reflection
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()


from ncbi.datasets.v2.reports import common_pb2 as ncbi_dot_datasets_dot_v2_dot_reports_dot_common__pb2
from ncbi.datasets.options import report_pb2 as ncbi_dot_datasets_dot_options_dot_report__pb2


DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n/ncbi/datasets/v2/reports/virus_annotation.proto\x12\x18ncbi.datasets.v2.reports\x1a%ncbi/datasets/v2/reports/common.proto\x1a\"ncbi/datasets/options/report.proto\"\xbe\x01\n\x0f\x43onservedDomain\x12\x38\n\taccession\x18\x01 \x01(\tB\x1a\xc2\xf3\x18\x16\n\taccession\x12\tAccessionR\taccession\x12$\n\x04name\x18\x02 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\x12K\n\x05range\x18\x03 \x01(\x0b\x32\x1f.ncbi.datasets.v2.reports.RangeB\x14\xc2\xf3\x18\x10\n\x06range-\x12\x06Range R\x05range\"\xbb\x02\n\x10SeqRangeSetFasta\x12\x30\n\x06seq_id\x18\x01 \x01(\tB\x19\xc2\xf3\x18\x15\n\x06seq-id\x12\x0bSequence IDR\x05seqId\x12G\n\x11\x61\x63\x63\x65ssion_version\x18\x02 \x01(\tB\x1a\xc2\xf3\x18\x16\n\taccession\x12\tAccessionR\x10\x61\x63\x63\x65ssionVersion\x12(\n\x05title\x18\x03 \x01(\tB\x12\xc2\xf3\x18\x0e\n\x05title\x12\x05TitleR\x05title\x12\x35\n\rsequence_hash\x18\x04 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04hash\x12\x04HashR\x0csequenceHash\x12K\n\x05range\x18\x05 \x03(\x0b\x32\x1f.ncbi.datasets.v2.reports.RangeB\x14\xc2\xf3\x18\x10\n\x06range-\x12\x06Range R\x05range\"\xa4\x08\n\x0cVirusPeptide\x12\x38\n\taccession\x18\x01 \x01(\tB\x1a\xc2\xf3\x18\x16\n\taccession\x12\tAccessionR\taccession\x12$\n\x04name\x18\x02 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\x12?\n\x0bother_names\x18\x03 \x03(\tB\x1e\xc2\xf3\x18\x1a\n\x0bother-names\x12\x0bOther NamesR\notherNames\x12o\n\nnucleotide\x18\x04 \x01(\x0b\x32*.ncbi.datasets.v2.reports.SeqRangeSetFastaB#\xc2\xf3\x18\x1f\n\nnuc-fasta-\x12\x11Nucleotide FASTA R\nnucleotide\x12j\n\x07protein\x18\x05 \x01(\x0b\x32*.ncbi.datasets.v2.reports.SeqRangeSetFastaB$\xc2\xf3\x18 \n\x0eprotein-fasta-\x12\x0eProtein FASTA R\x07protein\x12/\n\x07pdb_ids\x18\x06 \x03(\tB\x16\xc2\xf3\x18\x12\n\x07pdb-ids\x12\x07PDB IDsR\x06pdbIds\x12M\n\x03\x63\x64\x64\x18\x07 \x03(\x0b\x32).ncbi.datasets.v2.reports.ConservedDomainB\x10\xc2\xf3\x18\x0c\n\x04\x63\x64\x64-\x12\x04\x43\x44\x44 R\x03\x63\x64\x64\x12i\n\x0buni_prot_kb\x18\x08 \x01(\x0b\x32\x30.ncbi.datasets.v2.reports.VirusPeptide.UniProtIdB\x17\xc2\xf3\x18\x13\n\x08uniprot-\x12\x07UniProtR\tuniProtKb\x12r\n\x0emature_peptide\x18\t \x03(\x0b\x32&.ncbi.datasets.v2.reports.VirusPeptideB#\xc2\xf3\x18\x1f\n\x0cmat-peptide-\x12\x0fMature Peptide R\rmaturePeptide\x12\xa1\x01\n\x14protein_completeness\x18\n \x01(\x0e\x32?.ncbi.datasets.v2.reports.VirusPeptide.ViralPeptideCompletenessB-\xc2\xf3\x18)\n\x11prot-completeness\x12\x14Protein CompletenessR\x13proteinCompleteness\x1aO\n\tUniProtId\x12\x1c\n\x02id\x18\x01 \x01(\tB\x0c\xc2\xf3\x18\x08\n\x02id\x12\x02IDR\x02id\x12$\n\x04name\x18\x02 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\"B\n\x18ViralPeptideCompleteness\x12\x0b\n\x07UNKNOWN\x10\x00\x12\x0c\n\x08\x43OMPLETE\x10\x01\x12\x0b\n\x07PARTIAL\x10\x02\"\x98\x02\n\tVirusGene\x12$\n\x04name\x18\x01 \x01(\tB\x10\xc2\xf3\x18\x0c\n\x04name\x12\x04NameR\x04name\x12\x33\n\x07gene_id\x18\x02 \x01(\rB\x1a\xc2\xf3\x18\x16\n\x07gene-id\x12\x0bNCBI GeneIDR\x06geneId\x12\x64\n\nnucleotide\x18\x03 \x01(\x0b\x32*.ncbi.datasets.v2.reports.SeqRangeSetFastaB\x18\xc2\xf3\x18\x14\n\x08genomic-\x12\x08Genomic R\nnucleotide\x12J\n\x03\x63\x64s\x18\x04 \x03(\x0b\x32&.ncbi.datasets.v2.reports.VirusPeptideB\x10\xc2\xf3\x18\x0c\n\x04\x63\x64s-\x12\x04\x43\x44S R\x03\x63\x64s\"\xf2\x01\n\x15VirusAnnotationReport\x12\x45\n\taccession\x18\x01 \x01(\tB\'\xc2\xf3\x18#\n\taccession\x12\tAccession2\x0bNC_045512.2R\taccession\x12\x43\n\x0cisolate_name\x18\x02 \x01(\tB \xc2\xf3\x18\x1c\n\x0cisolate-name\x12\x0cIsolate nameR\x0bisolateName\x12M\n\x05genes\x18\x03 \x03(\x0b\x32#.ncbi.datasets.v2.reports.VirusGeneB\x12\xc2\xf3\x18\x0e\n\x05gene-\x12\x05Gene R\x05genes\"\xc1\x02\n\x19VirusAnnotationReportPage\x12I\n\x07reports\x18\x01 \x03(\x0b\x32/.ncbi.datasets.v2.reports.VirusAnnotationReportR\x07reports\x12\x1f\n\x0btotal_count\x18\x02 \x01(\rR\ntotalCount\x12&\n\x0fnext_page_token\x18\x03 \x01(\tR\rnextPageToken\x12!\n\x0c_report_type\x18\xf4\x03 \x01(\tR\nReportType\x12%\n\x0e_report_fields\x18\xf5\x03 \x03(\tR\x0cReportFields\x12\x1f\n\x0b_first_page\x18\xf6\x03 \x01(\x08R\tFirstPage\x12%\n\x0e_report_format\x18\xf7\x03 \x01(\tR\x0cReportFormatB\xbe\nZ\x18ncbi/datasets/v2/reports\xf8\x01\x01\xc2\xf3\x18\x9c\n\n\x17Virus Annotation Report\x12\x15VirusAnnotationReport\x1a\x45<p>The downloaded virus package contains a virus annotation report in\x1a:<a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\x1aUformat in the file:<br><br><em>ncbi_dataset/data/annotation_report.jsonl</em><br><br>\x1atEach line of the virus annotation report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\x1aqobject that represents a single virus record. The schema of the virus record is defined in the tables below where\x1a\x64\x65\x61\x63h row describes a single field in the report or a sub-structure, which is a collection of fields.\x1a\x44The outermost structure of the report is <em>VirusAssembly</em>.</p>\x1aQ<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\x1a\x66<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s\x1a\x37<nobr><code>--fields</code></nobr> option. Refer to the\x1al<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see\x1aghow you can use this tool to transform virus annotation reports from JSON Lines to tabular formats.</p>\"\x0bSeqRangeSet*\xa2\x01{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/sars2-virus-report.cli.output\" begin=\"START annotation-report\" end=\"END annotation-report\">}}b\x06proto3')



_CONSERVEDDOMAIN = DESCRIPTOR.message_types_by_name['ConservedDomain']
_SEQRANGESETFASTA = DESCRIPTOR.message_types_by_name['SeqRangeSetFasta']
_VIRUSPEPTIDE = DESCRIPTOR.message_types_by_name['VirusPeptide']
_VIRUSPEPTIDE_UNIPROTID = _VIRUSPEPTIDE.nested_types_by_name['UniProtId']
_VIRUSGENE = DESCRIPTOR.message_types_by_name['VirusGene']
_VIRUSANNOTATIONREPORT = DESCRIPTOR.message_types_by_name['VirusAnnotationReport']
_VIRUSANNOTATIONREPORTPAGE = DESCRIPTOR.message_types_by_name['VirusAnnotationReportPage']
_VIRUSPEPTIDE_VIRALPEPTIDECOMPLETENESS = _VIRUSPEPTIDE.enum_types_by_name['ViralPeptideCompleteness']
ConservedDomain = _reflection.GeneratedProtocolMessageType('ConservedDomain', (_message.Message,), {
  'DESCRIPTOR' : _CONSERVEDDOMAIN,
  '__module__' : 'ncbi.datasets.v2.reports.virus_annotation_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.ConservedDomain)
  })
_sym_db.RegisterMessage(ConservedDomain)

SeqRangeSetFasta = _reflection.GeneratedProtocolMessageType('SeqRangeSetFasta', (_message.Message,), {
  'DESCRIPTOR' : _SEQRANGESETFASTA,
  '__module__' : 'ncbi.datasets.v2.reports.virus_annotation_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.SeqRangeSetFasta)
  })
_sym_db.RegisterMessage(SeqRangeSetFasta)

VirusPeptide = _reflection.GeneratedProtocolMessageType('VirusPeptide', (_message.Message,), {

  'UniProtId' : _reflection.GeneratedProtocolMessageType('UniProtId', (_message.Message,), {
    'DESCRIPTOR' : _VIRUSPEPTIDE_UNIPROTID,
    '__module__' : 'ncbi.datasets.v2.reports.virus_annotation_pb2'
    # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.VirusPeptide.UniProtId)
    })
  ,
  'DESCRIPTOR' : _VIRUSPEPTIDE,
  '__module__' : 'ncbi.datasets.v2.reports.virus_annotation_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.VirusPeptide)
  })
_sym_db.RegisterMessage(VirusPeptide)
_sym_db.RegisterMessage(VirusPeptide.UniProtId)

VirusGene = _reflection.GeneratedProtocolMessageType('VirusGene', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSGENE,
  '__module__' : 'ncbi.datasets.v2.reports.virus_annotation_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.VirusGene)
  })
_sym_db.RegisterMessage(VirusGene)

VirusAnnotationReport = _reflection.GeneratedProtocolMessageType('VirusAnnotationReport', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSANNOTATIONREPORT,
  '__module__' : 'ncbi.datasets.v2.reports.virus_annotation_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.VirusAnnotationReport)
  })
_sym_db.RegisterMessage(VirusAnnotationReport)

VirusAnnotationReportPage = _reflection.GeneratedProtocolMessageType('VirusAnnotationReportPage', (_message.Message,), {
  'DESCRIPTOR' : _VIRUSANNOTATIONREPORTPAGE,
  '__module__' : 'ncbi.datasets.v2.reports.virus_annotation_pb2'
  # @@protoc_insertion_point(class_scope:ncbi.datasets.v2.reports.VirusAnnotationReportPage)
  })
_sym_db.RegisterMessage(VirusAnnotationReportPage)

if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  DESCRIPTOR._serialized_options = b'Z\030ncbi/datasets/v2/reports\370\001\001\302\363\030\234\n\n\027Virus Annotation Report\022\025VirusAnnotationReport\032E<p>The downloaded virus package contains a virus annotation report in\032:<a href=\"https://jsonlines.readthedocs.io/\">JSON Lines</a>\032Uformat in the file:<br><br><em>ncbi_dataset/data/annotation_report.jsonl</em><br><br>\032tEach line of the virus annotation report file is a hierarchical <a href=\"https://www.json.org/json-en.html\">JSON</a>\032qobject that represents a single virus record. The schema of the virus record is defined in the tables below where\032deach row describes a single field in the report or a sub-structure, which is a collection of fields.\032DThe outermost structure of the report is <em>VirusAssembly</em>.</p>\032Q<p>Table fields that include a <em>Table Field Mnemonic</em> can be used with the\032f<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> command-line tool\'s\0327<nobr><code>--fields</code></nobr> option. Refer to the\032l<a href=\"/datasets/docs/v2/reference-docs/command-line/dataformat/\">dataformat</a> CLI tool reference to see\032ghow you can use this tool to transform virus annotation reports from JSON Lines to tabular formats.</p>\"\013SeqRangeSet*\242\001{{< readfile-code-block lang=\"json\" file=\"/docs_samples/datasets-cli/sars2-virus-report.cli.output\" begin=\"START annotation-report\" end=\"END annotation-report\">}}'
  _CONSERVEDDOMAIN.fields_by_name['accession']._options = None
  _CONSERVEDDOMAIN.fields_by_name['accession']._serialized_options = b'\302\363\030\026\n\taccession\022\tAccession'
  _CONSERVEDDOMAIN.fields_by_name['name']._options = None
  _CONSERVEDDOMAIN.fields_by_name['name']._serialized_options = b'\302\363\030\014\n\004name\022\004Name'
  _CONSERVEDDOMAIN.fields_by_name['range']._options = None
  _CONSERVEDDOMAIN.fields_by_name['range']._serialized_options = b'\302\363\030\020\n\006range-\022\006Range '
  _SEQRANGESETFASTA.fields_by_name['seq_id']._options = None
  _SEQRANGESETFASTA.fields_by_name['seq_id']._serialized_options = b'\302\363\030\025\n\006seq-id\022\013Sequence ID'
  _SEQRANGESETFASTA.fields_by_name['accession_version']._options = None
  _SEQRANGESETFASTA.fields_by_name['accession_version']._serialized_options = b'\302\363\030\026\n\taccession\022\tAccession'
  _SEQRANGESETFASTA.fields_by_name['title']._options = None
  _SEQRANGESETFASTA.fields_by_name['title']._serialized_options = b'\302\363\030\016\n\005title\022\005Title'
  _SEQRANGESETFASTA.fields_by_name['sequence_hash']._options = None
  _SEQRANGESETFASTA.fields_by_name['sequence_hash']._serialized_options = b'\302\363\030\014\n\004hash\022\004Hash'
  _SEQRANGESETFASTA.fields_by_name['range']._options = None
  _SEQRANGESETFASTA.fields_by_name['range']._serialized_options = b'\302\363\030\020\n\006range-\022\006Range '
  _VIRUSPEPTIDE_UNIPROTID.fields_by_name['id']._options = None
  _VIRUSPEPTIDE_UNIPROTID.fields_by_name['id']._serialized_options = b'\302\363\030\010\n\002id\022\002ID'
  _VIRUSPEPTIDE_UNIPROTID.fields_by_name['name']._options = None
  _VIRUSPEPTIDE_UNIPROTID.fields_by_name['name']._serialized_options = b'\302\363\030\014\n\004name\022\004Name'
  _VIRUSPEPTIDE.fields_by_name['accession']._options = None
  _VIRUSPEPTIDE.fields_by_name['accession']._serialized_options = b'\302\363\030\026\n\taccession\022\tAccession'
  _VIRUSPEPTIDE.fields_by_name['name']._options = None
  _VIRUSPEPTIDE.fields_by_name['name']._serialized_options = b'\302\363\030\014\n\004name\022\004Name'
  _VIRUSPEPTIDE.fields_by_name['other_names']._options = None
  _VIRUSPEPTIDE.fields_by_name['other_names']._serialized_options = b'\302\363\030\032\n\013other-names\022\013Other Names'
  _VIRUSPEPTIDE.fields_by_name['nucleotide']._options = None
  _VIRUSPEPTIDE.fields_by_name['nucleotide']._serialized_options = b'\302\363\030\037\n\nnuc-fasta-\022\021Nucleotide FASTA '
  _VIRUSPEPTIDE.fields_by_name['protein']._options = None
  _VIRUSPEPTIDE.fields_by_name['protein']._serialized_options = b'\302\363\030 \n\016protein-fasta-\022\016Protein FASTA '
  _VIRUSPEPTIDE.fields_by_name['pdb_ids']._options = None
  _VIRUSPEPTIDE.fields_by_name['pdb_ids']._serialized_options = b'\302\363\030\022\n\007pdb-ids\022\007PDB IDs'
  _VIRUSPEPTIDE.fields_by_name['cdd']._options = None
  _VIRUSPEPTIDE.fields_by_name['cdd']._serialized_options = b'\302\363\030\014\n\004cdd-\022\004CDD '
  _VIRUSPEPTIDE.fields_by_name['uni_prot_kb']._options = None
  _VIRUSPEPTIDE.fields_by_name['uni_prot_kb']._serialized_options = b'\302\363\030\023\n\010uniprot-\022\007UniProt'
  _VIRUSPEPTIDE.fields_by_name['mature_peptide']._options = None
  _VIRUSPEPTIDE.fields_by_name['mature_peptide']._serialized_options = b'\302\363\030\037\n\014mat-peptide-\022\017Mature Peptide '
  _VIRUSPEPTIDE.fields_by_name['protein_completeness']._options = None
  _VIRUSPEPTIDE.fields_by_name['protein_completeness']._serialized_options = b'\302\363\030)\n\021prot-completeness\022\024Protein Completeness'
  _VIRUSGENE.fields_by_name['name']._options = None
  _VIRUSGENE.fields_by_name['name']._serialized_options = b'\302\363\030\014\n\004name\022\004Name'
  _VIRUSGENE.fields_by_name['gene_id']._options = None
  _VIRUSGENE.fields_by_name['gene_id']._serialized_options = b'\302\363\030\026\n\007gene-id\022\013NCBI GeneID'
  _VIRUSGENE.fields_by_name['nucleotide']._options = None
  _VIRUSGENE.fields_by_name['nucleotide']._serialized_options = b'\302\363\030\024\n\010genomic-\022\010Genomic '
  _VIRUSGENE.fields_by_name['cds']._options = None
  _VIRUSGENE.fields_by_name['cds']._serialized_options = b'\302\363\030\014\n\004cds-\022\004CDS '
  _VIRUSANNOTATIONREPORT.fields_by_name['accession']._options = None
  _VIRUSANNOTATIONREPORT.fields_by_name['accession']._serialized_options = b'\302\363\030#\n\taccession\022\tAccession2\013NC_045512.2'
  _VIRUSANNOTATIONREPORT.fields_by_name['isolate_name']._options = None
  _VIRUSANNOTATIONREPORT.fields_by_name['isolate_name']._serialized_options = b'\302\363\030\034\n\014isolate-name\022\014Isolate name'
  _VIRUSANNOTATIONREPORT.fields_by_name['genes']._options = None
  _VIRUSANNOTATIONREPORT.fields_by_name['genes']._serialized_options = b'\302\363\030\016\n\005gene-\022\005Gene '
  _CONSERVEDDOMAIN._serialized_start=153
  _CONSERVEDDOMAIN._serialized_end=343
  _SEQRANGESETFASTA._serialized_start=346
  _SEQRANGESETFASTA._serialized_end=661
  _VIRUSPEPTIDE._serialized_start=664
  _VIRUSPEPTIDE._serialized_end=1724
  _VIRUSPEPTIDE_UNIPROTID._serialized_start=1577
  _VIRUSPEPTIDE_UNIPROTID._serialized_end=1656
  _VIRUSPEPTIDE_VIRALPEPTIDECOMPLETENESS._serialized_start=1658
  _VIRUSPEPTIDE_VIRALPEPTIDECOMPLETENESS._serialized_end=1724
  _VIRUSGENE._serialized_start=1727
  _VIRUSGENE._serialized_end=2007
  _VIRUSANNOTATIONREPORT._serialized_start=2010
  _VIRUSANNOTATIONREPORT._serialized_end=2252
  _VIRUSANNOTATIONREPORTPAGE._serialized_start=2255
  _VIRUSANNOTATIONREPORTPAGE._serialized_end=2576
# @@protoc_insertion_point(module_scope)
